{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "91d595f9-b422-4ae7-96fe-30c080dd0777",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from langchain_community.document_loaders import PyPDFLoader\n",
    "from langchain_experimental.text_splitter import SemanticChunker\n",
    "from langchain_community.embeddings import HuggingFaceEmbeddings\n",
    "from langchain.schema import Document\n",
    "from transformers import AutoTokenizer\n",
    "import torch\n",
    "import requests\n",
    "from urllib.parse import urljoin\n",
    "from urllib.parse import urlencode\n",
    "import os, re, io, time, json, html\n",
    "import requests\n",
    "from pathlib import Path\n",
    "from urllib.parse import urljoin\n",
    "from PIL import Image, UnidentifiedImageError\n",
    "\n",
    "from selenium import webdriver\n",
    "from selenium.webdriver.chrome.options import Options\n",
    "from selenium.webdriver.common.by import By\n",
    "from selenium.webdriver.support.ui import WebDriverWait\n",
    "from selenium.webdriver.support import expected_conditions as EC\n",
    "import re\n",
    "import xml.etree.ElementTree as ET\n",
    "from typing import List, Dict, Optional\n",
    "\n",
    "\n",
    "import numpy as np\n",
    "from sentence_transformers import SentenceTransformer\n",
    "import chromadb\n",
    "from chromadb.config import Settings\n",
    "import uuid\n",
    "from typing import List, Dict, Any, Tuple\n",
    "from sklearn.metrics.pairwise import cosine_similarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1d622bb9-7e29-42cf-b0f0-86fe1cea3d61",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CUDA available: True\n",
      "Device: NVIDIA GeForce RTX 2070 SUPER\n"
     ]
    }
   ],
   "source": [
    "print(\"CUDA available:\", torch.cuda.is_available())\n",
    "print(\"Device:\", torch.cuda.get_device_name(0) if torch.cuda.is_available() else \"CPU only\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35efd180-b0d7-45e8-afbc-6e303861fcbe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "91a36f50-c636-4228-9bb4-7f884e698c83",
   "metadata": {},
   "source": [
    "# PubMed\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "896ee594-f1a0-4c3b-9932-f03045015908",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmcids = [\"PMC9866905\",\"PMC12449471\",\"PMC7037937\",\"PMC7411343\",\"PMC12449476\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "08ac2e41-7fdb-4975-ad29-212d7a81a711",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "pmcid = \"PMC9866905\"\n",
    "url = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi\"\n",
    "params = {\"db\": \"pmc\", \"id\": pmcid, \"retmode\": \"xml\"}\n",
    "r = requests.get(url, params=params)\n",
    "fulltext_xml = r.text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "19f7b2a9-ba8f-4969-bc0e-2d47247e6969",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\"  ?><!DOCTYPE pmc-articleset PUBLIC \"-//NLM//DTD ARTICLE SET 2.0//EN\" \"https://dtd.nlm.nih.gov/ncbi/pmc/articleset/nlm-articleset-2.0.dtd\"><pmc-articleset><article xmlns:mml=\"http://www.w3.org/1998/Math/MathML\" xmlns:ali=\"http://www.niso.org/schemas/ali/1.0/\" xml:lang=\"en\" article-type=\"research-article\" dtd-version=\"1.4\"><processing-meta base-tagset=\"archiving\" mathml-version=\"3.0\" table-model=\"xhtml\" tagset-family=\"jats\"><restricted-by>pmc</restricted-by></processing-meta><front><journal-meta><journal-id journal-id-type=\"nlm-ta\">Pharmaceuticals (Basel)</journal-id><journal-id journal-id-type=\"iso-abbrev\">Pharmaceuticals (Basel)</journal-id><journal-id journal-id-type=\"pmc-domain-id\">2102</journal-id><journal-id journal-id-type=\"pmc-domain\">pharmaceuticals</journal-id><journal-id journal-id-type=\"publisher-id\">pharmaceuticals</journal-id><journal-title-group><journal-title>Pharmaceuticals</journal-title></journal-title-group><issn pub-type=\"epub\">1424-8247</issn><publisher><publisher-name>Multidisciplinary Digital Publishing Institute  (MDPI)</publisher-name></publisher></journal-meta><article-meta><article-id pub-id-type=\"pmcid\">PMC9866905</article-id><article-id pub-id-type=\"pmcid-ver\">PMC9866905.1</article-id><article-id pub-id-type=\"pmcaid\">9866905</article-id><article-id pub-id-type=\"pmcaiid\">9866905</article-id><article-id pub-id-type=\"pmid\">36678511</article-id><article-id pub-id-type=\"doi\">10.3390/ph16010014</article-id><article-id pub-id-type=\"publisher-id\">pharmaceuticals-16-00014</article-id><article-version article-version-type=\"pmc-version\">1</article-version><article-categories><subj-group subj-group-type=\"heading\"><subject>Article</subject></subj-group></article-categories><title-group><article-title>Dehydroeburicoic Acid, a Dual Inhibitor against Oxidative Stress in Alcoholic Liver Disease</article-title></title-group><contrib-group><contrib contrib-type=\"author\"><name name-style=\"western\"><surname>Cheng</surname><given-names initials=\"S\">Shasha</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Conceptualization\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/conceptualization/\">Conceptualization</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Methodology\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/methodology/\">Methodology</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Formal analysis\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/formal-analysis/\">Formal analysis</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Investigation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/investigation/\">Investigation</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Data curation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/data-curation/\">Data curation</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Writing &#x2013; original draft\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/writing-original-draft/\">Writing &#8211; original draft</role><xref rid=\"af1-pharmaceuticals-16-00014\" ref-type=\"aff\">1</xref><xref rid=\"fn1-pharmaceuticals-16-00014\" ref-type=\"author-notes\">&#8224;</xref></contrib><contrib contrib-type=\"author\"><name name-style=\"western\"><surname>Kuang</surname><given-names initials=\"Y\">Yi</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Methodology\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/methodology/\">Methodology</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Software\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/software/\">Software</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Formal analysis\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/formal-analysis/\">Formal analysis</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Investigation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/investigation/\">Investigation</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Resources\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/resources/\">Resources</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Writing &#x2013; review &amp; editing\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/writing-review-editing/\">Writing &#8211; review &amp; editing</role><xref rid=\"af2-pharmaceuticals-16-00014\" ref-type=\"aff\">2</xref><xref rid=\"fn1-pharmaceuticals-16-00014\" ref-type=\"author-notes\">&#8224;</xref></contrib><contrib contrib-type=\"author\"><name name-style=\"western\"><surname>Li</surname><given-names initials=\"G\">Guodong</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Methodology\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/methodology/\">Methodology</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Validation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/validation/\">Validation</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Formal analysis\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/formal-analysis/\">Formal analysis</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Data curation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/data-curation/\">Data curation</role><xref rid=\"af1-pharmaceuticals-16-00014\" ref-type=\"aff\">1</xref><xref rid=\"af3-pharmaceuticals-16-00014\" ref-type=\"aff\">3</xref></contrib><contrib contrib-type=\"author\"><name name-style=\"western\"><surname>Wu</surname><given-names initials=\"J\">Jia</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Methodology\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/methodology/\">Methodology</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Formal analysis\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/formal-analysis/\">Formal analysis</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Investigation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/investigation/\">Investigation</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Data curation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/data-curation/\">Data curation</role><xref rid=\"af1-pharmaceuticals-16-00014\" ref-type=\"aff\">1</xref></contrib><contrib contrib-type=\"author\"><name name-style=\"western\"><surname>Ko</surname><given-names initials=\"CN\">Chung-Nga</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Methodology\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/methodology/\">Methodology</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Software\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/software/\">Software</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Data curation\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/data-curation/\">Data curation</role><xref rid=\"af4-pharmaceuticals-16-00014\" ref-type=\"aff\">4</xref></contrib><contrib contrib-type=\"author\"><name name-style=\"western\"><surname>Wang</surname><given-names initials=\"W\">Wanhe</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Methodology\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/methodology/\">Methodology</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Software\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/software/\">Software</role><xref rid=\"af4-pharmaceuticals-16-00014\" ref-type=\"aff\">4</xref><xref rid=\"af5-pharmaceuticals-16-00014\" ref-type=\"aff\">5</xref></contrib><contrib contrib-type=\"author\"><contrib-id contrib-id-type=\"orcid\" authenticated=\"true\">https://orcid.org/0000-0002-9515-340X</contrib-id><name name-style=\"western\"><surname>Ma</surname><given-names initials=\"DL\">Dik-Lung</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Software\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/software/\">Software</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Supervision\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/supervision/\">Supervision</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Project administration\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/project-administration/\">Project administration</role><xref rid=\"af4-pharmaceuticals-16-00014\" ref-type=\"aff\">4</xref><xref rid=\"c1-pharmaceuticals-16-00014\" ref-type=\"corresp\">*</xref></contrib><contrib contrib-type=\"author\"><name name-style=\"western\"><surname>Ye</surname><given-names initials=\"M\">Min</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Conceptualization\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/conceptualization/\">Conceptualization</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Software\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/software/\">Software</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Resources\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/resources/\">Resources</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Supervision\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/supervision/\">Supervision</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Project administration\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/project-administration/\">Project administration</role><xref rid=\"af2-pharmaceuticals-16-00014\" ref-type=\"aff\">2</xref><xref rid=\"c1-pharmaceuticals-16-00014\" ref-type=\"corresp\">*</xref></contrib><contrib contrib-type=\"author\"><contrib-id contrib-id-type=\"orcid\" authenticated=\"true\">https://orcid.org/0000-0003-2988-3786</contrib-id><name name-style=\"western\"><surname>Leung</surname><given-names initials=\"CH\">Chung-Hang</given-names></name><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Conceptualization\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/conceptualization/\">Conceptualization</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Visualization\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/visualization/\">Visualization</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Supervision\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/supervision/\">Supervision</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Project administration\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/project-administration/\">Project administration</role><role vocab=\"credit\" vocab-identifier=\"https://credit.niso.org/\" vocab-term=\"Funding acquisition\" vocab-term-identifier=\"https://credit.niso.org/contributor-roles/funding-acquisition/\">Funding acquisition</role><xref rid=\"af1-pharmaceuticals-16-00014\" ref-type=\"aff\">1</xref><xref rid=\"af3-pharmaceuticals-16-00014\" ref-type=\"aff\">3</xref><xref rid=\"af6-pharmaceuticals-16-00014\" ref-type=\"aff\">6</xref><xref rid=\"c1-pharmaceuticals-16-00014\" ref-type=\"corresp\">*</xref></contrib></contrib-group><contrib-group><contrib contrib-type=\"editor\"><name name-style=\"western\"><surname>Patil</surname><given-names initials=\"SP\">Sachin P.</given-names></name><role>Academic Editor</role></contrib></contrib-group><aff id=\"af1-pharmaceuticals-16-00014\"><label>1</label>State Key Laboratory of Quality Research in Chinese Medicine, Institute of Chinese Medical Sciences, University of Macau, Macau SAR 999078, China</aff><aff id=\"af2-pharmaceuticals-16-00014\"><label>2</label>State Key Laboratory of Natural and Biomimetic Drugs, School of Pharmaceutical Sciences, Peking University, Beijing 100191, China</aff><aff id=\"af3-pharmaceuticals-16-00014\"><label>3</label>Zhuhai UM Science and Technology Research Institute, Zhuhai 519031, China</aff><aff id=\"af4-pharmaceuticals-16-00014\"><label>4</label>Department of Chemistry, Hong Kong Baptist University, Hong Kong SAR 999077, China</aff><aff id=\"af5-pharmaceuticals-16-00014\"><label>5</label>Institute of Medical Research, Northwestern Polytechnical University, Xi&#8217;an 710072, China</aff><aff id=\"af6-pharmaceuticals-16-00014\"><label>6</label>Department of Biomedical Sciences, Faculty of Health Sciences, University of Macau, Macau SAR 999078, China</aff><author-notes><corresp id=\"c1-pharmaceuticals-16-00014\"><label>*</label>Correspondence: <email>edmondma@hkbu.edu.hk</email> (D.-L.M.); <email>yemin@bjmu.edu.cn</email> (M.Y.); <email>duncanleung@um.edu.mo</email> (C.-H.L.)</corresp><fn id=\"fn1-pharmaceuticals-16-00014\"><label>&#8224;</label><p>These authors contributed equally to this work.</p></fn></author-notes><pub-date pub-type=\"epub\"><day>22</day><month>12</month><year>2022</year></pub-date><pub-date pub-type=\"collection\"><month>1</month><year>2023</year></pub-date><volume>16</volume><issue>1</issue><issue-id pub-id-type=\"pmc-issue-id\">427075</issue-id><elocation-id>14</elocation-id><history><date date-type=\"received\"><day>15</day><month>11</month><year>2022</year></date><date date-type=\"rev-recd\"><day>04</day><month>12</month><year>2022</year></date><date date-type=\"accepted\"><day>08</day><month>12</month><year>2022</year></date></history><pub-history><event event-type=\"pmc-release\"><date><day>22</day><month>12</month><year>2022</year></date></event><event event-type=\"pmc-live\"><date><day>22</day><month>01</month><year>2023</year></date></event><event event-type=\"pmc-last-change\"><date iso-8601-date=\"2025-06-30 23:25:15.223\"><day>30</day><month>06</month><year>2025</year></date></event></pub-history><permissions><copyright-statement>&#169; 2022 by the authors.</copyright-statement><copyright-year>2022</copyright-year><license><ali:license_ref specific-use=\"textmining\" content-type=\"ccbylicense\">https://creativecommons.org/licenses/by/4.0/</ali:license_ref><license-p>Licensee MDPI, Basel, Switzerland. This article is an open access article distributed under the terms and conditions of the Creative Commons Attribution (CC BY) license (<ext-link xmlns:xlink=\"http://www.w3.org/1999/xlink\" ext-link-type=\"uri\" xlink:href=\"https://creativecommons.org/licenses/by/4.0/\">https://creativecommons.org/licenses/by/4.0/</ext-link>).</license-p></license></permissions><self-uri xmlns:xlink=\"http://www.w3.org/1999/xlink\" content-type=\"pmc-pdf\" xlink:href=\"pharmaceuticals-16-00014.pdf\"/><abstract><p>Alcoholic liver disease (ALD) is a complicated disease which can lead to hepatocellular carcinoma; however, there is a lack of satisfactory therapeutics. Dehydroeburicoic acid (DEA) (<bold>1</bold>), a triterpenoid isolated from <italic toggle=\"yes\">Antrodia cinnamomea</italic>, has been reported to act against ALD, but its mechanisms of action are still not clear. In this study, we report for the first time the use of DEA (<bold>1</bold>) as a dual inhibitor of the Keap1&#8211;Nrf2 protein&#8211;protein interaction (PPI) and GSK3&#946; in an in vitro ALD cell model. DEA (<bold>1</bold>) engages Keap1 to disrupt the Keap1&#8211;Nrf2 PPI and inhibits GSK3&#946; to restore Nrf2 activity in a Keap1-independent fashion. DEA (<bold>1</bold>) promotes Nrf2 nuclear translocation to activate downstream antioxidant genes. Importantly, DEA (<bold>1</bold>) restores the mitochondrial dysfunction induced by ethanol and generates antioxidant activity in the ALD cell model with minimal toxicity. We anticipate that DEA (<bold>1</bold>) could be a potential scaffold for the further development of clinical agents for treating ALD.</p></abstract><kwd-group><kwd>alcoholic liver disease (ALD)</kwd><kwd>Keap1&#8211;Nrf2 protein&#8211;protein interaction (PPI)</kwd><kwd>glycogen synthase kinase 3&#946; (GSK3&#946;)</kwd><kwd>hepatoprotective</kwd></kwd-group><funding-group><award-group><funding-source>Science and Technology Development Fund (Macau SAR, China)</funding-source><award-id>0007/2020/A1</award-id><award-id>0020/2022/A1</award-id></award-group><award-group><funding-source>State Key Laboratory of Quality Research in Chinese Medicine (University of Macau)</funding-source><award-id>SKL-QRCM(UM)-2020-2022</award-id></award-group><award-group><funding-source>University of Macau (University of Macau)</funding-source><award-id>MYRG2019-00002-ICMS</award-id><award-id>MYRG2020-00017-ICMS</award-id></award-group><award-group><funding-source>2022 Internal Research Grant of SKL-QRCM (University of Macau)</funding-source><award-id>22077109</award-id><award-id>21775131</award-id><award-id>82073715</award-id></award-group><award-group><funding-source>HKBU SKLEBA Research Grant</funding-source><award-id>SKLP_2223_P03</award-id></award-group><funding-statement>This work is supported by the Science and Technology Development Fund (Macau SAR, China) (0007/2020/A1, 0020/2022/A1), the State Key Laboratory of Quality Research in Chinese Medicine (University of Macau) (SKL-QRCM(UM)-2020-2022), the University of Macau (University of Macau) (MYRG2019-00002-ICMS, MYRG2020-00017-ICMS), 2022 Internal Research Grant of SKL-QRCM (University of Macau) (QRCM-IRG2022-011), the National Natural Science Foundation of China, China (22077109, 21775131 and 82073715), the HKBU SKLEBA Research Grant (SKLP_2223_P03). </funding-statement></funding-group><custom-meta-group><custom-meta><meta-name>pmc-status-qastatus</meta-name><meta-value>0</meta-value></custom-meta><custom-meta><meta-name>pmc-status-live</meta-name><meta-value>yes</meta-value></custom-meta><custom-meta><meta-name>pmc-status-embargo</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-status-released</meta-name><meta-value>yes</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-open-access</meta-name><meta-value>yes</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-olf</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-manuscript</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-legally-suppressed</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-has-pdf</meta-name><meta-value>yes</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-has-supplement</meta-name><meta-value>yes</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-pdf-only</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-suppress-copyright</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-is-real-version</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-is-scanned-article</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-preprint</meta-name><meta-value>no</meta-value></custom-meta><custom-meta><meta-name>pmc-prop-in-epmc</meta-name><meta-value>yes</meta-value></custom-meta><custom-meta><meta-name>pmc-license-ref</meta-name><meta-value>CC BY</meta-value></custom-meta></custom-meta-group></article-meta></front><body><sec sec-type=\"intro\" id=\"sec1-pharmaceuticals-16-00014\"><title>1. Introduction</title><p>Excessive alcohol consumption leads to oxidative stress in the liver as a result of alcohol metabolism, resulting in inflammatory damage and the injury of liver cells [<xref rid=\"B1-pharmaceuticals-16-00014\" ref-type=\"bibr\">1</xref>]. Almost 20% of alcoholics are eventually diagnosed with alcoholic liver disease (ALD) [<xref rid=\"B2-pharmaceuticals-16-00014\" ref-type=\"bibr\">2</xref>]. The most common therapeutic options for ALD can be divided into opioid receptor antagonists, supplements to modulate liver metabolism, and therapeutics that regulate alcohol metabolism [<xref rid=\"B3-pharmaceuticals-16-00014\" ref-type=\"bibr\">3</xref>]. However, these treatments have been associated with side effects including dizziness, drug dependence, dermatitis, vomiting, and leukopenia [<xref rid=\"B4-pharmaceuticals-16-00014\" ref-type=\"bibr\">4</xref>]. Thus, there are currently no highly satisfactory therapeutic options for ALD, and hence there is an unmet clinical need to develop more effective and safer drugs for patients with ALD.</p><p>Increasing evidence indicates that alcohol can damage mitochondria and destroy cellular homeostasis in the liver [<xref rid=\"B5-pharmaceuticals-16-00014\" ref-type=\"bibr\">5</xref>,<xref rid=\"B6-pharmaceuticals-16-00014\" ref-type=\"bibr\">6</xref>,<xref rid=\"B7-pharmaceuticals-16-00014\" ref-type=\"bibr\">7</xref>]. One mechanism that cells use to defend against alcohol&#8211;induced injury is through the activation of nuclear respiratory factor-2 (Nrf-2). Nrf2 orchestrates a complex, self-protective antioxidant response involving numerous signaling axes [<xref rid=\"B8-pharmaceuticals-16-00014\" ref-type=\"bibr\">8</xref>]. Upregulating Nrf2 has been envisioned as an effective tool for combating alcohol-induced acute liver injury [<xref rid=\"B9-pharmaceuticals-16-00014\" ref-type=\"bibr\">9</xref>,<xref rid=\"B10-pharmaceuticals-16-00014\" ref-type=\"bibr\">10</xref>]. Normally, the level of Nrf2 in the cytosol is low due to ubiquitination and proteasomal degradation regulated by a Kelch-like ECH-associated protein (Keap1) [<xref rid=\"B11-pharmaceuticals-16-00014\" ref-type=\"bibr\">11</xref>,<xref rid=\"B12-pharmaceuticals-16-00014\" ref-type=\"bibr\">12</xref>]. However, the presence of oxidants can oxidize cysteine residues in Keap1 into disulfides or conjugate them to electrophiles, resulting in the nuclear translocation of Nrf2 and the activation of antioxidant genes [<xref rid=\"B13-pharmaceuticals-16-00014\" ref-type=\"bibr\">13</xref>], including heme oxygenase-1 (<italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>), NAD(P)H dehydrogenase [quinone] 1 (<italic toggle=\"yes\">NQO1</italic>), and the mitochondrial superoxide dismutase 2 (<italic toggle=\"yes\">SOD2</italic>) [<xref rid=\"B14-pharmaceuticals-16-00014\" ref-type=\"bibr\">14</xref>,<xref rid=\"B15-pharmaceuticals-16-00014\" ref-type=\"bibr\">15</xref>]. In addition, Nrf2 activation increases the expression levels of nuclear respiratory factor-1 (Nrf-1) and peroxisome proliferator-activated receptor &#947; coactivator 1&#945; (PGC1&#945;), which in turn modulates the expression of mitochondrial respiratory subunits and translational components [<xref rid=\"B16-pharmaceuticals-16-00014\" ref-type=\"bibr\">16</xref>]. The genes activated by the antioxidant responsive element (ARE) protect cells against any damage caused by reactive oxygen species (ROS) [<xref rid=\"B17-pharmaceuticals-16-00014\" ref-type=\"bibr\">17</xref>]. In addition to Keap1, glycogen synthase kinase 3&#946; (GSK3&#946;) is another negative regulator of Nrf2 that acts to downregulate the antioxidant stress response [<xref rid=\"B18-pharmaceuticals-16-00014\" ref-type=\"bibr\">18</xref>,<xref rid=\"B19-pharmaceuticals-16-00014\" ref-type=\"bibr\">19</xref>]. GSK3&#946; promotes the nuclear exclusion and degradation of Nrf2 in a Keap1-independent fashion in stressed or injured cells with active Nrf2, thereby reducing the degree of protection conferred by Nrf2 activity [<xref rid=\"B20-pharmaceuticals-16-00014\" ref-type=\"bibr\">20</xref>,<xref rid=\"B21-pharmaceuticals-16-00014\" ref-type=\"bibr\">21</xref>]. Therefore, increasing the Nrf2 antioxidant activity via targeting Keap1 or GSK3&#946; could be a potential approach for treating ALD [<xref rid=\"B22-pharmaceuticals-16-00014\" ref-type=\"bibr\">22</xref>].</p><p>Most pharmacological approaches towards Nrf2 activation are focused on the inhibition of Keap1 [<xref rid=\"B23-pharmaceuticals-16-00014\" ref-type=\"bibr\">23</xref>,<xref rid=\"B24-pharmaceuticals-16-00014\" ref-type=\"bibr\">24</xref>]. Several Keap1 inhibitors have been reported in the last decade, and the vast majority of these covalently modify cysteine residues in Keap1 [<xref rid=\"B25-pharmaceuticals-16-00014\" ref-type=\"bibr\">25</xref>]. Covalently targeting cysteines in Keap1 may lack selectivity due to the presence of other reactive cysteine residues in the cell, resulting in adverse side effects [<xref rid=\"B26-pharmaceuticals-16-00014\" ref-type=\"bibr\">26</xref>,<xref rid=\"B27-pharmaceuticals-16-00014\" ref-type=\"bibr\">27</xref>]. However, because cysteine residues are abundant in cells, the safety and specificity of covalent drugs are a concern [<xref rid=\"B28-pharmaceuticals-16-00014\" ref-type=\"bibr\">28</xref>]. Recently, protein&#8211;protein interaction (PPI) inhibitors of Keap1 and Nrf2, developed based on the X-ray structure of Keap1, have emerged as a new class of Nrf2 activators [<xref rid=\"B29-pharmaceuticals-16-00014\" ref-type=\"bibr\">29</xref>,<xref rid=\"B30-pharmaceuticals-16-00014\" ref-type=\"bibr\">30</xref>,<xref rid=\"B31-pharmaceuticals-16-00014\" ref-type=\"bibr\">31</xref>]. Our group has previously reported a cyclometalated iridium(III) metal complex as a Keap1&#8211;Nrf2 PPI inhibitor, which is a promising therapeutic agent for acetaminophen&#8211;induced acute liver injury [<xref rid=\"B29-pharmaceuticals-16-00014\" ref-type=\"bibr\">29</xref>,<xref rid=\"B30-pharmaceuticals-16-00014\" ref-type=\"bibr\">30</xref>]. However, no Keap1&#8211;Nrf2 inhibitor has yet entered the clinic for the treatment of human diseases [<xref rid=\"B32-pharmaceuticals-16-00014\" ref-type=\"bibr\">32</xref>,<xref rid=\"B33-pharmaceuticals-16-00014\" ref-type=\"bibr\">33</xref>]. Meanwhile, although a few GSK3&#946; inhibitors have been reported for liver injury treatment, no GSK3&#946; inhibitor to date has made it to the market [<xref rid=\"B34-pharmaceuticals-16-00014\" ref-type=\"bibr\">34</xref>].</p><p><italic toggle=\"yes\">Antrodia cinnamomea</italic>, a basidiomycete that is endemic to Taiwan, is an edible fungus that is a component of many traditional herbal medicines [<xref rid=\"B35-pharmaceuticals-16-00014\" ref-type=\"bibr\">35</xref>,<xref rid=\"B36-pharmaceuticals-16-00014\" ref-type=\"bibr\">36</xref>]. Either crude extracts or molecules isolated from <italic toggle=\"yes\">A. cinnamomea</italic> have shown diverse biological properties, including antioxidant, anti-tumour, and anti-inflammatory activities [<xref rid=\"B37-pharmaceuticals-16-00014\" ref-type=\"bibr\">37</xref>,<xref rid=\"B38-pharmaceuticals-16-00014\" ref-type=\"bibr\">38</xref>]. Recently, research has shown that dehydroeburicoic acid (DEA) (<bold>1</bold>) from <italic toggle=\"yes\">A. cinnamomea</italic> could inhibit alcoholic fatty liver disease (AFLD) by upregulating aldehyde dehydrogenase 2 family member (ALDH2) activity [<xref rid=\"B39-pharmaceuticals-16-00014\" ref-type=\"bibr\">39</xref>,<xref rid=\"B40-pharmaceuticals-16-00014\" ref-type=\"bibr\">40</xref>]. Additionally, DEA (<bold>1</bold>) also exhibited protective effects against non-alcoholic fatty liver disease (NAFLD) through activating ALDH2 and accelerating the elimination of ROS and harmful aldehydes [<xref rid=\"B40-pharmaceuticals-16-00014\" ref-type=\"bibr\">40</xref>]. Therefore, DEA (<bold>1</bold>) has the potential ability to treat ALD. However, its mechanisms of hepatoprotection in ALD are still not clear. Besides DEA (<bold>1</bold>), other triterpenoids have also been studied as GSK3&#946; inhibitors and activators of Nrf2 because of their anti-inflammatory and antioxidant activities [<xref rid=\"B41-pharmaceuticals-16-00014\" ref-type=\"bibr\">41</xref>,<xref rid=\"B42-pharmaceuticals-16-00014\" ref-type=\"bibr\">42</xref>,<xref rid=\"B43-pharmaceuticals-16-00014\" ref-type=\"bibr\">43</xref>,<xref rid=\"B44-pharmaceuticals-16-00014\" ref-type=\"bibr\">44</xref>,<xref rid=\"B45-pharmaceuticals-16-00014\" ref-type=\"bibr\">45</xref>].</p><p>Research has shown that single&#8211;molecule drugs with two different biological activities may avoid some side effects and improve efficacy [<xref rid=\"B46-pharmaceuticals-16-00014\" ref-type=\"bibr\">46</xref>]. In this study, we discovered a potent dual Keap1&#8211;Nrf2 PPI and GSK3&#946; inhibitor, DEA (<bold>1</bold>), and investigated its underlying mechanisms of action. Through in vitro fluorescence polarization (FP) screening, we identified DEA (<bold>1</bold>) as the most potent candidate for disrupting the Keap1&#8211;Nrf2 PPI (EC<sub>50</sub> = 14.1 &#181;M) from thirteen analogues isolated from <italic toggle=\"yes\">A. cinnamomea</italic>. Moreover, DEA (<bold>1</bold>) inhibited GSK3&#946; kinase activity with an EC<sub>50</sub> value of 8.0 &#181;M. In cells, low-toxicity compound DEA (<bold>1</bold>) engaged Keap1 and GSK3&#946; to promote Nrf2 accumulation in the nucleus, leading to the upregulation of ARE transcriptional activity and increasing the expression of downstream antioxidant factors with more potent than the reported Keap1&#8211;Nrf2 PPI inhibitor ML334 in ALD model cells.</p></sec><sec sec-type=\"results\" id=\"sec2-pharmaceuticals-16-00014\"><title>2. Results</title><sec id=\"sec2dot1-pharmaceuticals-16-00014\"><title>2.1. DEA (<bold>1</bold>) Inhibits the Keap1&#8211;Nrf2 PPI and GSK3&#946; Activity</title><p>The fluorescence polarization assay is a commonly used high-throughput assay to detect PPI inhibitors in solution. We used labeled Nrf2 peptides, bearing the high&#8211;affinity ETGE motif, which is recognized by Keap1, as fluorescent tracers to monitor for inhibition of the Keap1&#8211;Nrf2 interaction. ML334 is the first non&#8211;covalent small molecule inhibitor of Keap1&#8211;Nrf2 interaction reported and is distinct from other Nrf2 inducers [<xref rid=\"B47-pharmaceuticals-16-00014\" ref-type=\"bibr\">47</xref>]. In this paper, ML334 was used as a positive control. Compounds <bold>1</bold>&#8211;<bold>13</bold> from <italic toggle=\"yes\">A. cinnamomea</italic> and ML334 were screened against the Keap1&#8211;Nrf2 interaction by FP (<xref rid=\"pharmaceuticals-16-00014-f001\" ref-type=\"fig\">Figure 1</xref>A). The preliminary screening results showed that DEA (<bold>1</bold>) was the most potent compound, which disrupted the Keap1&#8211;Nrf2 interaction by 62% inhibition at 50 &#181;M (cf. ML334: 60% inhibition). The NMR spectrum of DEA (<bold>1</bold>) is presented in <xref rid=\"app1-pharmaceuticals-16-00014\" ref-type=\"app\">Supplementary Figure S1</xref>. The potency of DEA (<bold>1</bold>) was further investigated in a dose-response assay, revealing an EC<sub>50</sub> value of 14.1 &#177; 0.1 &#181;M against Keap1&#8211;Nrf2 PPI (<xref rid=\"pharmaceuticals-16-00014-f001\" ref-type=\"fig\">Figure 1</xref>B). We also explored the inhibitory effect of DEA (<bold>1</bold>) against GSK3&#946;, which negatively regulates Nrf2 in a fashion independent of Keap1. 10 &#181;M of DEA (<bold>1</bold>) reduced GSK3&#946; activity by 58.9%, making it more potent than ML334 at the same concentration (11.3% inhibition) (<xref rid=\"pharmaceuticals-16-00014-f001\" ref-type=\"fig\">Figure 1</xref>C). Moreover, a dose-response assay revealed an EC<sub>50</sub> value of 8.0 &#177; 0.7 &#181;M against GSK3&#946; activity, indicating that DEA (<bold>1</bold>) is an effective GSK3&#946; inhibitor (<xref rid=\"pharmaceuticals-16-00014-f001\" ref-type=\"fig\">Figure 1</xref>D). In the nucleus, Nrf2 binds to the ARE to activate antioxidant gene transcription [<xref rid=\"B14-pharmaceuticals-16-00014\" ref-type=\"bibr\">14</xref>]. Thus, we tested the transcriptional activity of ARE after treatment of LO2 cells with DEA (<bold>1</bold>) and ML334 (10 &#181;M) using the dual luciferase assay kit (<xref rid=\"pharmaceuticals-16-00014-f001\" ref-type=\"fig\">Figure 1</xref>E). The activity of ARE was significantly increased by 54% in the treatment of DEA (<bold>1</bold>) compared to 27% in the ML334 group. The results showed that DEA (<bold>1</bold>) notably activates the transcriptional activity of ARE compared to ML334 in cellulo.</p></sec><sec id=\"sec2dot2-pharmaceuticals-16-00014\"><title>2.2. DEA (<bold>1</bold>) Exhibits Low Cytotoxicity In Cellulo</title><p>The cytotoxicity of compound DEA (<bold>1</bold>) was detected in both human normal cell lines LO2 and HEK 293T, and the hepatocellular carcinoma cell line HepG2. After treatment with different concentrations of DEA (<bold>1</bold>), no significant cytotoxicity was observed (<xref rid=\"pharmaceuticals-16-00014-f001\" ref-type=\"fig\">Figure 1</xref>F&#8211;H). These data indicate that DEA (<bold>1</bold>) could be potentially safe for treating ALD in vivo.</p></sec><sec id=\"sec2dot3-pharmaceuticals-16-00014\"><title>2.3. DEA (<bold>1</bold>) as a Dual Inhibitor of Keap1 and GSK3&#946;</title><p>The ability of DEA (<bold>1</bold>) to disrupt the Keap1&#8211;Nrf2 binding in cellulo was monitored through a Co&#8211;IP experiment using human normal liver cell line LO2 cells (<xref rid=\"pharmaceuticals-16-00014-f002\" ref-type=\"fig\">Figure 2</xref>A). After treatment with DEA (<bold>1</bold>) (10 &#181;M) or ML334 (10 &#181;M) for 8 h, there was a 45% reduction of Keap1 co-precipitated with Nrf2 for DEA (<bold>1</bold>) as compared to 12% inhibition for ML334, indicating that DEA (<bold>1</bold>) could more effectively inhibit the Keap1&#8211;Nrf2 interaction compared to ML334 in living cells (<xref rid=\"pharmaceuticals-16-00014-f002\" ref-type=\"fig\">Figure 2</xref>B).</p><p>We investigated the ability of DEA (<bold>1</bold>) to target Keap1, Nrf2, and GSK3&#946; in the cellular environment using CETSA. LO2 cell lysates were incubated with 10 &#181;M of DEA (<bold>1</bold>) at room temperature for 30 min. Then, aliquots were heated individually at different temperatures and the protein in the soluble fraction was quantified by WB. Compound DEA (<bold>1</bold>) significantly stabilized Keap1 (<italic toggle=\"yes\">&#916;T<sub>m</sub></italic>: ca. 4.5 &#176;C) and GSK3&#946; (<italic toggle=\"yes\">&#916;T<sub>m</sub></italic>: ca. 3.9 &#176;C), while having no observable effect on the thermal stability of Nrf2 and &#946;&#8211;actin (<xref rid=\"pharmaceuticals-16-00014-f002\" ref-type=\"fig\">Figure 2</xref>C,D). This result suggests that DEA (<bold>1</bold>) can bind with Keap1 and GSK3&#946;, even within the complicated environment of cell lysates.</p></sec><sec id=\"sec2dot4-pharmaceuticals-16-00014\"><title>2.4. DEA (<bold>1</bold>) Targets Keap1 and GSK3&#946; to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins</title><p>To verify whether Keap1 or GSK3&#946; are direct targets of DEA (<bold>1</bold>), knockdown assays were performed. <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> are target antioxidant genes of Nrf2 that increase cytoprotection against oxidative stress [<xref rid=\"B48-pharmaceuticals-16-00014\" ref-type=\"bibr\">48</xref>] As expected, Keap1 knockdown in LO2 cells induced a noticeable rise of the expression levels of both Nrf2 and its gene targets, <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> (<xref rid=\"pharmaceuticals-16-00014-f003\" ref-type=\"fig\">Figure 3</xref>A,B). However, treating Keap1 knockdown cells with DEA (<bold>1</bold>) led to no further rises in the <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> level, in contrast to the control cells, where DEA (<bold>1</bold>) treatment produced noticeable increases of <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>. These results suggest that DEA (<bold>1</bold>) activates <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> through targeting Keap1. Similarly, GSK3&#946; knockdown induced Nrf2 and downstream antioxidant protein expression but without influencing Keap1 protein level (<xref rid=\"pharmaceuticals-16-00014-f003\" ref-type=\"fig\">Figure 3</xref>C,D). In addition, DEA (<bold>1</bold>) had a diminished effect on <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> level in GSK3&#946; knockdown cells as compared to the control cells. Finally, we knocked down GSK3&#946; and Keap1 at the same time to verify the dual binding mechanism of DEA (<bold>1</bold>) on GSK3&#946; and Keap1 (<xref rid=\"pharmaceuticals-16-00014-f003\" ref-type=\"fig\">Figure 3</xref>E,F). The results showed that treating the GSK3&#946; and Keap1 knockdown cells with DEA (<bold>1</bold>) did not cause a further significant induction of <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> expression.</p><p>Evidence for the hypothesis that DEA (<bold>1</bold>) acts through targeting both Keap1 and GSK3&#946; comes from considering the Nrf2 level. In either the Keap1 knockdown or GSK3&#946; knockdown cells, DEA (<bold>1</bold>) treatment resulted in slight increases of Nrf2 expression relative to DMSO. However, in the double Keap1 and GSK3&#946; knockdown cells, treatment with DEA (<bold>1</bold>) did not lead to increases of Nrf2 expression. These results suggest that DEA (<bold>1</bold>) treatment mimicked the effects of double knockdown in increasing Nrf2 expression. Therefore, DEA (<bold>1</bold>) exerts its antioxidant effects by targeting both Keap1 and GSK3&#946;.</p><p>Inhibition of the Keap1&#8211;Nrf2 interaction is known to increase Nrf2 accumulation in the nucleus [<xref rid=\"B47-pharmaceuticals-16-00014\" ref-type=\"bibr\">47</xref>]. Meanwhile, GSK3&#946; is also a negative regulator of Nrf2 accumulation by acting independently of Keap1 [<xref rid=\"B49-pharmaceuticals-16-00014\" ref-type=\"bibr\">49</xref>]. Therefore, we evaluated the Nrf2 level in the cytoplasm and nucleus of LO2 cells via WB after the treatment of cells with DEA (<bold>1</bold>) (10 &#181;M) for 8 h. As shown in <xref rid=\"pharmaceuticals-16-00014-f004\" ref-type=\"fig\">Figure 4</xref>A, DEA (<bold>1</bold>) significantly increased Nrf2 translocation into the nucleus in a dose-dependent manner. GSK3&#946; knockdown also phenocopied compound DEA (<bold>1</bold>) treatment at inducing Nrf2 nuclear accumulation without influencing Keap1 level (<xref rid=\"pharmaceuticals-16-00014-f004\" ref-type=\"fig\">Figure 4</xref>B). In summary, DEA (<bold>1</bold>) disrupts the Keap1&#8211;Nrf2 PPI and inhibits GSK3&#946; to increase the translocation of liberated Nrf2 into the nucleus, thereby presumably increasing ARE transcriptional activity.</p></sec><sec id=\"sec2dot5-pharmaceuticals-16-00014\"><title>2.5. DEA (<bold>1</bold>) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells</title><p>Nuclear Nrf2 heterodimerizes with musculoaponeurotic fibrosarcoma (Maf) to recognize ARE and promote the transcription of antioxidant genes, including <italic toggle=\"yes\">Nrf1</italic>, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">NQO1</italic> [<xref rid=\"B14-pharmaceuticals-16-00014\" ref-type=\"bibr\">14</xref>]. Therefore, we evaluated the ability of DEA (<bold>1</bold>) to increase <italic toggle=\"yes\">Nrf1</italic>, <italic toggle=\"yes\">NQO1</italic>, and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> levels in ALD model cells, which were prepared by treating LO2 cells with 0.3% EtOH for 8 h. 0.3% EtOH was reported to induce oxidative stress in living cells [<xref rid=\"B50-pharmaceuticals-16-00014\" ref-type=\"bibr\">50</xref>,<xref rid=\"B51-pharmaceuticals-16-00014\" ref-type=\"bibr\">51</xref>,<xref rid=\"B52-pharmaceuticals-16-00014\" ref-type=\"bibr\">52</xref>]. In addition, a preliminary time-dependent assay was performed, which identified an optimal incubation time length of 8 h (<xref rid=\"app1-pharmaceuticals-16-00014\" ref-type=\"app\">Supplementary Figure S2</xref>). As shown in <xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>A, DEA (<bold>1</bold>) (30 &#956;M) induced <italic toggle=\"yes\">Nrf1</italic> (<xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>B), <italic toggle=\"yes\">NQO1</italic> (<xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>C) and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> (<xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>D) protein levels by 1.2, 3.8 and 1.8-fold, respectively, in ALD model cells, making it more potent than ML334 (30 &#956;M), which increased <italic toggle=\"yes\">Nrf1</italic>, <italic toggle=\"yes\">NQO1</italic>, and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> protein levels by 1.0, 1.1, and 1.5-fold, respectively.</p><p>Nrf2 directly controls the expression of the transcriptional factor <italic toggle=\"yes\">PGC</italic>&#8211;<italic toggle=\"yes\">1&#945;</italic>, a regulator of mitochondrial biogenesis, while it also directly regulates the expression of mitochondrial antioxidant enzymes, such as superoxide dismutase 2 (<italic toggle=\"yes\">SOD2</italic>) [<xref rid=\"B53-pharmaceuticals-16-00014\" ref-type=\"bibr\">53</xref>,<xref rid=\"B54-pharmaceuticals-16-00014\" ref-type=\"bibr\">54</xref>]. Hence, the effect of DEA (<bold>1</bold>) on mRNA levels of <italic toggle=\"yes\">SOD2</italic>, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, <italic toggle=\"yes\">NQO1</italic>, <italic toggle=\"yes\">Nrf1</italic>, and <italic toggle=\"yes\">PGC</italic>&#8211;<italic toggle=\"yes\">1&#945;</italic> in 0.3% EtOH&#8211;treated LO2 cells was also evaluated. 30 &#956;M of DEA (<bold>1</bold>) rescued <italic toggle=\"yes\">SOD2</italic> mRNA level in ALD model cells by 2-fold, compared to 1.5-fold with ML334 (<xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>E). DEA (<bold>1</bold>) (30 &#956;M) also increased <italic toggle=\"yes\">NQO1</italic> (<xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>F) and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> (<xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>G) mRNA levels by 2.0 and 2.5-fold, respectively, making it more potent than ML334 which increased <italic toggle=\"yes\">NQO1</italic> and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> mRNA levels by 1.8 and 1.4-fold, respectively. In addition, the mRNA levels of <italic toggle=\"yes\">Nrf1</italic> and <italic toggle=\"yes\">PGC</italic>&#8211;<italic toggle=\"yes\">1&#945;</italic> were also increased 2-fold by compound DEA (<bold>1</bold>) (30 &#956;M) (<xref rid=\"pharmaceuticals-16-00014-f005\" ref-type=\"fig\">Figure 5</xref>H,I). The results also show a slight increase of <italic toggle=\"yes\">NQO1</italic>, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">PGC</italic>&#8211;<italic toggle=\"yes\">1&#945;</italic> levels by 0.3% EtOH treatment alone, which has been previously observed in other studies [<xref rid=\"B55-pharmaceuticals-16-00014\" ref-type=\"bibr\">55</xref>]. However, the subsequent increases of antioxidant genes were much greater upon further treatment of DEA (<bold>1</bold>). Taken together, these results indicate that DEA (<bold>1</bold>) can activate Nrf2 downstream antioxidant genes in EtOH&#8211;treated human liver cells, a cellular model of ALD.</p></sec><sec id=\"sec2dot6-pharmaceuticals-16-00014\"><title>2.6. DEA (<bold>1</bold>) Repairs the EtOH Induced Mitochondrial Dysfunction and Improves the Antioxidant Activity in LO2 Cells</title><p>Inflammation and oxidative stress increase ROS production, and the activated Nrf2 was able to maintain low intracellular ROS level to protect the cell from oxidant injury [<xref rid=\"B17-pharmaceuticals-16-00014\" ref-type=\"bibr\">17</xref>]. Moreover, growing evidence shows that mitochondrial and oxidant injury induced by ethanol consumption play key roles in alcohol&#8211;induced liver injury [<xref rid=\"B56-pharmaceuticals-16-00014\" ref-type=\"bibr\">56</xref>]. Alcohol&#8211;induced ROS causes mitochondrial membrane depolarization and mitochondrial permeability transition (MPT), leading to hepatic apoptosis and necrosis [<xref rid=\"B54-pharmaceuticals-16-00014\" ref-type=\"bibr\">54</xref>]. Therefore, we detected the effect of DEA (<bold>1</bold>) and ML334 on ROS level in 0.3% EtOH&#8211;treated LO2 cells (<xref rid=\"pharmaceuticals-16-00014-f006\" ref-type=\"fig\">Figure 6</xref>A). The results indicated that both <bold>1</bold> and ML334 could significantly decrease ROS level induced by EtOH. The MPTP assay showed that DEA (<bold>1</bold>) (30 &#956;M) can restore MPTP function (40%) with a higher activity as compared to ML334 (34%) (<xref rid=\"pharmaceuticals-16-00014-f006\" ref-type=\"fig\">Figure 6</xref>B,C). ATP is an indicator of mitochondrial function and ATP level are significantly reduced in ethanol&#8211;fed Nrf2<sup>&#8722;/&#8722;</sup> mice [<xref rid=\"B57-pharmaceuticals-16-00014\" ref-type=\"bibr\">57</xref>]. DEA (<bold>1</bold>) (30 &#956;M) rescued 50% of ATP in LO2 cells after EtOH treatment, making it more potent than ML334, which rescued 20% of the ATP level under the same conditions (<xref rid=\"pharmaceuticals-16-00014-f006\" ref-type=\"fig\">Figure 6</xref>D). Finally, DEA (<bold>1</bold>) (30 &#956;M) induced a higher increase of antioxidant activity (4-fold) compared to ML334 (2-fold) under the same conditions (<xref rid=\"pharmaceuticals-16-00014-f006\" ref-type=\"fig\">Figure 6</xref>E).</p></sec></sec><sec sec-type=\"discussion\" id=\"sec3-pharmaceuticals-16-00014\"><title>3. Discussion</title><p>ALD is a serious chronic liver disease caused by oxidative stress and alcohol metabolism, which leads to massive global deaths [<xref rid=\"B1-pharmaceuticals-16-00014\" ref-type=\"bibr\">1</xref>]. Unfortunately, until now, there are no effective therapeutic drugs approved by the FDA for the treatment of ALD [<xref rid=\"B58-pharmaceuticals-16-00014\" ref-type=\"bibr\">58</xref>]. The pathogenesis of ALD is poorly characterized and research on the mechanism is still not clear. A number of reports have shown that the Keap1&#8211;Nrf2 pathway plays an important role in activating ARE antioxidant signaling [<xref rid=\"B14-pharmaceuticals-16-00014\" ref-type=\"bibr\">14</xref>]. Nrf2 is regulated via Keap1-independent pathways, including mitogen-activated protein kinase&#8211;Erk and PI3K&#8211;Akt pathways [<xref rid=\"B59-pharmaceuticals-16-00014\" ref-type=\"bibr\">59</xref>]. Among these regulatory pathways, GSK3&#946; has emerged as a convergent point [<xref rid=\"B60-pharmaceuticals-16-00014\" ref-type=\"bibr\">60</xref>,<xref rid=\"B61-pharmaceuticals-16-00014\" ref-type=\"bibr\">61</xref>]. Because Nrf2 degradation and nuclear exclusion are regulated by GSK3&#946;, it is critical in switching off the self&#8211;protective antioxidant stress response after injury [<xref rid=\"B59-pharmaceuticals-16-00014\" ref-type=\"bibr\">59</xref>,<xref rid=\"B62-pharmaceuticals-16-00014\" ref-type=\"bibr\">62</xref>]. Hence, targeting Keap1 and GSK3&#946; to activate Nrf2 are potential therapeutic strategies for ALD.</p><p>Natural products provide diverse scaffolds with high bioactivity and low toxicity, and could be developed as promising candidates in drug development [<xref rid=\"B63-pharmaceuticals-16-00014\" ref-type=\"bibr\">63</xref>]. <italic toggle=\"yes\">A. cinnamomea</italic> is a traditional Chinese herb that is a component of various medicines [<xref rid=\"B64-pharmaceuticals-16-00014\" ref-type=\"bibr\">64</xref>]. Triterpenoids are the major constituent in <italic toggle=\"yes\">A. cinnamomea</italic>, and several studies have demonstrated their antioxidative and hepatoprotective effects [<xref rid=\"B65-pharmaceuticals-16-00014\" ref-type=\"bibr\">65</xref>,<xref rid=\"B66-pharmaceuticals-16-00014\" ref-type=\"bibr\">66</xref>]. However, the hepatoprotective mechanism of these compounds has not been extensively explored.</p><p>DEA (<bold>1</bold>) has previously shown promising hepatoprotective activity in some studies [<xref rid=\"B67-pharmaceuticals-16-00014\" ref-type=\"bibr\">67</xref>], including in mouse models [<xref rid=\"B39-pharmaceuticals-16-00014\" ref-type=\"bibr\">39</xref>,<xref rid=\"B67-pharmaceuticals-16-00014\" ref-type=\"bibr\">67</xref>,<xref rid=\"B68-pharmaceuticals-16-00014\" ref-type=\"bibr\">68</xref>,<xref rid=\"B69-pharmaceuticals-16-00014\" ref-type=\"bibr\">69</xref>]. In recent years, many Keap1&#8211;Nrf2 PPI inhibitors and GSK3&#946; inhibitors have been reported for various indications, such as Alzheimer&#8217;s disease, cancers, Parkinson&#8217;s disease, kidney diseases, Acetaminophen (APAP)&#8211;induced liver injury, and ferroptosis [<xref rid=\"B30-pharmaceuticals-16-00014\" ref-type=\"bibr\">30</xref>,<xref rid=\"B70-pharmaceuticals-16-00014\" ref-type=\"bibr\">70</xref>,<xref rid=\"B71-pharmaceuticals-16-00014\" ref-type=\"bibr\">71</xref>,<xref rid=\"B72-pharmaceuticals-16-00014\" ref-type=\"bibr\">72</xref>,<xref rid=\"B73-pharmaceuticals-16-00014\" ref-type=\"bibr\">73</xref>,<xref rid=\"B74-pharmaceuticals-16-00014\" ref-type=\"bibr\">74</xref>]. However, few have been studied in the context of ALD. In our present study, we discovered DEA (<bold>1</bold>) as a potent Keap1&#8211;Nrf2 PPI inhibitor from <italic toggle=\"yes\">A. cinnamomea</italic> extracts. We explored the detailed mechanism of hepatoprotective action of DEA (<bold>1</bold>) in ALD, which acted via targeting both the Keap1&#8211;Nrf2 PPI and GSK3&#946; (<xref rid=\"pharmaceuticals-16-00014-f007\" ref-type=\"fig\">Figure 7</xref>). To our best knowledge, the dual inhibition mechanism of DEA (<bold>1</bold>) in ALD has not been reported before.</p><p>In in vitro assays, DEA (<bold>1</bold>) inhibited the Keap1&#8211;Nrf2 PPI (EC<sub>50</sub> = 14.1 &#181;M) as well as GSK3&#946; kinase activity (EC<sub>50</sub> = 8.0 &#177; 0.7 &#181;M) with greater potency than the known Keap1&#8211;Nrf2 inhibitor, ML334. Cellular experiments confirmed the ability of DEA (<bold>1</bold>) to engage Keap1 and GSK3&#946; in cellulo to activate Nrf2 and stimulate the expression of ARE&#8211;controlled genes. The specificity of DEA (<bold>1</bold>) was further confirmed using Keap1 and GSK3&#946; knockdown experiments. DEA (<bold>1</bold>) treatment phenocopied dual knockdown of Keap1 and GSK3&#946;, suggesting that DEA (<bold>1</bold>) acted simultaneously through both pathways to exert its biological effects. Moreover, DEA (<bold>1</bold>) increased Nrf2 nuclear accumulation, mimicking GSK3&#946; knockdown, without influencing Keap1 level. Finally, we showed that DEA (<bold>1</bold>) could restore MPTP function and reduce ROS level induced by EtOH in ALD model cells. DEA (<bold>1</bold>) displayed low cytotoxicity in both normal and cancerous liver cells, indicating its potential safety for treating ALD in vivo. In summary, DEA (<bold>1</bold>) is a promising scaffold for the further development of dual inhibitors of the Keap1&#8211;Nrf2 PPI and GSK3&#946; as therapeutic agents against ALD. The structural modification of compound DEA (<bold>1</bold>) is in progress and animal studies will be performed in due course.</p></sec><sec id=\"sec4-pharmaceuticals-16-00014\"><title>4. Materials and Methods</title><sec id=\"sec4dot1-pharmaceuticals-16-00014\"><title>4.1. Cell Lines and Culture</title><p>The human normal liver cell line (LO2), human embryonic kidney 293 T cell line, and HepG2 cell line were purchased from American Type Culture Collection (Manassas, VA, USA). Cells were cultured in Dulbecco&#8217;s modified Eagle&#8217;s medium (DMEM), with 10% FBS, 100 units/mL penicillin, 100 &#956;g/mL streptomycin. Fetal bovine serum (FBS), DMEM, penicillin, and streptomycin were purchased from Gibco BRL (Gaithersburg, MD, USA).</p></sec><sec id=\"sec4dot2-pharmaceuticals-16-00014\"><title>4.2. Chemical</title><p>Compounds <bold>1</bold>&#8211;<bold>13</bold> (purity &gt; 98% by HPLC analysis) were isolated from <italic toggle=\"yes\">A. cinnamomea</italic> in a previous study [<xref rid=\"B75-pharmaceuticals-16-00014\" ref-type=\"bibr\">75</xref>]. New <sup>1</sup>H and <sup>13</sup>C nuclear magnetic resonance (NMR) were recorded for DEA (<bold>1</bold>) to verify its identity for this study (<xref rid=\"app1-pharmaceuticals-16-00014\" ref-type=\"app\">Supplementary Figure S1</xref>). Spectral data were consistent with our previous report [<xref rid=\"B75-pharmaceuticals-16-00014\" ref-type=\"bibr\">75</xref>].</p></sec><sec id=\"sec4dot3-pharmaceuticals-16-00014\"><title>4.3. Fluorescence Polarization Assay</title><p>The ability of compounds <bold>1</bold>&#8211;<bold>13</bold> and ML334 to inhibit Keap1&#8211;Nrf2 peptide binding was evaluated using a FP assay according to the manufacturer&#8217;s instruction (BPS Bioscience, San Diego, CA, USA). ML334 was a non&#8211;covalent small molecule inhibitor of the Keap1&#8211;Nrf2 and purchased from MedChemExpress (Princeton, NJ, USA). The tested compounds were dissolved in the provided buffer at the indicated concentrations. Then, the required volumes of Nrf2 peptide, Bovine Serum Albumin (BSA), Keap1 protein, and assay buffer were added to each well, followed by incubation for 30 min and the fluorescent polarization value was recorded.</p></sec><sec id=\"sec4dot4-pharmaceuticals-16-00014\"><title>4.4. Western Blot (WB) and Co&#8211;Immunoprecipitation (Co&#8211;IP)</title><p>The protocol was reported in a previous study [<xref rid=\"B76-pharmaceuticals-16-00014\" ref-type=\"bibr\">76</xref>]. In brief, LO2 cells were seeded at a density of 2 &#215; 10<sup>6</sup> cells in a six&#8211;well plate. LO2 lysate is harvested after incubation with 10 &#956;M of compound DEA (<bold>1</bold>) or ML334 for 8 h. A 10% SDS&#8211;PAGE gel was used to separate the LO2 cell lysates, which were transferred to a PVDF membrane (Bio&#8211;Rad, Hercules, CA, USA). The membranes were incubated with antibodies anti&#8211;Keap1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 8047S), Nrf2 (1:1000, Proteintech, Wuhan, China; Cat# 16396-1-AP), GSK3&#946; (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 9832S), <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 5853S), <italic toggle=\"yes\">NQO1</italic> (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 3187S), &#946;&#8211;actin (1:1000, Absin, Shanghai, China; Cat#: abs137975), or anti-lamin B (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat # 12586S) overnight. The membranes were washed with wash buffer three times. A secondary antibody (1:1000) was added to the membranes. After incubated for 2 h, the proteins bands were detected using enhanced chemiluminescent Plus reagents (GE Healthcare, Boston, MA, USA) and analyzed by Image Lab. The antibodies were purchased from Abcam (Waltham, MA, USA).</p><p>Co&#8211;IP between Keap1 and Nrf2 was performed following the protocol from Life Technologies. Briefly, lysates from 2 &#215; 10<sup>6</sup> LO2 cells were incubated with anti&#8211;Nrf2 or Rabbit mAb IgG (Abcam, Waltham, MA, USA; Cat#: ab205718) overnight. Then, samples were incubated with 10 &#956;L Dynabeads (Life technologies, Foster, CA, USA) for 6 h. The Dynabeads were washed and analyzed by WB with the Keap1 antibody.</p></sec><sec id=\"sec4dot5-pharmaceuticals-16-00014\"><title>4.5. Real-Time Quantitative Polymerase Chain Reaction (RT&#8211;qPCR)</title><p>LO2 cells were incubated with 10 &#181;M or 30 &#181;M compound DEA (<bold>1</bold>) for 8 h and the total RNA was extracted using the NucleoSpin<sup>&#174;</sup> RNA Plus kit (Takara, Tokyo, Japan). We checked the RNA quality during the RT&#8211;qPCR operation and ensured that all the A260/280 value of samples were around 2, which is consistent with the literature [<xref rid=\"B77-pharmaceuticals-16-00014\" ref-type=\"bibr\">77</xref>]. We also performed an RT negative control and found no PCR amplification. Then, cDNA was synthesized using the PrimeScript&#8482; RT Reagent Kit (Takara, Tokyo, Japan). The RT&#8211;qPCR conditions were as previously reported, with minor modifications [<xref rid=\"B77-pharmaceuticals-16-00014\" ref-type=\"bibr\">77</xref>]. DNA denaturation temperature: 95 &#176;C (30 s); annealing temperature: 58 &#176;C (30 s); extension temperature: 72 &#176;C (45 s), number of cycles: 40. RT&#8211;qPCR was performed and analyzed as described previously [<xref rid=\"B78-pharmaceuticals-16-00014\" ref-type=\"bibr\">78</xref>]. The related primers are listed in <xref rid=\"app1-pharmaceuticals-16-00014\" ref-type=\"app\">Supplementary Table S1</xref>.</p></sec><sec id=\"sec4dot6-pharmaceuticals-16-00014\"><title>4.6. MTT Assay</title><p>The LO2, HUVEC and HEK 293 T cell lines were seeded at 1 &#215; 10<sup>4</sup> cells per well in 96&#8211;well plates and incubated with DMEM overnight at 37 &#176;C in a humidified CO<sub>2</sub> incubator to ensure attachment. The medium was then replaced with a medium that contained the different concentrations of compound for 48 h at 37 &#176;C. The final concentration of DMSO was 0.1% or lower in all cases (including controls). The medium with 100 &#956;L of MTT (3-(4,5-dimethylthiazol-2-yl)-2,5-tetrazolium bromide) reagent (1 mg/mL) was then replaced. After 4 h incubation, it was replaced with 100 &#956;L of DMSO and measured using a microplate reader at 570 nm.</p></sec><sec id=\"sec4dot7-pharmaceuticals-16-00014\"><title>4.7. Detection of ROS and Antioxidant Activity</title><p>LO2 cells at a density of 2 &#215; 10<sup>6</sup> cells/well in a six-well plate were treated with different concentrations of DEA (<bold>1</bold>) or 10 &#956;M ML334 for 8 h. ROS level and antioxidant activity were measured using the Reactive Oxygen Species Assay kit and the Total Antioxidant Capacity Assay Kit with FRAP, respectively, according to the manufacturer&#8217;s instructions (Beyotime, Shanghai, China).</p></sec><sec id=\"sec4dot8-pharmaceuticals-16-00014\"><title>4.8. Nuclear and Cytoplasmic Extraction</title><p>LO2 cells were incubated with 10 &#181;M or 30 &#181;M of DEA (<bold>1</bold>) for 8 h. The nuclear and cytoplasmic extraction followed the protocol from the manufacturer (Epigentek, New York, NY, USA, EpiQuik&#8482; Nuclear Extraction Kit I, Catalog # OP-0002), Then, protein expression was analyzed by WB with the indicated antibodies.</p></sec><sec id=\"sec4dot9-pharmaceuticals-16-00014\"><title>4.9. GSK3&#946; Kinase Assay</title><p>The ability of DEA (<bold>1</bold>) and ML334 to inhibit GSK3&#946; activity was evaluated using luminescent kinase assay according to the manufacturer&#8217;s instruction (Promega Corporation, Madison, WI, USA). Briefly, the 384 well white plate was mixed with 1 &#956;L of inhibitor, 2 &#956;L of enzyme, and 2 &#956;L of substrate/ATP mix and incubated at room temperature for 60 min. After 60 min, 5 &#956;L of ADP&#8211;Glo&#8482; reagent was added and incubated at room temperature for 40 min. Then 10 &#956;L of kinase detection reagent was added, incubated at room temperature for 30 min and the luminescence value was recorded (integration time 0.5&#8211;1 s).</p></sec><sec id=\"sec4dot10-pharmaceuticals-16-00014\"><title>4.10. Dual Luciferase Reporter Gene Assay</title><p>LO2 cells were seeded in a six&#8211;well plate with 80% confluence in a DMEM medium for 24 h. The ARE&#8211;luciferase vector or a control plasmid with TurboFect transfection reagent (Thermo Fisher Scientific, Waltham, MA, USA) were transfected into LO2 cells according to the manufacturer&#8217;s instruction, and cells were incubated for 48 h. Then, LO2 cells were incubated with 10 &#181;M compound DEA (<bold>1</bold>), 10 &#181;M ML334 or DMSO for 4 h. The ARE activity was detected through the manufacturer&#8217;s instruction (Beyotime Dual Luciferase Reporter Gene Assay Kit, Shanghai, China).</p></sec><sec id=\"sec4dot11-pharmaceuticals-16-00014\"><title>4.11. Cellular Thermal Shift Assay (CETSA)</title><p>LO2 cell lysates were incubated with 10 &#956;M compound DEA (<bold>1</bold>) and DMSO at room temperature for 30 min. The lysates are divided and heated individually at the indicated temperature. The supernatants of the heated lysates were collected after centrifugation, and then analyzed by WB with the Keap1, Nrf2, and &#946;&#8211;actin antibodies.</p></sec><sec id=\"sec4dot12-pharmaceuticals-16-00014\"><title>4.12. Mitochondrial Permeability Transition Pore (MPTP) Assay</title><p>MPTP kit was used for evaluation of mitochondrial permeability transition according to the manufacturer&#8217;s instructions (Beyotime, Shanghai, China). A density of 1 &#215; 10<sup>4</sup> cells in a six&#8211;well plate, EtOH&#8211;induced LO2 cells were treated with 10 &#181;M of ML334, 10 or 30 &#181;M compound DEA (<bold>1</bold>), DMSO for 8 h, and then pretreated with calcein or calcein and CoCl<sub>2</sub>. Cells were stained by calcein acetoxymethyl ester (Calcein AM) and MPT function was detected by fluorescence spectroscopy.</p></sec><sec id=\"sec4dot13-pharmaceuticals-16-00014\"><title>4.13. SiRNA Gene Knockdown</title><p>The sequences of Keap1 siRNA are 5&#8242;-GGCCUUUGGCAUCAUGAACTT-3&#8242; (sense) and 5&#8242;-GUUCAUGAUGCCAAAGGCCTG-3&#8242; (antisense) [<xref rid=\"B79-pharmaceuticals-16-00014\" ref-type=\"bibr\">79</xref>]. GSK3&#946; siRNA is 5&#8242;-GCAUUUAUCGUUAACCUAA-3&#8242; (sense) and 5&#8242;-UUAGGUUAACGA UAAAUGC-3&#8242; (antisense) [<xref rid=\"B80-pharmaceuticals-16-00014\" ref-type=\"bibr\">80</xref>]. Keap1 and GSK3&#946; siRNA was transfected with Lipo3000 reagent (Thermo Fisher Scientific, Waltham, MA, USA) into 1 &#215; 10<sup>6</sup> LO2 cells following the manufacturer&#8217;s instruction. After incubation for 48 h, Keap1 and GSK3&#946; protein expression and functional experiments were performed.</p></sec><sec id=\"sec4dot14-pharmaceuticals-16-00014\"><title>4.14. Statistical Analysis</title><p>Statistical significance was determined using the Student&#8217;s t-test for experiments comparing two groups. The data is homogeneous and follows a normal distribution before proceeding to the statistical analysis. Comparisons among groups were analysed using analysis of variance (ANOVA). All statistical tests were done using the GraphPad Prism version 8.0 software (GraphPad Software Inc. San Diego, CA, USA). All values were expressed as the mean &#177; standard deviation (SD). <italic toggle=\"yes\">p</italic> &lt; 0.05 means statistical significance.</p></sec></sec><sec sec-type=\"conclusions\" id=\"sec5-pharmaceuticals-16-00014\"><title>5. Conclusions</title><p>Recent studies on ALD are scarce compared to research on other liver diseases, and there are still no highly satisfactory therapeutic options for ALD. To our knowledge, no dual inhibitor of the Keap1&#8211;Nrf2 PPI and GSK3&#946; to treat alcohol-induced liver injury has been reported. This paper has revealed that DEA (<bold>1</bold>), one of the active triterpenes isolated from <italic toggle=\"yes\">A. cinnamomea</italic>, could be a potential lead compound for treating ALD. Compound DEA (<bold>1</bold>) inhibited the Keap1&#8211;Nrf2 interaction and GSK3&#946; activity both in vitro and in cellulo, and displayed low cytotoxicity at the concentrations required for inducing antioxidant activity. Moreover, DEA (<bold>1</bold>) could engage Keap1 and GSK3&#946; in liver cells and increase Nrf2 nucleus translocation, thereby activating ARE transcriptional activity and upregulating antioxidant genes and mitochondrial biogenesis regulators. Significantly, compound DEA (<bold>1</bold>) displayed more potent hepatoprotective activity when compared with the classic Keap1&#8211;Nrf2 PPI inhibitor ML334. Therefore, the structure of DEA (<bold>1</bold>) may be optimized to expand the diversity, improve the activity, and be used in animal research. To our knowledge, this paper firstly reports DEA (<bold>1</bold>) acts as the dual inhibitor of Keap1&#8211;Nrf2 PPI and GSK3&#946; with the potential for protecting against ALD. We anticipate that natural product DEA (<bold>1</bold>) can be considered as a potential scaffold for the development of clinical agents for treating ALD.</p></sec></body><back><fn-group><fn><p><bold>Disclaimer/Publisher&#8217;s Note:</bold> The statements, opinions and data contained in all publications are solely those of the individual author(s) and contributor(s) and not of MDPI and/or the editor(s). MDPI and/or the editor(s) disclaim responsibility for any injury to people or property resulting from any ideas, methods, instructions or products referred to in the content.</p></fn></fn-group><app-group><app id=\"app1-pharmaceuticals-16-00014\"><title>Supplementary Materials</title><p>The following supporting information can be downloaded at: <uri xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"https://www.mdpi.com/article/10.3390/ph16010014/s1\">https://www.mdpi.com/article/10.3390/ph16010014/s1</uri>, Figure S1: NMR data of dehydroeburicoic acid; Figure S2: The time dependent of ML334, compound 2, compound DEA (1) and induces antioxidant factor expression in the ALD cell model; Table S1: Primer sequences used for PCR analysis.</p><supplementary-material id=\"pharmaceuticals-16-00014-s001\" position=\"float\" content-type=\"local-data\" orientation=\"portrait\"><media xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"pharmaceuticals-16-00014-s001.zip\" position=\"float\" orientation=\"portrait\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material></app></app-group><notes><title>Author Contributions</title><p>Conceptualization, M.Y. and C.-H.L.; Methodology, S.C., Y.K., G.L., J.W., C.-N.K. and W.W.; Software, Y.K., C.-N.K., W.W., D.-L.M. and M.Y.; Validation, G.L.; Formal analysis, S.C., Y.K., G.L. and J.W.; Investigation, S.C., Y.K. and J.W.; Resources, Y.K. and M.Y.; Data curation, S.C., G.L., J.W. and C.-N.K.; Writing&#8212;original draft, S.C.; Writing&#8212;review &amp; editing, Y.K.; Supervision, D.-L.M., M.Y. and C.-H.L.; Project administration, D.-L.M., M.Y. and C.-H.L.; Funding acquisition, D.-L.M., M.Y. and C.-H.L. All authors have read and agreed to the published version of the manuscript.</p></notes><notes><title>Institutional Review Board Statement</title><p>Not applicable.</p></notes><notes><title>Informed Consent Statement</title><p>Not applicable.</p></notes><notes notes-type=\"data-availability\"><title>Data Availability Statement</title><p>Data is contained within the article and <xref rid=\"app1-pharmaceuticals-16-00014\" ref-type=\"app\">Supplementary Material</xref>.</p></notes><notes notes-type=\"COI-statement\"><title>Conflicts of Interest</title><p>The authors declare no conflict of interest.</p></notes><ref-list><title>References</title><ref id=\"B1-pharmaceuticals-16-00014\"><label>1.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Avila</surname><given-names>M.A.</given-names></name><name name-style=\"western\"><surname>Dufour</surname><given-names>J.F.</given-names></name><name name-style=\"western\"><surname>Gerbes</surname><given-names>A.L.</given-names></name><name name-style=\"western\"><surname>Zoulim</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Bataller</surname><given-names>R.</given-names></name><name name-style=\"western\"><surname>Burra</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Cortez-Pinto</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Gao</surname><given-names>B.</given-names></name><name name-style=\"western\"><surname>Gilmore</surname><given-names>I.</given-names></name><name name-style=\"western\"><surname>Mathurin</surname><given-names>P.</given-names></name><etal/></person-group><article-title>Recent advances in alcohol-related liver disease (ALD): Summary of a Gut round table meeting</article-title><source>Gut</source><year>2019</year><volume>69</volume><fpage>764</fpage><lpage>780</lpage><pub-id pub-id-type=\"doi\">10.1136/gutjnl-2019-319720</pub-id><pub-id pub-id-type=\"pmid\">31879281</pub-id><pub-id pub-id-type=\"pmcid\">PMC7236084</pub-id></element-citation></ref><ref id=\"B2-pharmaceuticals-16-00014\"><label>2.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Bakhautdin</surname><given-names>B.</given-names></name><name name-style=\"western\"><surname>Das</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Mandal</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Roychowdhury</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Danner</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Bush</surname><given-names>K.</given-names></name><name name-style=\"western\"><surname>Pollard</surname><given-names>K.</given-names></name><name name-style=\"western\"><surname>Kaspar</surname><given-names>J.W.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Salomon</surname><given-names>R.G.</given-names></name><etal/></person-group><article-title>Protective role of HO-1 and carbon monoxide in ethanol-induced hepatocyte cell death and liver injury in mice</article-title><source>J. Hepatol.</source><year>2014</year><volume>61</volume><fpage>1029</fpage><lpage>1037</lpage><pub-id pub-id-type=\"doi\">10.1016/j.jhep.2014.06.007</pub-id><pub-id pub-id-type=\"pmid\">24946281</pub-id><pub-id pub-id-type=\"pmcid\">PMC4203703</pub-id></element-citation></ref><ref id=\"B3-pharmaceuticals-16-00014\"><label>3.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Higuera-de Tijera</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Serv&#237;n-Caama&#241;o</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Serralde-Z&#250;&#241;iga</surname><given-names>A.E.</given-names></name><name name-style=\"western\"><surname>Cruz-Herrera</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>P&#233;rez-Torres</surname><given-names>E.</given-names></name><name name-style=\"western\"><surname>Abdo-Francis</surname><given-names>J.M.</given-names></name><name name-style=\"western\"><surname>Salas-Gordillo</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>P&#233;rez-Hern&#225;ndez</surname><given-names>J.L.</given-names></name></person-group><article-title>Metadoxine improves the three-and six-month survival rates in patients with severe alcoholic hepatitis</article-title><source>World J. Gastroenterol.</source><year>2015</year><volume>21</volume><fpage>4975</fpage><lpage>4985</lpage><pub-id pub-id-type=\"doi\">10.3748/wjg.v21.i16.4975</pub-id><pub-id pub-id-type=\"pmid\">25945012</pub-id><pub-id pub-id-type=\"pmcid\">PMC4408471</pub-id></element-citation></ref><ref id=\"B4-pharmaceuticals-16-00014\"><label>4.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Triantafyllou</surname><given-names>K.</given-names></name><name name-style=\"western\"><surname>Vlachogiannakos</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Ladas</surname><given-names>S.D.</given-names></name></person-group><article-title>Gastrointestinal and liver side effects of drugs in elderly patients</article-title><source>Best Pract. Res. Clin. Gastroenterol.</source><year>2010</year><volume>24</volume><fpage>203</fpage><lpage>215</lpage><pub-id pub-id-type=\"doi\">10.1016/j.bpg.2010.02.004</pub-id><pub-id pub-id-type=\"pmid\">20227033</pub-id></element-citation></ref><ref id=\"B5-pharmaceuticals-16-00014\"><label>5.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Li</surname><given-names>X.X.</given-names></name><name name-style=\"western\"><surname>Jiang</surname><given-names>Z.H.</given-names></name><name name-style=\"western\"><surname>Zhou</surname><given-names>B.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>C.</given-names></name></person-group><article-title>Hepatoprotective effect of gastrodin against alcohol-induced liver injury in mice</article-title><source>J. Physiol. Biochem.</source><year>2019</year><volume>75</volume><fpage>29</fpage><lpage>37</lpage><pub-id pub-id-type=\"doi\">10.1007/s13105-018-0647-8</pub-id><pub-id pub-id-type=\"pmid\">30242628</pub-id></element-citation></ref><ref id=\"B6-pharmaceuticals-16-00014\"><label>6.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Nassir</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Ibdah</surname><given-names>J.A.</given-names></name></person-group><article-title>Role of mitochondria in alcoholic liver disease</article-title><source>World J. Gastroenterol.</source><year>2014</year><volume>20</volume><fpage>2136</fpage><lpage>2342</lpage><pub-id pub-id-type=\"doi\">10.3748/wjg.v20.i9.2136</pub-id><pub-id pub-id-type=\"pmid\">24605012</pub-id><pub-id pub-id-type=\"pmcid\">PMC3942818</pub-id></element-citation></ref><ref id=\"B7-pharmaceuticals-16-00014\"><label>7.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Abdallah</surname><given-names>M.A.</given-names></name><name name-style=\"western\"><surname>Singal</surname><given-names>A.K.</given-names></name></person-group><article-title>Mitochondrial dysfunction and alcohol-associated liver disease: A novel pathway and therapeutic target</article-title><source>Signal Transduct. Target. Ther.</source><year>2020</year><volume>5</volume><fpage>1</fpage><lpage>2</lpage><pub-id pub-id-type=\"doi\">10.1038/s41392-020-0128-8</pub-id><pub-id pub-id-type=\"pmid\">32296016</pub-id><pub-id pub-id-type=\"pmcid\">PMC7054548</pub-id></element-citation></ref><ref id=\"B8-pharmaceuticals-16-00014\"><label>8.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Fang</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Zhao</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>He</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Guo</surname><given-names>T.S.</given-names></name><name name-style=\"western\"><surname>Song</surname><given-names>Q.</given-names></name><name name-style=\"western\"><surname>Guo</surname><given-names>N.</given-names></name><name name-style=\"western\"><surname>Yuan</surname><given-names>Z.Y.</given-names></name></person-group><article-title>Overexpression of FGF19 alleviates hypoxia/reoxygenation-induced injury of cardiomyocytes by regulating GSK-3&#946;/Nrf2/ARE signaling</article-title><source>Biochem. Biophys. Res. Commun.</source><year>2018</year><volume>503</volume><fpage>2355</fpage><lpage>2362</lpage><pub-id pub-id-type=\"doi\">10.1016/j.bbrc.2018.06.161</pub-id><pub-id pub-id-type=\"pmid\">29964017</pub-id></element-citation></ref><ref id=\"B9-pharmaceuticals-16-00014\"><label>9.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Han</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Ybanez</surname><given-names>M.D.</given-names></name><name name-style=\"western\"><surname>Johnson</surname><given-names>H.S.</given-names></name><name name-style=\"western\"><surname>McDonald</surname><given-names>J.N.</given-names></name><name name-style=\"western\"><surname>Mesropyan</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Sancheti</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Martin</surname><given-names>G.</given-names></name><name name-style=\"western\"><surname>Martin</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Lim</surname><given-names>A.M.</given-names></name><name name-style=\"western\"><surname>Dara</surname><given-names>L.</given-names></name><etal/></person-group><article-title>Dynamic adaptation of liver mitochondria to chronic alcohol feeding in mice biogenesis, remodeling, and functional alterations</article-title><source>J. Biol. Chem.</source><year>2021</year><volume>287</volume><fpage>42165</fpage><lpage>42179</lpage><pub-id pub-id-type=\"doi\">10.1074/jbc.M112.377374</pub-id><pub-id pub-id-type=\"pmcid\">PMC3516762</pub-id><pub-id pub-id-type=\"pmid\">23086958</pub-id></element-citation></ref><ref id=\"B10-pharmaceuticals-16-00014\"><label>10.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Song</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Cui</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Meng</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Xia</surname><given-names>Q.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Hou</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Jia</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>J.</given-names></name></person-group><article-title>Glucopyranose from Pleurotus geesteranus prevent alcoholic liver diseases by regulating Nrf2/HO-1-TLR4/NF-&#954;B signalling pathways and gut microbiota</article-title><source>Food Funct.</source><year>2022</year><volume>13</volume><fpage>2441</fpage><lpage>2455</lpage><pub-id pub-id-type=\"doi\">10.1039/D1FO03486C</pub-id><pub-id pub-id-type=\"pmid\">35048917</pub-id></element-citation></ref><ref id=\"B11-pharmaceuticals-16-00014\"><label>11.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Jiang</surname><given-names>Z.Y.</given-names></name><name name-style=\"western\"><surname>Lu</surname><given-names>M.C.</given-names></name><name name-style=\"western\"><surname>You</surname><given-names>Q.D.</given-names></name></person-group><article-title>Discovery and development of Kelch-like ECH-associated protein 1. nuclear factor erythroid 2-related factor 2 (KEAP1: NRF2) protein&#8211;protein interaction inhibitors: Achievements, challenges, and future directions</article-title><source>J. Med. Chem.</source><year>2016</year><volume>59</volume><fpage>10837</fpage><lpage>10858</lpage><pub-id pub-id-type=\"doi\">10.1021/acs.jmedchem.6b00586</pub-id><pub-id pub-id-type=\"pmid\">27690435</pub-id></element-citation></ref><ref id=\"B12-pharmaceuticals-16-00014\"><label>12.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Zhou</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Zheng</surname><given-names>Q.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Z.</given-names></name></person-group><article-title>The Nrf2 Pathway in Liver Diseases</article-title><source>Front. Cell Dev. Biol.</source><year>2022</year><volume>10</volume><fpage>826204</fpage><pub-id pub-id-type=\"doi\">10.3389/fcell.2022.826204</pub-id><pub-id pub-id-type=\"pmid\">35223849</pub-id><pub-id pub-id-type=\"pmcid\">PMC8866876</pub-id></element-citation></ref><ref id=\"B13-pharmaceuticals-16-00014\"><label>13.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Itoh</surname><given-names>K.</given-names></name><name name-style=\"western\"><surname>Wakabayashi</surname><given-names>N.</given-names></name><name name-style=\"western\"><surname>Katoh</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Ishii</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Igarashi</surname><given-names>K.</given-names></name><name name-style=\"western\"><surname>Engel</surname><given-names>J.D.</given-names></name><name name-style=\"western\"><surname>Yamamoto</surname><given-names>M.</given-names></name></person-group><article-title>Keap1 represses nuclear activation of antioxidant responsive elements by Nrf2 through binding to the amino-terminal Neh2 domain</article-title><source>Genes Dev.</source><year>1999</year><volume>13</volume><fpage>76</fpage><lpage>86</lpage><pub-id pub-id-type=\"doi\">10.1101/gad.13.1.76</pub-id><pub-id pub-id-type=\"pmid\">9887101</pub-id><pub-id pub-id-type=\"pmcid\">PMC316370</pub-id></element-citation></ref><ref id=\"B14-pharmaceuticals-16-00014\"><label>14.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Michali&#269;kov&#225;</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Hrn&#269;&#237;&#345;</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Canov&#225;</surname><given-names>N.K.</given-names></name><name name-style=\"western\"><surname>Slana&#345;</surname><given-names>O.</given-names></name></person-group><article-title>Targeting Keap1/Nrf2/ARE signaling pathway in multiple sclerosis</article-title><source>Eur. J. Pharmacol.</source><year>2020</year><volume>873</volume><fpage>172973</fpage><lpage>172988</lpage><pub-id pub-id-type=\"doi\">10.1016/j.ejphar.2020.172973</pub-id><pub-id pub-id-type=\"pmid\">32017935</pub-id></element-citation></ref><ref id=\"B15-pharmaceuticals-16-00014\"><label>15.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Mansouri</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Reiner</surname><given-names>&#381;.</given-names></name><name name-style=\"western\"><surname>Ruscica</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Tedeschi-Reiner</surname><given-names>E.</given-names></name><name name-style=\"western\"><surname>Radbakhsh</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Bagheri Ekta</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Sahebkar</surname><given-names>A.</given-names></name></person-group><article-title>Antioxidant Effects of Statins by Modulating Nrf2 and Nrf2/HO-1 Signaling in Different Diseases</article-title><source>J. Clin. Med.</source><year>2022</year><volume>11</volume><elocation-id>1313</elocation-id><pub-id pub-id-type=\"doi\">10.3390/jcm11051313</pub-id><pub-id pub-id-type=\"pmid\">35268403</pub-id><pub-id pub-id-type=\"pmcid\">PMC8911353</pub-id></element-citation></ref><ref id=\"B16-pharmaceuticals-16-00014\"><label>16.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Mitsuishi</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Taguchi</surname><given-names>K.</given-names></name><name name-style=\"western\"><surname>Kawatani</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Shibata</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Nukiwa</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Aburatani</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Yamamoto</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Motohashi</surname><given-names>H.</given-names></name></person-group><article-title>Nrf2 redirects glucose and glutamine into anabolic pathways in metabolic reprogramming</article-title><source>Cancer Cell</source><year>2012</year><volume>22</volume><fpage>66</fpage><lpage>79</lpage><pub-id pub-id-type=\"doi\">10.1016/j.ccr.2012.05.016</pub-id><pub-id pub-id-type=\"pmid\">22789539</pub-id></element-citation></ref><ref id=\"B17-pharmaceuticals-16-00014\"><label>17.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Zhang</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Xiong</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Pang</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Su</surname><given-names>Z.W.</given-names></name><name name-style=\"western\"><surname>Lai</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Lin</surname><given-names>H.Q.</given-names></name><name name-style=\"western\"><surname>Jian</surname><given-names>B.Q.</given-names></name><name name-style=\"western\"><surname>He</surname><given-names>W.H.</given-names></name><name name-style=\"western\"><surname>Yang</surname><given-names>H.D.</given-names></name><name name-style=\"western\"><surname>Zheng</surname><given-names>Y.Q.</given-names></name></person-group><article-title>Nrf2 activation protects auditory hair cells from cisplatin-induced ototoxicity independent on mitochondrial ROS production</article-title><source>Toxicol. Lett.</source><year>2020</year><volume>331</volume><fpage>1</fpage><lpage>10</lpage><pub-id pub-id-type=\"doi\">10.1016/j.toxlet.2020.04.005</pub-id><pub-id pub-id-type=\"pmid\">32428544</pub-id></element-citation></ref><ref id=\"B18-pharmaceuticals-16-00014\"><label>18.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Xu</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Jeong</surname><given-names>S.S.</given-names></name><name name-style=\"western\"><surname>Qian</surname><given-names>Y.H.</given-names></name><name name-style=\"western\"><surname>Wu</surname><given-names>H.L.</given-names></name><name name-style=\"western\"><surname>Xia</surname><given-names>Q.</given-names></name><name name-style=\"western\"><surname>Kong</surname><given-names>X.N.</given-names></name></person-group><article-title>The role of Nrf2 in liver disease: Novel molecular mechanisms and therapeutic approaches</article-title><source>Front. Pharmacol.</source><year>2019</year><volume>8</volume><fpage>1428</fpage><lpage>1435</lpage><pub-id pub-id-type=\"doi\">10.3389/fphar.2018.01428</pub-id><pub-id pub-id-type=\"pmcid\">PMC6331455</pub-id><pub-id pub-id-type=\"pmid\">30670963</pub-id></element-citation></ref><ref id=\"B19-pharmaceuticals-16-00014\"><label>19.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Huang</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Xie</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Lai</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Mo</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Zeng</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Kuang</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Deng</surname><given-names>G.</given-names></name><name name-style=\"western\"><surname>Zhou</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Y.</given-names></name><etal/></person-group><article-title>Hepatic TGF&#946;r1 Deficiency Attenuates Lipopolysaccharide/D-Galactosamine-Induced Acute Liver Failure Through Inhibiting GSK3&#946;-Nrf2-Mediated Hepatocyte Apoptosis and Ferroptosis</article-title><source>Cell Mol. Gastroenterol. Hepatol.</source><year>2022</year><volume>13</volume><fpage>1649</fpage><lpage>1672</lpage><pub-id pub-id-type=\"doi\">10.1016/j.jcmgh.2022.02.009</pub-id><pub-id pub-id-type=\"pmid\">35202887</pub-id><pub-id pub-id-type=\"pmcid\">PMC9046809</pub-id></element-citation></ref><ref id=\"B20-pharmaceuticals-16-00014\"><label>20.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Shu</surname><given-names>G.</given-names></name><name name-style=\"western\"><surname>Qiu</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Hao</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Fu</surname><given-names>Q.</given-names></name><name name-style=\"western\"><surname>Deng</surname><given-names>X.</given-names></name></person-group><article-title>&#947;-Oryzanol alleviates acetaminophen-induced liver injury: Roles of modulating AMPK/GSK3&#946;/Nrf2 and NF-&#954;B signaling pathways</article-title><source>Food Funct.</source><year>2019</year><volume>10</volume><fpage>6858</fpage><lpage>6872</lpage><pub-id pub-id-type=\"doi\">10.1039/C9FO01808E</pub-id><pub-id pub-id-type=\"pmid\">31584590</pub-id></element-citation></ref><ref id=\"B21-pharmaceuticals-16-00014\"><label>21.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Rojo</surname><given-names>A.I.</given-names></name><name name-style=\"western\"><surname>Medina-Campos</surname><given-names>O.N.</given-names></name><name name-style=\"western\"><surname>RadaP</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Z&#250;&#241;iga-Toal&#225;c</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>L&#243;pez-Gazc&#243;n</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Espada</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Pedraza-Chaverri</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Cuadradoa</surname><given-names>A.</given-names></name></person-group><article-title>Signaling pathways activated by the phytochemical nordihydroguaiaretic acid contribute to a Keap1-independent regulation of Nrf2 stability: Role of glycogen synthase kinase-3</article-title><source>Free Radic. Biol. Med.</source><year>2012</year><volume>52</volume><fpage>473</fpage><lpage>487</lpage><pub-id pub-id-type=\"doi\">10.1016/j.freeradbiomed.2011.11.003</pub-id><pub-id pub-id-type=\"pmid\">22142471</pub-id></element-citation></ref><ref id=\"B22-pharmaceuticals-16-00014\"><label>22.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Chen</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Zou</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Wu</surname><given-names>T.</given-names></name></person-group><article-title>The protective effect of glycyrrhetinic acid on carbon tetrachloride-induced chronic liver fibrosis in mice via upregulation of Nrf2</article-title><source>PLoS ONE</source><year>2013</year><volume>8</volume><elocation-id>e53662</elocation-id><pub-id pub-id-type=\"doi\">10.1371/journal.pone.0053662</pub-id><pub-id pub-id-type=\"pmid\">23341968</pub-id><pub-id pub-id-type=\"pmcid\">PMC3544925</pub-id></element-citation></ref><ref id=\"B23-pharmaceuticals-16-00014\"><label>23.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Yamamoto</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Kensler</surname><given-names>T.W.</given-names></name><name name-style=\"western\"><surname>Motohashi</surname><given-names>H.</given-names></name></person-group><article-title>The KEAP1-NRF2 system: A thiol-based sensor-effector apparatus for maintaining redox homeostasis</article-title><source>Physiol. Rev.</source><year>2018</year><volume>98</volume><fpage>1169</fpage><lpage>1203</lpage><pub-id pub-id-type=\"doi\">10.1152/physrev.00023.2017</pub-id><pub-id pub-id-type=\"pmid\">29717933</pub-id><pub-id pub-id-type=\"pmcid\">PMC9762786</pub-id></element-citation></ref><ref id=\"B24-pharmaceuticals-16-00014\"><label>24.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Mundal</surname><given-names>S.B.</given-names></name><name name-style=\"western\"><surname>Rakner</surname><given-names>J.J.</given-names></name><name name-style=\"western\"><surname>Silva</surname><given-names>G.B.</given-names></name><name name-style=\"western\"><surname>Gierman</surname><given-names>L.M.</given-names></name><name name-style=\"western\"><surname>Austdal</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Basnet</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Elschot</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Bakke</surname><given-names>S.S.</given-names></name><name name-style=\"western\"><surname>Ostrop</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Thomsen</surname><given-names>L.C.V.</given-names></name><etal/></person-group><article-title>Divergent Regulation of Decidual Oxidative-Stress Response by NRF2 and KEAP1 in Preeclampsia with and without Fetal Growth Restriction</article-title><source>Int. J. Mol. Sci.</source><year>2022</year><volume>10</volume><elocation-id>1966</elocation-id><pub-id pub-id-type=\"doi\">10.3390/ijms23041966</pub-id><pub-id pub-id-type=\"pmid\">35216082</pub-id><pub-id pub-id-type=\"pmcid\">PMC8875334</pub-id></element-citation></ref><ref id=\"B25-pharmaceuticals-16-00014\"><label>25.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Liby</surname><given-names>K.T.</given-names></name><name name-style=\"western\"><surname>Yore</surname><given-names>M.M.</given-names></name><name name-style=\"western\"><surname>Sporn</surname><given-names>M.B.</given-names></name></person-group><article-title>Triterpenoids and rexinoids as multifunctional agents for the prevention and treatment of cancer</article-title><source>Nat. Rev. Cancer</source><year>2007</year><volume>7</volume><fpage>357</fpage><lpage>369</lpage><pub-id pub-id-type=\"doi\">10.1038/nrc2129</pub-id><pub-id pub-id-type=\"pmid\">17446857</pub-id></element-citation></ref><ref id=\"B26-pharmaceuticals-16-00014\"><label>26.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Lu</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Zhou</surname><given-names>H.S.</given-names></name><name name-style=\"western\"><surname>You</surname><given-names>Q.D.</given-names></name><name name-style=\"western\"><surname>Jiang</surname><given-names>Z.</given-names></name></person-group><article-title>Design, synthesis, and initial evaluation of affinity-based small-molecule probes for fluorescent visualization and specific detection of Keap1</article-title><source>J. Med. Chem.</source><year>2016</year><volume>59</volume><fpage>7305</fpage><lpage>7310</lpage><pub-id pub-id-type=\"doi\">10.1021/acs.jmedchem.6b00775</pub-id><pub-id pub-id-type=\"pmid\">27409246</pub-id></element-citation></ref><ref id=\"B27-pharmaceuticals-16-00014\"><label>27.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Kim</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Indu Viswanath</surname><given-names>A.N.</given-names></name><name name-style=\"western\"><surname>Park</surname><given-names>J.H.</given-names></name><name name-style=\"western\"><surname>Lee</surname><given-names>H.E.</given-names></name><name name-style=\"western\"><surname>Park</surname><given-names>A.Y.</given-names></name><name name-style=\"western\"><surname>Choi</surname><given-names>J.W.</given-names></name><name name-style=\"western\"><surname>Kim</surname><given-names>H.J.</given-names></name><name name-style=\"western\"><surname>Londhe</surname><given-names>A.M.</given-names></name><name name-style=\"western\"><surname>Jang</surname><given-names>B.K.</given-names></name><name name-style=\"western\"><surname>Lee</surname><given-names>J.</given-names></name><etal/></person-group><article-title>Nrf2 activator via interference of Nrf2-Keap1 interaction has antioxidant and anti-inflammatory properties in Parkinson&#8217;s disease animal model</article-title><source>Neuropharmacology</source><year>2020</year><volume>167</volume><fpage>107989</fpage><pub-id pub-id-type=\"doi\">10.1016/j.neuropharm.2020.107989</pub-id><pub-id pub-id-type=\"pmid\">32032607</pub-id></element-citation></ref><ref id=\"B28-pharmaceuticals-16-00014\"><label>28.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Cuadrado</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Rojo</surname><given-names>A.I.</given-names></name><name name-style=\"western\"><surname>Wells</surname><given-names>G.</given-names></name><name name-style=\"western\"><surname>Hayes</surname><given-names>J.D.</given-names></name><name name-style=\"western\"><surname>Cousin</surname><given-names>S.P.</given-names></name><name name-style=\"western\"><surname>Rumsey</surname><given-names>W.L.</given-names></name><name name-style=\"western\"><surname>Attucks</surname><given-names>O.C.</given-names></name><name name-style=\"western\"><surname>Franklin</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Levonen</surname><given-names>A.L.</given-names></name><name name-style=\"western\"><surname>Kensler</surname><given-names>T.W.</given-names></name><etal/></person-group><article-title>Therapeutic targeting of the NRF2 and KEAP1 partnership in chronic diseases</article-title><source>Nat. Rev. Drug Discov.</source><year>2019</year><volume>18</volume><fpage>295</fpage><lpage>317</lpage><pub-id pub-id-type=\"doi\">10.1038/s41573-018-0008-x</pub-id><pub-id pub-id-type=\"pmid\">30610225</pub-id></element-citation></ref><ref id=\"B29-pharmaceuticals-16-00014\"><label>29.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Jiang</surname><given-names>Z.Y.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>L.L.</given-names></name><name name-style=\"western\"><surname>Lu</surname><given-names>M.C.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Z.Y.</given-names></name><name name-style=\"western\"><surname>Yuan</surname><given-names>Z.W.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>X.L.</given-names></name><name name-style=\"western\"><surname>Guo</surname><given-names>X.K.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>X.J.</given-names></name><name name-style=\"western\"><surname>Sun</surname><given-names>H.P.</given-names></name><name name-style=\"western\"><surname>You</surname><given-names>Q.D.</given-names></name></person-group><article-title>Structure&#8211;activity and structure&#8211;property relationship and exploratory in vivo evaluation of the nanomolar Keap1&#8211;Nrf2 protein&#8211;protein interaction inhibitor</article-title><source>J. Med. Chem.</source><year>2015</year><volume>58</volume><fpage>6410</fpage><lpage>6421</lpage><pub-id pub-id-type=\"doi\">10.1021/acs.jmedchem.5b00185</pub-id><pub-id pub-id-type=\"pmid\">26258437</pub-id></element-citation></ref><ref id=\"B30-pharmaceuticals-16-00014\"><label>30.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Li</surname><given-names>G.D.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Feng</surname><given-names>R.B.</given-names></name><name name-style=\"western\"><surname>Kang</surname><given-names>T.S.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>W.H.</given-names></name><name name-style=\"western\"><surname>Ko</surname><given-names>C.N.</given-names></name><name name-style=\"western\"><surname>Wong</surname><given-names>C.Y.</given-names></name><name name-style=\"western\"><surname>Ye</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Ma</surname><given-names>D.L.</given-names></name><name name-style=\"western\"><surname>Wan</surname><given-names>J.B.</given-names></name><etal/></person-group><article-title>A bioactive ligand-conjugated iridium (III) metal-based complex as a Keap1&#8211;Nrf2 protein-protein interaction inhibitor against acetaminophen-induced acute liver injury</article-title><source>Redox Biol.</source><year>2021</year><volume>48</volume><fpage>102129</fpage><pub-id pub-id-type=\"doi\">10.1016/j.redox.2021.102129</pub-id><pub-id pub-id-type=\"pmid\">34526248</pub-id><pub-id pub-id-type=\"pmcid\">PMC8710994</pub-id></element-citation></ref><ref id=\"B31-pharmaceuticals-16-00014\"><label>31.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Sun</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Han</surname><given-names>B.</given-names></name><name name-style=\"western\"><surname>Zhai</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Zhao</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Qian</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Hao</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>Q.</given-names></name><name name-style=\"western\"><surname>Shen</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Kai</surname><given-names>G.</given-names></name></person-group><article-title>Dihydrotanshinone I inhibits ovarian tumor growth by activating oxidative stress through Keap1-mediated Nrf2 ubiquitination degradation</article-title><source>Free Radic. Biol. Med.</source><year>2022</year><volume>20</volume><fpage>220</fpage><lpage>235</lpage><pub-id pub-id-type=\"doi\">10.1016/j.freeradbiomed.2022.01.015</pub-id><pub-id pub-id-type=\"pmid\">35074488</pub-id></element-citation></ref><ref id=\"B32-pharmaceuticals-16-00014\"><label>32.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Leung</surname><given-names>C.H.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>J.T.</given-names></name><name name-style=\"western\"><surname>Yang</surname><given-names>G.J.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Han</surname><given-names>Q.B.</given-names></name><name name-style=\"western\"><surname>Ma</surname><given-names>D.L.</given-names></name></person-group><article-title>Emerging Screening Approaches in the Development of Nrf2&#8211;Keap1 Protein&#8211;Protein Interaction Inhibitors</article-title><source>Int. J. Mol. Sci.</source><year>2019</year><volume>20</volume><fpage>4445</fpage><lpage>4464</lpage><pub-id pub-id-type=\"doi\">10.3390/ijms20184445</pub-id><pub-id pub-id-type=\"pmid\">31509940</pub-id><pub-id pub-id-type=\"pmcid\">PMC6770765</pub-id></element-citation></ref><ref id=\"B33-pharmaceuticals-16-00014\"><label>33.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Feng</surname><given-names>R.B.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>He</surname><given-names>C.W.</given-names></name><name name-style=\"western\"><surname>Yang</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Wan</surname><given-names>J.B.</given-names></name></person-group><article-title>Gallic acid, a natural polyphenol, protects against tert-butyl hydroperoxide-induced hepatotoxicity by activating ERK-Nrf2-Keap1-mediated antioxidative response</article-title><source>Food Chem. Toxicol.</source><year>2018</year><volume>119</volume><fpage>479</fpage><lpage>488</lpage><pub-id pub-id-type=\"doi\">10.1016/j.fct.2017.10.033</pub-id><pub-id pub-id-type=\"pmid\">29066411</pub-id></element-citation></ref><ref id=\"B34-pharmaceuticals-16-00014\"><label>34.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Noori</surname><given-names>M.S.</given-names></name><name name-style=\"western\"><surname>Bhatt</surname><given-names>P.M.</given-names></name><name name-style=\"western\"><surname>Courreges</surname><given-names>M.C.</given-names></name><name name-style=\"western\"><surname>Ghazanfari</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Cuckler</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Orac</surname><given-names>C.M.</given-names></name><name name-style=\"western\"><surname>McMills</surname><given-names>M.C.</given-names></name><name name-style=\"western\"><surname>Schwartz</surname><given-names>F.L.</given-names></name><name name-style=\"western\"><surname>Deosarkar</surname><given-names>S.P.</given-names></name><name name-style=\"western\"><surname>Bergmeier</surname><given-names>S.C.</given-names></name><etal/></person-group><article-title>Identification of a novel selective and potent inhibitor of glycogen synthase kinase-3</article-title><source>Am. J. Physiol.-Cell Physiol.</source><year>2019</year><volume>317</volume><fpage>C1289</fpage><lpage>C1303</lpage><pub-id pub-id-type=\"doi\">10.1152/ajpcell.00061.2019</pub-id><pub-id pub-id-type=\"pmid\">31553649</pub-id><pub-id pub-id-type=\"pmcid\">PMC6962522</pub-id></element-citation></ref><ref id=\"B35-pharmaceuticals-16-00014\"><label>35.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Chang</surname><given-names>T.T.</given-names></name><name name-style=\"western\"><surname>Chou</surname><given-names>W.N.</given-names></name></person-group><article-title>Antrodia cinnamomea reconsidered and <italic toggle=\"yes\">A. salmonea</italic> sp. nov. on Cunninghamia konishii in Taiwan</article-title><source>Bot. Bull. Acad. Sin.</source><year>2004</year><volume>45</volume><fpage>347</fpage><lpage>352</lpage></element-citation></ref><ref id=\"B36-pharmaceuticals-16-00014\"><label>36.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Li</surname><given-names>H.X.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>J.J.</given-names></name><name name-style=\"western\"><surname>Lu</surname><given-names>C.L.</given-names></name><name name-style=\"western\"><surname>Gao</surname><given-names>Y.J.</given-names></name><name name-style=\"western\"><surname>Gao</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Yang</surname><given-names>Z.Q.</given-names></name></person-group><article-title>Review of Bioactivity, Isolation, and Identification of Active Compounds from Antrodia cinnamomea</article-title><source>Bioengineering</source><year>2022</year><volume>9</volume><elocation-id>494</elocation-id><pub-id pub-id-type=\"doi\">10.3390/bioengineering9100494</pub-id><pub-id pub-id-type=\"pmid\">36290462</pub-id><pub-id pub-id-type=\"pmcid\">PMC9598228</pub-id></element-citation></ref><ref id=\"B37-pharmaceuticals-16-00014\"><label>37.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Li</surname><given-names>B.</given-names></name><name name-style=\"western\"><surname>Kuang</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>He</surname><given-names>J.B.</given-names></name><name name-style=\"western\"><surname>Tang</surname><given-names>R.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>L.L.</given-names></name><name name-style=\"western\"><surname>Leung</surname><given-names>C.H.</given-names></name><name name-style=\"western\"><surname>Ma</surname><given-names>D.L.</given-names></name><name name-style=\"western\"><surname>Qiao</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Ye</surname><given-names>M.</given-names></name></person-group><article-title>Antcamphorols A-K, Cytotoxic and ROS Scavenging Triterpenoids from <italic toggle=\"yes\">Antrodia camphorata</italic></article-title><source>J. Nat. Prod.</source><year>2020</year><volume>83</volume><fpage>45</fpage><lpage>54</lpage><pub-id pub-id-type=\"doi\">10.1021/acs.jnatprod.9b00580</pub-id><pub-id pub-id-type=\"pmid\">31891260</pub-id></element-citation></ref><ref id=\"B38-pharmaceuticals-16-00014\"><label>38.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Yang</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Lin</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Lim</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Cao</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Zheng</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Fu</surname><given-names>K.C.</given-names></name><etal/></person-group><article-title>Structure and Anti-Inflammatory Activity Relationship of Ergostanes and Lanostanes in Antrodia cinnamomea</article-title><source>Foods</source><year>2022</year><volume>11</volume><elocation-id>1831</elocation-id><pub-id pub-id-type=\"doi\">10.3390/foods11131831</pub-id><pub-id pub-id-type=\"pmid\">35804645</pub-id><pub-id pub-id-type=\"pmcid\">PMC9266224</pub-id></element-citation></ref><ref id=\"B39-pharmaceuticals-16-00014\"><label>39.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Xu</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Peng</surname><given-names>A.K.</given-names></name><name name-style=\"western\"><surname>Cao</surname><given-names>Y.N.</given-names></name><name name-style=\"western\"><surname>Qiao</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Yue</surname><given-names>S.S.</given-names></name><name name-style=\"western\"><surname>Ye</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Qi</surname><given-names>R.</given-names></name></person-group><article-title>Protective Effects of Antrodia cinnamomea and Its Constituent Compound Dehydroeburicoic Acid 32 against Alcoholic Fatty Liver Disease</article-title><source>Curr. Mol. Pharmacol.</source><year>2021</year><volume>14</volume><fpage>871</fpage><lpage>882</lpage><pub-id pub-id-type=\"doi\">10.2174/1874467214666210120152140</pub-id><pub-id pub-id-type=\"pmid\">33494689</pub-id></element-citation></ref><ref id=\"B40-pharmaceuticals-16-00014\"><label>40.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Cao</surname><given-names>Y.N.</given-names></name><name name-style=\"western\"><surname>Yue</surname><given-names>S.S.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>A.Y.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Hu</surname><given-names>Y.T.</given-names></name><name name-style=\"western\"><surname>Qiao</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Wu</surname><given-names>T.Y.</given-names></name><name name-style=\"western\"><surname>Ye</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Wu</surname><given-names>Y.C.</given-names></name><name name-style=\"western\"><surname>Qi</surname><given-names>R.</given-names></name></person-group><article-title>Antrodia cinnamomea and its compound dehydroeburicoic acid attenuate nonalcoholic fatty liver disease by upregulating ALDH2 activity</article-title><source>J. Ethnopharmacol.</source><year>2022</year><volume>28</volume><fpage>115146</fpage><pub-id pub-id-type=\"doi\">10.1016/j.jep.2022.115146</pub-id><pub-id pub-id-type=\"pmid\">35304272</pub-id></element-citation></ref><ref id=\"B41-pharmaceuticals-16-00014\"><label>41.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>To</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Roy</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Chan</surname><given-names>E.</given-names></name><name name-style=\"western\"><surname>Prado</surname><given-names>M.A.M.</given-names></name><name name-style=\"western\"><surname>Guglielmo</surname><given-names>G.M.D.</given-names></name></person-group><article-title>Synthetic triterpenoids inhibit GSK3&#946; activity and localization and affect focal adhesions and cell migration</article-title><source>Biochim. Biophys. Acta-Mol. Cell Res.</source><year>2017</year><volume>1864</volume><fpage>1274</fpage><lpage>1284</lpage><pub-id pub-id-type=\"doi\">10.1016/j.bbamcr.2017.03.012</pub-id><pub-id pub-id-type=\"pmid\">28366661</pub-id></element-citation></ref><ref id=\"B42-pharmaceuticals-16-00014\"><label>42.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Jacobs</surname><given-names>K.M.</given-names></name><name name-style=\"western\"><surname>Bhave</surname><given-names>S.R.</given-names></name><name name-style=\"western\"><surname>Ferraro</surname><given-names>D.J.</given-names></name><name name-style=\"western\"><surname>Jaboin</surname><given-names>J.J.</given-names></name><name name-style=\"western\"><surname>Hallahan</surname><given-names>D.E.</given-names></name><name name-style=\"western\"><surname>Thotala</surname><given-names>D.</given-names></name></person-group><article-title>GSK-3&#946;: A Bifunctional Role in Cell Death Pathways</article-title><source>Int J Cell Biol.</source><year>2012</year><volume>2012</volume><fpage>930710</fpage><lpage>930722</lpage><pub-id pub-id-type=\"doi\">10.1155/2012/930710</pub-id><pub-id pub-id-type=\"pmid\">22675363</pub-id><pub-id pub-id-type=\"pmcid\">PMC3364548</pub-id></element-citation></ref><ref id=\"B43-pharmaceuticals-16-00014\"><label>43.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Ven&#232;</surname><given-names>R.</given-names></name><name name-style=\"western\"><surname>Larghero</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Arena</surname><given-names>G.</given-names></name><name name-style=\"western\"><surname>Sporn</surname><given-names>M.B.</given-names></name><name name-style=\"western\"><surname>Albini</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Tosetti</surname><given-names>F.</given-names></name></person-group><article-title>Glycogen synthase kinase 3beta regulates cell death induced by synthetic triterpenoids</article-title><source>Cancer Res.</source><year>2008</year><volume>68</volume><fpage>6987</fpage><lpage>6996</lpage><pub-id pub-id-type=\"doi\">10.1158/0008-5472.CAN-07-6362</pub-id><pub-id pub-id-type=\"pmid\">18757413</pub-id></element-citation></ref><ref id=\"B44-pharmaceuticals-16-00014\"><label>44.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Kamble</surname><given-names>S.M.</given-names></name><name name-style=\"western\"><surname>Patel</surname><given-names>H.M.</given-names></name><name name-style=\"western\"><surname>Goyal</surname><given-names>S.N.</given-names></name><name name-style=\"western\"><surname>Noolvi</surname><given-names>M.N.</given-names></name><name name-style=\"western\"><surname>Mahajan</surname><given-names>U.B.</given-names></name><name name-style=\"western\"><surname>Ojha</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Patil</surname><given-names>C.R.</given-names></name></person-group><article-title>In silico Evidence for Binding of Pentacyclic Triterpenoids to Keap1-Nrf2 Protein-Protein Binding Site</article-title><source>Comb. Chem. High Throughput Screen.</source><year>2017</year><volume>20</volume><fpage>215</fpage><lpage>234</lpage><pub-id pub-id-type=\"doi\">10.2174/1386207319666161214111822</pub-id><pub-id pub-id-type=\"pmid\">28024463</pub-id></element-citation></ref><ref id=\"B45-pharmaceuticals-16-00014\"><label>45.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Kou</surname><given-names>R.W.</given-names></name><name name-style=\"western\"><surname>Xia</surname><given-names>B.</given-names></name><name name-style=\"western\"><surname>Han</surname><given-names>R.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>Z.Q.</given-names></name><name name-style=\"western\"><surname>Yang</surname><given-names>J.R.</given-names></name><name name-style=\"western\"><surname>Yin</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Gao</surname><given-names>Y.Q.</given-names></name><name name-style=\"western\"><surname>Gao</surname><given-names>J.M.</given-names></name></person-group><article-title>Neuroprotective effects of a new triterpenoid from edible mushroom on oxidative stress and apoptosis through the BDNF/TrkB/ERK/CREB and Nrf2 signaling pathway in vitro and in vivo</article-title><source>Food Funct.</source><year>2022</year><volume>13</volume><fpage>12121</fpage><lpage>12134</lpage><pub-id pub-id-type=\"doi\">10.1039/D2FO02854A</pub-id><pub-id pub-id-type=\"pmid\">36321740</pub-id></element-citation></ref><ref id=\"B46-pharmaceuticals-16-00014\"><label>46.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Patyar</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Prakash</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Medhi</surname><given-names>B.</given-names></name></person-group><article-title>Dual inhibition: A novel promising pharmacological approach for different disease conditions</article-title><source>J. Pharm. Pharmacol.</source><year>2011</year><volume>63</volume><fpage>459</fpage><lpage>471</lpage><pub-id pub-id-type=\"doi\">10.1111/j.2042-7158.2010.01236.x</pub-id><pub-id pub-id-type=\"pmid\">21401597</pub-id></element-citation></ref><ref id=\"B47-pharmaceuticals-16-00014\"><label>47.</label><element-citation publication-type=\"book\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Wang</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Lewis</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>Y.L.</given-names></name><name name-style=\"western\"><surname>Khodier</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Magesh</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Inoyama</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Zhen</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Hu</surname><given-names>L.Q.</given-names></name></person-group><article-title>The identification and characterization of non-reactive inhibitor of Keap1-Nrf2 interaction through HTS using a fluorescence polarization assay</article-title><source>Probe Reports from the NIH Molecular Libraries Program</source><publisher-name>National Center for Biotechnology Information (US)</publisher-name><publisher-loc>Bethesda, MD, USA</publisher-loc><year>2010</year><pub-id pub-id-type=\"pmid\">24260785</pub-id></element-citation></ref><ref id=\"B48-pharmaceuticals-16-00014\"><label>48.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Kobayashi</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Yamamoto</surname><given-names>M.</given-names></name></person-group><article-title>Molecular mechanisms activating the Nrf2-Keap1 pathway of antioxidant gene regulation</article-title><source>Antioxid. Redox Signal.</source><year>2005</year><volume>7</volume><fpage>385</fpage><lpage>394</lpage><pub-id pub-id-type=\"doi\">10.1089/ars.2005.7.385</pub-id><pub-id pub-id-type=\"pmid\">15706085</pub-id></element-citation></ref><ref id=\"B49-pharmaceuticals-16-00014\"><label>49.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Lu</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Qiao</surname><given-names>Y.J.</given-names></name><name name-style=\"western\"><surname>Ge</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Flickinger</surname><given-names>B.</given-names></name><name name-style=\"western\"><surname>Malhotrac</surname><given-names>D.K.</given-names></name><name name-style=\"western\"><surname>Dworkin</surname><given-names>L.D.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>Z.S.</given-names></name><name name-style=\"western\"><surname>Gong</surname><given-names>R.J.</given-names></name></person-group><article-title>GSK3&#946;-mediated Keap1-independent regulation of Nrf2 antioxidant response: A molecular rheostat of acute kidney injury to chronic kidney disease transition</article-title><source>Redox Biol.</source><year>2019</year><volume>26</volume><fpage>101275</fpage><lpage>101291</lpage><pub-id pub-id-type=\"doi\">10.1016/j.redox.2019.101275</pub-id><pub-id pub-id-type=\"pmid\">31349118</pub-id><pub-id pub-id-type=\"pmcid\">PMC6669347</pub-id></element-citation></ref><ref id=\"B50-pharmaceuticals-16-00014\"><label>50.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Qiu</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Cai</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Zhao</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Fang</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Tang</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Guo</surname><given-names>C.</given-names></name></person-group><article-title>Inhibition of aldose reductase ameliorates ethanol-induced steatosis in HepG2 cells</article-title><source>Mol. Med. Rep.</source><year>2017</year><volume>15</volume><fpage>2732</fpage><lpage>2736</lpage><pub-id pub-id-type=\"doi\">10.3892/mmr.2017.6313</pub-id><pub-id pub-id-type=\"pmid\">28447762</pub-id></element-citation></ref><ref id=\"B51-pharmaceuticals-16-00014\"><label>51.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Lled&#237;as</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Hansberg</surname><given-names>W.</given-names></name></person-group><article-title>Oxidation of human catalase by singlet oxygen in myeloid leukemia cells</article-title><source>Photochem. Photobiol.</source><year>1999</year><volume>70</volume><fpage>887</fpage><lpage>892</lpage><pub-id pub-id-type=\"doi\">10.1111/j.1751-1097.1999.tb08298.x</pub-id><pub-id pub-id-type=\"pmid\">10628300</pub-id></element-citation></ref><ref id=\"B52-pharmaceuticals-16-00014\"><label>52.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Ren</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Sha</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Shang</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Yuan</surname><given-names>E.</given-names></name></person-group><article-title>Hepatoprotective peptides purified from Corbicula fluminea and its effect against ethanol-induced LO2 cells injury</article-title><source>Int. J. Food Sci. Technol.</source><year>2021</year><volume>56</volume><fpage>352</fpage><lpage>361</lpage><pub-id pub-id-type=\"doi\">10.1111/ijfs.14649</pub-id></element-citation></ref><ref id=\"B53-pharmaceuticals-16-00014\"><label>53.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Kang</surname><given-names>T.C.</given-names></name></person-group><article-title>Nuclear Factor-Erythroid 2-Related Factor 2 (Nrf2) and Mitochondrial Dynamics/Mitophagy in Neurological Diseases</article-title><source>Antioxidants</source><year>2020</year><volume>9</volume><fpage>617</fpage><lpage>637</lpage><pub-id pub-id-type=\"doi\">10.3390/antiox9070617</pub-id><pub-id pub-id-type=\"pmid\">32679689</pub-id><pub-id pub-id-type=\"pmcid\">PMC7402121</pub-id></element-citation></ref><ref id=\"B54-pharmaceuticals-16-00014\"><label>54.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Ping</surname><given-names>Z.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>L.F.</given-names></name><name name-style=\"western\"><surname>Cui</surname><given-names>Y.J.</given-names></name><name name-style=\"western\"><surname>Chang</surname><given-names>Y.M.</given-names></name><name name-style=\"western\"><surname>Jiang</surname><given-names>C.W.</given-names></name><name name-style=\"western\"><surname>Meng</surname><given-names>Z.Z.</given-names></name><name name-style=\"western\"><surname>Xu</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>H.Y.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>D.Y.</given-names></name><name name-style=\"western\"><surname>Cao</surname><given-names>X.B.</given-names></name></person-group><article-title>The protective effects of salidroside from exhaustive exercise-induced heart injury by enhancing the PGC-1&#945;&#8211;NRF1/NRF2 pathway and mitochondrial respiratory function in rats</article-title><source>Oxidative Med. Cell. Longev.</source><year>2015</year><volume>2015</volume><fpage>876825</fpage><lpage>876834</lpage><pub-id pub-id-type=\"doi\">10.1155/2015/876825</pub-id><pub-id pub-id-type=\"pmid\">26167242</pub-id><pub-id pub-id-type=\"pmcid\">PMC4488012</pub-id></element-citation></ref><ref id=\"B55-pharmaceuticals-16-00014\"><label>55.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Yeligar</surname><given-names>S.M.</given-names></name><name name-style=\"western\"><surname>Machida</surname><given-names>K.</given-names></name><name name-style=\"western\"><surname>Kalra</surname><given-names>V.K.</given-names></name></person-group><article-title>Ethanol-induced HO-1 and NQO1 are differentially regulated by HIF-1alpha and Nrf2 to attenuate inflammatory cytokine expression</article-title><source>J. Biol. Chem.</source><year>2010</year><volume>285</volume><fpage>35359</fpage><lpage>35373</lpage><pub-id pub-id-type=\"doi\">10.1074/jbc.M110.138636</pub-id><pub-id pub-id-type=\"pmid\">20833713</pub-id><pub-id pub-id-type=\"pmcid\">PMC2975160</pub-id></element-citation></ref><ref id=\"B56-pharmaceuticals-16-00014\"><label>56.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>El-Assal</surname><given-names>O.</given-names></name><name name-style=\"western\"><surname>Hong</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Kim</surname><given-names>W.H.</given-names></name><name name-style=\"western\"><surname>Radaeva</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Gao</surname><given-names>B.</given-names></name></person-group><article-title>IL-6-deficient mice are susceptible to ethanol-induced hepatic steatosis: IL-6 protects against ethanol-induced oxidative stress and mitochondrial permeability transition in the liver</article-title><source>Cell. Mol. Immunol.</source><year>2004</year><volume>1</volume><fpage>205</fpage><lpage>211</lpage><pub-id pub-id-type=\"pmid\">16219169</pub-id></element-citation></ref><ref id=\"B57-pharmaceuticals-16-00014\"><label>57.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Dudonne</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Vitrac</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Coutiere</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Woillez</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>M&#233;rillon</surname><given-names>J.M.</given-names></name></person-group><article-title>Comparative study of antioxidant properties and total phenolic content of 30 plant extracts of industrial interest using DPPH, ABTS, FRAP, SOD, and ORAC assays</article-title><source>J. Agric. Food Chem.</source><year>2009</year><volume>57</volume><fpage>1768</fpage><lpage>1774</lpage><pub-id pub-id-type=\"doi\">10.1021/jf803011r</pub-id><pub-id pub-id-type=\"pmid\">19199445</pub-id></element-citation></ref><ref id=\"B58-pharmaceuticals-16-00014\"><label>58.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Wang</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>W.Y.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Gobejishvili</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Barve</surname><given-names>S.S.</given-names></name><name name-style=\"western\"><surname>McClain</surname><given-names>C.J.</given-names></name><name name-style=\"western\"><surname>Joshi-Barve</surname><given-names>S.</given-names></name></person-group><article-title>Elevated Fructose and Uric Acid through Aldose Reductase Contribute to Experimental and Human Alcoholic Liver Disease</article-title><source>Hepatology.</source><year>2020</year><volume>72</volume><fpage>1617</fpage><lpage>1637</lpage><pub-id pub-id-type=\"doi\">10.1002/hep.31197</pub-id><pub-id pub-id-type=\"pmid\">32086945</pub-id></element-citation></ref><ref id=\"B59-pharmaceuticals-16-00014\"><label>59.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Zhou</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>P.</given-names></name><name name-style=\"western\"><surname>Qiao</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Ge</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>Y.Z.</given-names></name><name name-style=\"western\"><surname>Quan</surname><given-names>S.X.</given-names></name><name name-style=\"western\"><surname>Yao</surname><given-names>R.</given-names></name><name name-style=\"western\"><surname>Zhuang</surname><given-names>S.G.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>L.J.</given-names></name><name name-style=\"western\"><surname>Du</surname><given-names>Y.</given-names></name><etal/></person-group><article-title>Genetic and pharmacologic targeting of glycogen synthase kinase 3&#946; reinforces the Nrf2 antioxidant defense against podocytopathy</article-title><source>J. Am. Soc. Nephrol.</source><year>2016</year><volume>27</volume><fpage>2289</fpage><lpage>2308</lpage><pub-id pub-id-type=\"doi\">10.1681/ASN.2015050565</pub-id><pub-id pub-id-type=\"pmid\">26647425</pub-id><pub-id pub-id-type=\"pmcid\">PMC4978041</pub-id></element-citation></ref><ref id=\"B60-pharmaceuticals-16-00014\"><label>60.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Bryan</surname><given-names>H.K.</given-names></name><name name-style=\"western\"><surname>Olayanju</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Goldring</surname><given-names>C.E.</given-names></name><name name-style=\"western\"><surname>Park</surname><given-names>B.K.</given-names></name></person-group><article-title>The Nrf2 cell defence pathway: Keap1-dependent and-independent mechanisms of regulation</article-title><source>Biochem. Pharmacol.</source><year>2013</year><volume>85</volume><fpage>705</fpage><lpage>717</lpage><pub-id pub-id-type=\"doi\">10.1016/j.bcp.2012.11.016</pub-id><pub-id pub-id-type=\"pmid\">23219527</pub-id></element-citation></ref><ref id=\"B61-pharmaceuticals-16-00014\"><label>61.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Soni</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Kumar</surname><given-names>P.</given-names></name></person-group><article-title>GSK-3&#946;-mediated regulation of Nrf2/HO-1 signaling as a new therapeutic approach in the treatment of movement disorders</article-title><source>Pharmacol. Rep.</source><year>2022</year><volume>74</volume><fpage>557</fpage><lpage>569</lpage><pub-id pub-id-type=\"doi\">10.1007/s43440-022-00390-z</pub-id><pub-id pub-id-type=\"pmid\">35882765</pub-id></element-citation></ref><ref id=\"B62-pharmaceuticals-16-00014\"><label>62.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Yousef</surname><given-names>M.H.</given-names></name><name name-style=\"western\"><surname>Salama</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>El-Fawal</surname><given-names>H.A.N.</given-names></name><name name-style=\"western\"><surname>Abdelnaser</surname><given-names>A.</given-names></name></person-group><article-title>Selective GSK3&#946; Inhibition Mediates an Nrf2-Independent Anti-inflammatory Microglial Response</article-title><source>Mol. Neurobiol.</source><year>2022</year><volume>59</volume><fpage>5591</fpage><lpage>5611</lpage><pub-id pub-id-type=\"doi\">10.1007/s12035-022-02923-2</pub-id><pub-id pub-id-type=\"pmid\">35739410</pub-id><pub-id pub-id-type=\"pmcid\">PMC9395457</pub-id></element-citation></ref><ref id=\"B63-pharmaceuticals-16-00014\"><label>63.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Li</surname><given-names>G.D.</given-names></name><name name-style=\"western\"><surname>Boyle</surname><given-names>J.W.</given-names></name><name name-style=\"western\"><surname>Ko</surname><given-names>C.N.</given-names></name><name name-style=\"western\"><surname>Zeng</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Wong</surname><given-names>V.K.W.</given-names></name><name name-style=\"western\"><surname>Wan</surname><given-names>J.B.</given-names></name><name name-style=\"western\"><surname>Chan</surname><given-names>P.W.H.</given-names></name><name name-style=\"western\"><surname>Ma</surname><given-names>D.L.</given-names></name><name name-style=\"western\"><surname>Leung</surname><given-names>C.H.</given-names></name></person-group><article-title>Aurone derivatives as Vps34 inhibitors that modulate autophagy</article-title><source>Acta Pharm. Sin. B</source><year>2019</year><volume>9</volume><fpage>537</fpage><lpage>544</lpage><pub-id pub-id-type=\"doi\">10.1016/j.apsb.2019.01.016</pub-id><pub-id pub-id-type=\"pmid\">31193773</pub-id><pub-id pub-id-type=\"pmcid\">PMC6543056</pub-id></element-citation></ref><ref id=\"B64-pharmaceuticals-16-00014\"><label>64.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Chou</surname><given-names>M.C.</given-names></name><name name-style=\"western\"><surname>Chang</surname><given-names>R.</given-names></name><name name-style=\"western\"><surname>Hung</surname><given-names>Y.H.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Y.C.</given-names></name><name name-style=\"western\"><surname>Chiu</surname><given-names>C.H.</given-names></name></person-group><article-title>Antrodia camphorata ameliorates high-fat-diet induced hepatic steatosis via improving lipid metabolism and antioxidative status</article-title><source>J. Funct. Foods</source><year>2013</year><volume>5</volume><fpage>1317</fpage><lpage>1325</lpage><pub-id pub-id-type=\"doi\">10.1016/j.jff.2013.04.019</pub-id></element-citation></ref><ref id=\"B65-pharmaceuticals-16-00014\"><label>65.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Tien</surname><given-names>A.J.</given-names></name><name name-style=\"western\"><surname>Chien</surname><given-names>C.Y.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Y.H.</given-names></name><name name-style=\"western\"><surname>Lin</surname><given-names>L.C.</given-names></name><name name-style=\"western\"><surname>Chien</surname><given-names>C.T.</given-names></name></person-group><article-title>Fruiting Bodies of Antrodia cinnamomea and Its Active Triterpenoid, Antcin K, Ameliorates N-Nitrosodiethylamine-Induced Hepatic Inflammation, Fibrosis and Carcinogenesis in Rats</article-title><source>Am. J. Chin. Med.</source><year>2017</year><volume>45</volume><fpage>173</fpage><lpage>198</lpage><pub-id pub-id-type=\"doi\">10.1142/S0192415X17500124</pub-id><pub-id pub-id-type=\"pmid\">28081627</pub-id></element-citation></ref><ref id=\"B66-pharmaceuticals-16-00014\"><label>66.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Wang</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>W.H.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>R.</given-names></name><name name-style=\"western\"><surname>Ge</surname><given-names>Y.P.</given-names></name><name name-style=\"western\"><surname>Yang</surname><given-names>S.D.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Wu</surname><given-names>Q.P.</given-names></name><name name-style=\"western\"><surname>Cheng</surname><given-names>X.H.</given-names></name></person-group><article-title>Study on Characteristics of Triterpenoids and Hepatoprotective Effects of Fruit Body of Stout Camphor Mushroom, Taiwanofungus camphoratus (Agaricomycetes), Cultivated with Apple-Wood</article-title><source>Int. J. Med. Mushrooms</source><year>2022</year><volume>24</volume><fpage>53</fpage><lpage>65</lpage><pub-id pub-id-type=\"doi\">10.1615/IntJMedMushrooms.2022044118</pub-id><pub-id pub-id-type=\"pmid\">35993961</pub-id></element-citation></ref><ref id=\"B67-pharmaceuticals-16-00014\"><label>67.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Huang</surname><given-names>G.J.</given-names></name><name name-style=\"western\"><surname>Deng</surname><given-names>J.S.</given-names></name><name name-style=\"western\"><surname>Huang</surname><given-names>S.S.</given-names></name><name name-style=\"western\"><surname>Lee</surname><given-names>C.Y.</given-names></name><name name-style=\"western\"><surname>Hou</surname><given-names>W.C.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>S.Y.</given-names></name><name name-style=\"western\"><surname>Sung</surname><given-names>P.J.</given-names></name><name name-style=\"western\"><surname>Kuo</surname><given-names>Y.H.</given-names></name></person-group><article-title>Hepatoprotective effects of eburicoic acid and dehydroeburicoic acid from Antrodia camphorata in a mouse model of acute hepatic injury</article-title><source>Food Chem.</source><year>2013</year><volume>141</volume><fpage>3020</fpage><lpage>3027</lpage><pub-id pub-id-type=\"doi\">10.1016/j.foodchem.2013.03.061</pub-id><pub-id pub-id-type=\"pmid\">23871054</pub-id></element-citation></ref><ref id=\"B68-pharmaceuticals-16-00014\"><label>68.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Yi</surname><given-names>Z.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Liang</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>G.</given-names></name><name name-style=\"western\"><surname>Xiong</surname><given-names>Z.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Song</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Ai</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Xia</surname><given-names>Y.</given-names></name></person-group><article-title>Antrodin A from Antrodia camphorata modulates the gut microbiome and liver metabolome in mice exposed to acute alcohol intake</article-title><source>Food Funct.</source><year>2021</year><volume>12</volume><fpage>2925</fpage><lpage>2937</lpage><pub-id pub-id-type=\"doi\">10.1039/D0FO03345F</pub-id><pub-id pub-id-type=\"pmid\">33720247</pub-id></element-citation></ref><ref id=\"B69-pharmaceuticals-16-00014\"><label>69.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Huang</surname><given-names>C.H.</given-names></name><name name-style=\"western\"><surname>Chang</surname><given-names>Y.Y.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>C.W.</given-names></name><name name-style=\"western\"><surname>Kang</surname><given-names>W.Y.</given-names></name><name name-style=\"western\"><surname>Lin</surname><given-names>Y.L.</given-names></name><name name-style=\"western\"><surname>Chang</surname><given-names>H.C.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Y.C.</given-names></name></person-group><article-title>Fruiting body of Niuchangchih (<italic toggle=\"yes\">Antrodia camphorata)</italic> protects livers against chronic alcohol consumption damage</article-title><source>J. Agric. Food Chem.</source><year>2010</year><volume>58</volume><fpage>3859</fpage><lpage>3866</lpage><pub-id pub-id-type=\"doi\">10.1021/jf100530c</pub-id><pub-id pub-id-type=\"pmid\">20192205</pub-id></element-citation></ref><ref id=\"B70-pharmaceuticals-16-00014\"><label>70.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Qin</surname><given-names>J.J.</given-names></name><name name-style=\"western\"><surname>Cheng</surname><given-names>X.D.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>W.D.</given-names></name></person-group><article-title>Dual roles and therapeutic potential of Keap1-Nrf2 pathway in pancreatic cancer: A systematic review</article-title><source>Cell Commun. Signal.</source><year>2019</year><volume>17</volume><fpage>1</fpage><lpage>15</lpage><pub-id pub-id-type=\"doi\">10.1186/s12964-019-0435-2</pub-id><pub-id pub-id-type=\"pmid\">31511020</pub-id><pub-id pub-id-type=\"pmcid\">PMC6740038</pub-id></element-citation></ref><ref id=\"B71-pharmaceuticals-16-00014\"><label>71.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Lu</surname><given-names>M.C.</given-names></name><name name-style=\"western\"><surname>Zhao</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>Y.T.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Tao</surname><given-names>M.M.</given-names></name><name name-style=\"western\"><surname>You</surname><given-names>Q.D.</given-names></name><name name-style=\"western\"><surname>Jiang</surname><given-names>Z.Y.</given-names></name></person-group><article-title>CPUY192018, a potent inhibitor of the Keap1-Nrf2 protein-protein interaction, alleviates renal inflammation in mice by restricting oxidative stress and NF-&#954;B activation</article-title><source>Redox Biol.</source><year>2019</year><volume>26</volume><fpage>101266</fpage><pub-id pub-id-type=\"doi\">10.1016/j.redox.2019.101266</pub-id><pub-id pub-id-type=\"pmid\">31279986</pub-id><pub-id pub-id-type=\"pmcid\">PMC6614503</pub-id></element-citation></ref><ref id=\"B72-pharmaceuticals-16-00014\"><label>72.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Sun</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>He</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Hua</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Qin</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Wang</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Gu</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>T.</given-names></name><name name-style=\"western\"><surname>Li</surname><given-names>N.</given-names></name><etal/></person-group><article-title>Activation of p62-Keap1-Nrf2 Pathway Protects 6-Hydroxydopamine-Induced Ferroptosis in Dopaminergic Cells</article-title><source>Mol. Neurobiol.</source><year>2020</year><volume>57</volume><fpage>4628</fpage><lpage>4641</lpage><pub-id pub-id-type=\"doi\">10.1007/s12035-020-02049-3</pub-id><pub-id pub-id-type=\"pmid\">32770451</pub-id></element-citation></ref><ref id=\"B73-pharmaceuticals-16-00014\"><label>73.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Sun</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Huang</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Chen</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Shang</surname><given-names>H.</given-names></name><name name-style=\"western\"><surname>Zhang</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Yu</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>He</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Xing</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Zhuang</surname><given-names>C.</given-names></name></person-group><article-title>Direct inhibition of Keap1-Nrf2 Protein-Protein interaction as a potential therapeutic strategy for Alzheimer&#8217;s disease</article-title><source>Bioorganic Chem.</source><year>2020</year><volume>103</volume><fpage>104172</fpage><pub-id pub-id-type=\"doi\">10.1016/j.bioorg.2020.104172</pub-id><pub-id pub-id-type=\"pmid\">32890991</pub-id></element-citation></ref><ref id=\"B74-pharmaceuticals-16-00014\"><label>74.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Di Martino</surname><given-names>R.M.C.</given-names></name><name name-style=\"western\"><surname>Pruccoli</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Bisi</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Gobbi</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Rampa</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>Martinez</surname><given-names>A.</given-names></name><name name-style=\"western\"><surname>P&#233;rez</surname><given-names>C.</given-names></name><name name-style=\"western\"><surname>Martinez-Gonzalez</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Paglione</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Di Schiavi</surname><given-names>E.</given-names></name><etal/></person-group><article-title>Novel Curcumin-Diethyl Fumarate Hybrid as a Dualistic GSK-3&#946; Inhibitor/Nrf2 Inducer for the Treatment of Parkinson&#8217;s Disease</article-title><source>ACS Chem. Neurosci.</source><year>2020</year><volume>11</volume><fpage>2728</fpage><lpage>2740</lpage><pub-id pub-id-type=\"doi\">10.1021/acschemneuro.0c00363</pub-id><pub-id pub-id-type=\"pmid\">32663009</pub-id><pub-id pub-id-type=\"pmcid\">PMC8009478</pub-id></element-citation></ref><ref id=\"B75-pharmaceuticals-16-00014\"><label>75.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Huang</surname><given-names>Y.</given-names></name><name name-style=\"western\"><surname>Lin</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Qiao</surname><given-names>X.</given-names></name><name name-style=\"western\"><surname>Ji</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>K.D.</given-names></name><name name-style=\"western\"><surname>Yeh</surname><given-names>C.T.</given-names></name><name name-style=\"western\"><surname>Tzeng</surname><given-names>Y.M.</given-names></name><name name-style=\"western\"><surname>Guo</surname><given-names>D.</given-names></name><name name-style=\"western\"><surname>Ye</surname><given-names>M.</given-names></name></person-group><article-title>Antcamphins A-L, ergostanoids from Antrodia camphorate</article-title><source>J. Nat. Prod.</source><year>2014</year><volume>77</volume><fpage>118</fpage><lpage>124</lpage><pub-id pub-id-type=\"doi\">10.1021/np400741s</pub-id><pub-id pub-id-type=\"pmid\">24387703</pub-id></element-citation></ref><ref id=\"B76-pharmaceuticals-16-00014\"><label>76.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Yang</surname><given-names>G.J.</given-names></name><name name-style=\"western\"><surname>Ko</surname><given-names>C.N.</given-names></name><name name-style=\"western\"><surname>Zhong</surname><given-names>H.J.</given-names></name><name name-style=\"western\"><surname>Leung</surname><given-names>C.H.</given-names></name><name name-style=\"western\"><surname>Ma</surname><given-names>D.L.</given-names></name></person-group><article-title>Structure-Based Discovery of a Selective KDM5A Inhibitor that Exhibits Anti-Cancer Activity via Inducing Cell Cycle Arrest and Senescence in Breast Cancer Cell Lines</article-title><source>Cancers</source><year>2019</year><volume>11</volume><fpage>92</fpage><lpage>107</lpage><pub-id pub-id-type=\"doi\">10.3390/cancers11010092</pub-id><pub-id pub-id-type=\"pmid\">30650517</pub-id><pub-id pub-id-type=\"pmcid\">PMC6360022</pub-id></element-citation></ref><ref id=\"B77-pharmaceuticals-16-00014\"><label>77.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Taylor</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Wakem</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Dijkman</surname><given-names>G.</given-names></name><name name-style=\"western\"><surname>Alsarraj</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Nguyen</surname><given-names>M.</given-names></name></person-group><article-title>A practical approach to RT-qPCR-Publishing data that conform to the MIQE guidelines</article-title><source>Methods</source><year>2010</year><volume>50</volume><fpage>S1</fpage><lpage>S5</lpage><pub-id pub-id-type=\"doi\">10.1016/j.ymeth.2010.01.005</pub-id><pub-id pub-id-type=\"pmid\">20215014</pub-id></element-citation></ref><ref id=\"B78-pharmaceuticals-16-00014\"><label>78.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Dai</surname><given-names>W.</given-names></name><name name-style=\"western\"><surname>Zhao</surname><given-names>F.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>J.</given-names></name><name name-style=\"western\"><surname>Liu</surname><given-names>H.</given-names></name></person-group><article-title>ASCT2 Is Involved in SARS-Mediated &#946;-Casein Synthesis of Bovine Mammary Epithelial Cells with Methionine Supply</article-title><source>J. Agric. Food Chem.</source><year>2019</year><volume>68</volume><fpage>13038</fpage><lpage>13045</lpage><pub-id pub-id-type=\"doi\">10.1021/acs.jafc.9b03833</pub-id><pub-id pub-id-type=\"pmid\">31597423</pub-id></element-citation></ref><ref id=\"B79-pharmaceuticals-16-00014\"><label>79.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Devling</surname><given-names>T.W.</given-names></name><name name-style=\"western\"><surname>Lindsay</surname><given-names>C.D.</given-names></name><name name-style=\"western\"><surname>McLellan</surname><given-names>L.I.</given-names></name><name name-style=\"western\"><surname>McMahon</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Hayes</surname><given-names>J.D.</given-names></name></person-group><article-title>Utility of siRNA against Keap1 as a strategy to stimulate a cancer chemopreventive phenotype</article-title><source>Proc. Natl. Acad. Sci. USA</source><year>2005</year><volume>102</volume><fpage>7280</fpage><lpage>7285</lpage><pub-id pub-id-type=\"doi\">10.1073/pnas.0501475102</pub-id><pub-id pub-id-type=\"pmid\">15883370</pub-id><pub-id pub-id-type=\"pmcid\">PMC1091750</pub-id></element-citation></ref><ref id=\"B80-pharmaceuticals-16-00014\"><label>80.</label><element-citation publication-type=\"journal\"><person-group person-group-type=\"author\"><name name-style=\"western\"><surname>Ciotti</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Iuliano</surname><given-names>L.</given-names></name><name name-style=\"western\"><surname>Cefal&#249;</surname><given-names>S.</given-names></name><name name-style=\"western\"><surname>Comelli</surname><given-names>M.</given-names></name><name name-style=\"western\"><surname>Mavelli</surname><given-names>I.</given-names></name><name name-style=\"western\"><surname>Di Giorgio</surname><given-names>E.</given-names></name><name name-style=\"western\"><surname>Brancolini</surname><given-names>C.</given-names></name></person-group><article-title>GSK3&#946; is a key regulator of the ROS-dependent necrotic death induced by the quinone DMNQ</article-title><source>Cell Death Dis.</source><year>2020</year><volume>11</volume><fpage>2</fpage><pub-id pub-id-type=\"doi\">10.1038/s41419-019-2202-0</pub-id><pub-id pub-id-type=\"pmid\">31919413</pub-id><pub-id pub-id-type=\"pmcid\">PMC6952365</pub-id></element-citation></ref></ref-list></back><floats-group><fig position=\"float\" id=\"pharmaceuticals-16-00014-f001\" orientation=\"portrait\"><label>Figure 1</label><caption><p>Screening of compounds <bold>1</bold>&#8211;<bold>13</bold> and identification of DEA (<bold>1</bold>) as a potent Keap1&#8211;Nrf2 inhibitor in vitro with low toxicity. (<bold>A</bold>) The binding activity of Keap1&#8211;Nrf2 was measured via an FP assay with the compounds added at 50 &#181;M. (<bold>B</bold>) Dose-response effect of compound DEA (<bold>1</bold>) on Keap1&#8211;Nrf2 binding. (<bold>C</bold>) Compound DEA (<bold>1</bold>) inhibited GSK3&#946; activity in vitro. (<bold>D</bold>) Dose-response effect of compound DEA (<bold>1</bold>) on GSK3&#946; activity. (<bold>E</bold>) Compound DEA (<bold>1</bold>) enhances ARE activity in LO2 cells determined by the dual luciferase assay. (<bold>F</bold>&#8211;<bold>H</bold>) The cytotoxicity of DEA (<bold>1</bold>) and ML334 on LO2, HEK 293T, and HepG2 cell lines for 48 h. Cells were treated with 0&#8211;100 &#181;M of DEA (<bold>1</bold>) and ML334 for 48 h and cytotoxicity was detected using the MTT assay. Data are represented as mean &#177; SD. * <italic toggle=\"yes\">p</italic> &lt; 0.05, ** <italic toggle=\"yes\">p</italic> &lt; 0.01, *** <italic toggle=\"yes\">p</italic> &lt; 0.001 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. DEA: Dehydroeburicoic acid; FP: fluorescence polarization; DMSO: Dimethyl sulfoxide; GSK3&#946;: Glycogen synthase kinase 3&#946;; IgG: Immunoglobulin G; Keap1: Kelch-like erythroid cell&#8211;derived protein with CNC homology&#8211;associated protein 1; Nrf2: Nuclear respiratory factor-2; MTT: (3-(4,5-dimethylthiazol-2-yl)-2,5-diphenyltetrazolium bromide) tetrazolium reduction.</p></caption><graphic xmlns:xlink=\"http://www.w3.org/1999/xlink\" position=\"float\" orientation=\"portrait\" xlink:href=\"pharmaceuticals-16-00014-g001.jpg\"/></fig><fig position=\"float\" id=\"pharmaceuticals-16-00014-f002\" orientation=\"portrait\"><label>Figure 2</label><caption><p>Compound DEA (<bold>1</bold>) releases Nrf2 by binding with Keap1 and GSK3&#946;. (<bold>A</bold>) LO2 cells were treated with 10 &#181;M DEA (<bold>1</bold>) and positive control ML334 for 8 h. Interactions between Keap1 and Nrf2 in LO2 cells were examined by WB and Co&#8211;IP. (<bold>B</bold>) Quantification analysis of Keap1 in WB. Error bars represent the standard deviations of the results from three independent experiments. (<bold>C</bold>) LO2 lysates were treated with 10 &#181;M DEA (<bold>1</bold>) or DMSO and incubated at room temperature for 30 minutes. Then, CETSA was performed to assess Keap1, Nrf2, GSK3&#946;, and &#946;&#8211;actin thermal stability. (<bold>D</bold>) Quantification analysis of Keap1, Nrf2, GSK3&#946;, and &#946;&#8211;actin in WB. Data are represented as mean &#177; SD. * <italic toggle=\"yes\">p</italic> &lt; 0.05, ** <italic toggle=\"yes\">p</italic> &lt; 0.01 vs. DMSO group (Student&#8217;s <italic toggle=\"yes\">t</italic>-test). Error bars represent the standard deviations of the results from three independent experiments. Anti&#8211;Nrf2: anti&#8211;Nrf2 antibody; WB: Western blot; Co&#8211;IP: co&#8211;immunoprecipitation.</p></caption><graphic xmlns:xlink=\"http://www.w3.org/1999/xlink\" position=\"float\" orientation=\"portrait\" xlink:href=\"pharmaceuticals-16-00014-g002.jpg\"/></fig><fig position=\"float\" id=\"pharmaceuticals-16-00014-f003\" orientation=\"portrait\"><label>Figure 3</label><caption><p>Compound DEA (<bold>1</bold>) regulates the antioxidant effects, at least in part, by targeting Keap1 and GSK3&#946;. (<bold>A</bold>) Keap1 knockdown decreased the antioxidant effect of DEA (<bold>1</bold>) in LO2 cells. Keap1, Nrf2, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">NQO1</italic> were blotted to control for total protein levels. (<bold>B</bold>) Quantification analysis of Keap1, Nrf2, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">NQO1</italic> levels in Keap1 knockdown cells. Data are represented as mean &#177; SD. <sup>#</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.05, <sup>##</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.01 vs. control group, * <italic toggle=\"yes\">p</italic> &lt; 0.05 vs. DMSO. Error bars represent the standard deviations of the results from three independent experiments. (<bold>C</bold>) GSK3&#946; knockdown decreased the antioxidant effect of DEA (<bold>1</bold>) in LO2 cells. GSK3&#946;, Nrf2, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">NQO1</italic> were blotted to control for total protein levels. (<bold>D</bold>) Quantification analysis of GSK3&#946;, Nrf2, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">NQO1</italic> levels in GSK3&#946; knockdown cells. Data are represented as mean &#177; SD. <sup>#</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.05, <sup>##</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.01 vs. control group, * <italic toggle=\"yes\">p</italic> &lt; 0.05, ** <italic toggle=\"yes\">p</italic> &lt; 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. (<bold>E</bold>) Keap1 and GSK3&#946; knockdown decreased the antioxidant effect of DEA (<bold>1</bold>) in LO2 cells. GSK3&#946;, Nrf2, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">NQO1</italic> were blotted to control for total protein levels. (<bold>F</bold>) Quantification analysis of GSK3&#946;, Keap1, Nrf2, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, and <italic toggle=\"yes\">NQO1</italic> levels in Keap1 and GSK3&#946; knockdown cells. Data are represented as mean &#177; SD. <sup>#</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.05, <sup>##</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.01 vs. control group, ** <italic toggle=\"yes\">p</italic> &lt; 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA Keap1: SiKeap1; SiRNA GSK3&#946;: SiGSK3&#946;; SiGSK3&#946; + SiKeap1: SiRNA GSK3&#946; + SiRNA Keap1.</p></caption><graphic xmlns:xlink=\"http://www.w3.org/1999/xlink\" position=\"float\" orientation=\"portrait\" xlink:href=\"pharmaceuticals-16-00014-g003.jpg\"/></fig><fig position=\"float\" id=\"pharmaceuticals-16-00014-f004\" orientation=\"portrait\"><label>Figure 4</label><caption><p>Compound DEA (<bold>1</bold>) increases Nrf2 translocation into the nucleus. (<bold>A</bold>) LO2 cells were treated with 10 or 30 &#181;M compound DEA (<bold>1</bold>) for 8 h and then analyzed by WB to detect the level of Nrf2 in the nucleus and cytoplasm. &#946;&#8211;actin and lamin B were used as internal controls. (<bold>B</bold>) GSK3&#946; knockdown increased nuclear Nrf2 level. Keap1 and Nrf2 were blotted to control for the total protein levels. The data represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA GSK3&#946;: SiGSK3&#946;.</p></caption><graphic xmlns:xlink=\"http://www.w3.org/1999/xlink\" position=\"float\" orientation=\"portrait\" xlink:href=\"pharmaceuticals-16-00014-g004.jpg\"/></fig><fig position=\"float\" id=\"pharmaceuticals-16-00014-f005\" orientation=\"portrait\"><label>Figure 5</label><caption><p>Compound DEA (<bold>1</bold>) induces antioxidant factor expression in the ALD cell model. (<bold>A</bold>) Effects of 10 &#181;M, 30 &#181;M DEA (<bold>1</bold>), and 30 &#181;M ML334 on <italic toggle=\"yes\">Nrf1</italic>, <italic toggle=\"yes\">NQO1</italic>, <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, Keap1, and Nrf2 protein levels in LO2 cells after 8 h treatment. (<bold>B</bold>&#8211;<bold>D</bold>) Quantification analysis of <italic toggle=\"yes\">Nrf1</italic> (<bold>B</bold>), <italic toggle=\"yes\">NQO1</italic> (<bold>C</bold>), and <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic> (<bold>D</bold>) in WB. (<bold>E</bold>&#8211;<bold>I</bold>) Effects of DEA (<bold>1</bold>) and ML334 on (<bold>E</bold>) <italic toggle=\"yes\">SOD2</italic>, (<bold>F</bold>) <italic toggle=\"yes\">NQO1</italic>, (<bold>G</bold>) <italic toggle=\"yes\">HO</italic>&#8211;<italic toggle=\"yes\">1</italic>, (<bold>H</bold>) <italic toggle=\"yes\">Nrf1</italic>, (<bold>I</bold>) <italic toggle=\"yes\">PGC</italic>&#8211;<italic toggle=\"yes\">1&#945;</italic> mRNA levels in 0.3% EtOH&#8211;induced LO2 cells. Data are represented as mean &#177; SD. * <italic toggle=\"yes\">p</italic> &lt; 0.05, *** <italic toggle=\"yes\">p</italic> &lt; 0.001 vs. control group. The data represent the standard deviations of the results from three independent experiments. EtOH: Ethyl alcohol. <italic toggle=\"yes\">Nrf1</italic>: Nuclear respiratory factor&#8211;1; <italic toggle=\"yes\">NQO1</italic>: NAD(P)H dehydrogenase [quinone] 1; <italic toggle=\"yes\">PGC</italic>&#8211;<italic toggle=\"yes\">1&#945;</italic>: Peroxisome proliferator&#8211;activated receptor &#947; coactivator 1&#945;; SOD 2: Superoxide dismutase 2.</p></caption><graphic xmlns:xlink=\"http://www.w3.org/1999/xlink\" position=\"float\" orientation=\"portrait\" xlink:href=\"pharmaceuticals-16-00014-g005.jpg\"/></fig><fig position=\"float\" id=\"pharmaceuticals-16-00014-f006\" orientation=\"portrait\"><label>Figure 6</label><caption><p>Compound DEA (<bold>1</bold>) protects mitochondrial dysfunction from 0.3% EtOH&#8211;induced injury. (<bold>A</bold>) 0.3% EtOH&#8211;induced LO2 cells were treated with 30 &#181;M ML334, 10 or 30 &#181;M compound DEA (<bold>1</bold>) for 8 h. Intracellular ROS level were detected by DCFH&#8211;DA probe and measured using the confocal laser scanning microscopy. (<bold>B</bold>) EtOH&#8211;induced LO2 cells were treated with 10 &#181;M ML334, 10 or 30 &#181;M compound DEA (<bold>1</bold>), DMSO for 8 h, and MPTP function was detected by fluorescence spectroscopy. (<bold>C</bold>) Quantification analysis of fluorescence in MPTP assay. (<bold>D</bold>) ATP level were detected using an ATP assay kit in 0.3% EtOH&#8211;induced LO2 cells. (<bold>E</bold>) Antioxidant activity of DEA (<bold>1</bold>) was detected in 0.3% EtOH&#8211;induced LO2 cells. Data are represented as mean &#177; SD. <sup>#</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.05, <sup>##</sup>&#160;<italic toggle=\"yes\">p</italic> &lt; 0.01, vs. control group. The data represent the standard deviations of the results from three independent experiments. ATP: Adenosine triphosphate; DCFH&#8211;DA: Dichlorodihydrofluorescein diacetate; MPTP: Mitochondrial Permeability Transition Pore Assay; ROS: Reactive oxygen species.</p></caption><graphic xmlns:xlink=\"http://www.w3.org/1999/xlink\" position=\"float\" orientation=\"portrait\" xlink:href=\"pharmaceuticals-16-00014-g006.jpg\"/></fig><fig position=\"float\" id=\"pharmaceuticals-16-00014-f007\" orientation=\"portrait\"><label>Figure 7</label><caption><p>Hepatoprotective mechanism of the dual inhibitor DEA (<bold>1</bold>) against EtOH&#8211;induced cell injury.</p></caption><graphic xmlns:xlink=\"http://www.w3.org/1999/xlink\" position=\"float\" orientation=\"portrait\" xlink:href=\"pharmaceuticals-16-00014-g007.jpg\"/></fig></floats-group></article></pmc-articleset>\n"
     ]
    }
   ],
   "source": [
    "print(fulltext_xml)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e3fa3f9-e4e3-40c0-a3fa-189e73ce47c1",
   "metadata": {},
   "source": [
    "# Parsiranjea XML format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "50e1b62c-e6e1-4bdf-901c-035f7da4fac2",
   "metadata": {},
   "outputs": [],
   "source": [
    "XLINK = \"{http://www.w3.org/1999/xlink}\"\n",
    "WS_RE = re.compile(r\"\\s+\")\n",
    "\n",
    "def norm_ws(s: str) -> str:\n",
    "    return WS_RE.sub(\" \", (s or \"\")).strip()\n",
    "\n",
    "# skip links/tables/formulas; DO NOT skip <fig> here, we handle it explicitly\n",
    "SKIP_TAGS = {\"ext-link\", \"uri\", \"inline-formula\", \"disp-formula\", \"table-wrap\"}\n",
    "INLINE_KEEP_TAGS = {\"italic\", \"bold\", \"sup\", \"sub\"}\n",
    "\n",
    "def _fig_token(fig_el: ET.Element) -> str:\n",
    "    \"\"\"Build a stable placeholder for a <fig> element.\"\"\"\n",
    "    fid = fig_el.get(\"id\")\n",
    "    if not fid:\n",
    "        g = fig_el.find(\".//graphic\")\n",
    "        if g is not None:\n",
    "            href = g.get(f\"{XLINK}href\") or \"\"\n",
    "            fid = (href.rsplit(\"/\", 1)[-1] or href).rsplit(\".\", 1)[0]\n",
    "    if not fid:\n",
    "        lbl = (fig_el.findtext(\"label\") or \"\").strip() or \"fig\"\n",
    "        fid = lbl\n",
    "    return f\"[[FIG::{fid}]]\"\n",
    "\n",
    "def extract_text_tokenizing_inline_figs(node: ET.Element) -> str:\n",
    "    parts = []\n",
    "    def walk(el: ET.Element):\n",
    "        tag = el.tag\n",
    "\n",
    "        if tag in {\"ext-link\", \"uri\", \"inline-formula\", \"disp-formula\", \"table-wrap\"}:\n",
    "            if el.tail:\n",
    "                parts.append(el.tail)\n",
    "            return\n",
    "\n",
    "        if tag == \"fig\":\n",
    "            parts.append(_fig_token(el))          # single token where the figure sits\n",
    "            if el.tail:\n",
    "                parts.append(el.tail)\n",
    "            return\n",
    "\n",
    "        if tag == \"xref\" and (el.get(\"ref-type\") or \"\").lower() == \"fig\":\n",
    "            # do NOT tokenize mentions like \"Figure 2A\"; just keep flow\n",
    "            if el.tail:\n",
    "                parts.append(el.tail)\n",
    "            return\n",
    "\n",
    "        if el.text:\n",
    "            parts.append(el.text)\n",
    "        for child in list(el):\n",
    "            walk(child)\n",
    "        if el.tail:\n",
    "            parts.append(el.tail)\n",
    "\n",
    "    walk(node)\n",
    "    return norm_ws(\"\".join(parts))\n",
    "\n",
    "def first(el: Optional[ET.Element], path: str) -> Optional[ET.Element]:\n",
    "    return el.find(path) if el is not None else None\n",
    "\n",
    "def section_title(sec: ET.Element) -> str:\n",
    "    t = first(sec, \"title\")\n",
    "    return extract_text_tokenizing_inline_figs(t) if t is not None else \"\"\n",
    "\n",
    "def collect_sections(sec: ET.Element, parent_label: str = \"\") -> List[Dict]:\n",
    "    chunks = []\n",
    "    sid = sec.get(\"id\")\n",
    "    title_el = sec.find(\"title\")\n",
    "    title = extract_text_tokenizing_inline_figs(title_el) if title_el is not None else \"\"\n",
    "    label = title or parent_label\n",
    "\n",
    "    body_parts = []\n",
    "    for child in list(sec):\n",
    "        if child.tag == \"title\":\n",
    "            if child.tail:\n",
    "                body_parts.append(child.tail)\n",
    "            continue\n",
    "        if child.tag == \"sec\":\n",
    "            continue\n",
    "        body_parts.append(extract_text_tokenizing_inline_figs(child))\n",
    "\n",
    "    body_text = norm_ws(\" \".join(p for p in body_parts if p))\n",
    "    if body_text:\n",
    "        chunks.append({\"type\": \"body\", \"id\": sid, \"title\": title, \"text\": body_text})\n",
    "\n",
    "    for sub in sec.findall(\"sec\"):\n",
    "        chunks.extend(collect_sections(sub, parent_label=label))\n",
    "    return chunks\n",
    "def _abstract_text(abs_el: ET.Element) -> str:\n",
    "    \"\"\"Flatten a (possibly structured) <abstract> into plain text using your tokenizer.\"\"\"\n",
    "    parts = []\n",
    "\n",
    "    def walk_abstract(el: ET.Element):\n",
    "        # if structured abstract with <sec>, keep each sec title inline then its text\n",
    "        if el.tag == \"sec\":\n",
    "            t = el.find(\"title\")\n",
    "            if t is not None:\n",
    "                parts.append(extract_text_tokenizing_inline_figs(t) + \": \")\n",
    "            # body of this sec\n",
    "            for child in list(el):\n",
    "                if child.tag == \"title\":\n",
    "                    continue\n",
    "                walk_abstract(child)\n",
    "            return\n",
    "\n",
    "        # paragraphs or mixed content\n",
    "        if el.text:\n",
    "            parts.append(el.text)\n",
    "        for child in list(el):\n",
    "            walk_abstract(child)\n",
    "        if el.tail:\n",
    "            parts.append(el.tail)\n",
    "\n",
    "    walk_abstract(abs_el)\n",
    "    return norm_ws(\"\".join(parts))\n",
    "\n",
    "def extract_abstract_chunk(article: ET.Element) -> Optional[Dict]:\n",
    "    \"\"\"Return a single combined abstract chunk or None.\"\"\"\n",
    "    am = article.find(\"front/article-meta\")\n",
    "    if am is None:\n",
    "        return None\n",
    "\n",
    "    # collect candidate abstracts (exclude graphical/visual/teaser/toc)\n",
    "    def is_narrative(a: ET.Element) -> bool:\n",
    "        at = (a.get(\"abstract-type\") or \"\").lower()\n",
    "        return not any(k in at for k in (\"graphical\", \"visual\", \"teaser\", \"toc\"))\n",
    "\n",
    "    abstracts = [a for a in am.findall(\"abstract\") if is_narrative(a)]\n",
    "    # optional: include translated abstracts the same way\n",
    "    trans_abstracts = [a for a in am.findall(\"trans-abstract\") if is_narrative(a)]\n",
    "\n",
    "    if not abstracts and not trans_abstracts:\n",
    "        return None\n",
    "\n",
    "    texts = []\n",
    "    for a in abstracts + trans_abstracts:\n",
    "        # prefer an explicit title inside the abstract if present\n",
    "        t_el = a.find(\"title\")\n",
    "        if t_el is not None:\n",
    "            title_text = extract_text_tokenizing_inline_figs(t_el)\n",
    "            if title_text:\n",
    "                texts.append(title_text + \". \")\n",
    "        texts.append(_abstract_text(a))\n",
    "\n",
    "    all_text = norm_ws(\" \".join(t for t in texts if t))\n",
    "    if not all_text:\n",
    "        return None\n",
    "\n",
    "    # try to keep a stable id and title\n",
    "    abs_id = (abstracts[0].get(\"id\") if abstracts else None) or \"abstract\"\n",
    "    abs_title = \"Abstract\"\n",
    "    return {\"type\": \"abstract\", \"id\": abs_id, \"title\": abs_title, \"text\": all_text}\n",
    "\n",
    "def jats_to_chunks(xml_string: str) -> List[Dict]:\n",
    "    root = ET.fromstring(xml_string)\n",
    "    article = root if root.tag == \"article\" else (root.find(\"article\") or None)\n",
    "    if article is None:\n",
    "        raise ValueError(\"No <article> element found.\")\n",
    "\n",
    "    chunks: List[Dict] = []\n",
    "\n",
    "    # --- front matter (if you have any, keep) ---\n",
    "\n",
    "    # --- abstract (NEW) ---\n",
    "    abs_chunk = extract_abstract_chunk(article)\n",
    "    if abs_chunk is not None:\n",
    "        chunks.append(abs_chunk)\n",
    "\n",
    "    # --- body sections ---\n",
    "    body = article.find(\"body\")\n",
    "    if body is not None:\n",
    "        for sec in body.findall(\"sec\"):\n",
    "            chunks.extend(collect_sections(sec))\n",
    "\n",
    "    # ---------- stand-alone figure placeholders (keep your code) ----------\n",
    "    inline_fig_ids = set()\n",
    "    if body is not None:\n",
    "        for fig in body.findall(\".//sec//fig\"):\n",
    "            fid = fig.get(\"id\")\n",
    "            if fid:\n",
    "                inline_fig_ids.add(fid)\n",
    "\n",
    "    def _add_fig_chunk(fig_el: ET.Element):\n",
    "        tok = _fig_token(fig_el)\n",
    "        chunks.append({\n",
    "            \"type\": \"body\",\n",
    "            \"id\": fig_el.get(\"id\"),\n",
    "            \"title\": (fig_el.findtext(\"label\") or \"\").strip(),\n",
    "            \"text\": tok\n",
    "        })\n",
    "\n",
    "    if body is not None:\n",
    "        for fig in body.findall(\"fig\"):\n",
    "            fid = fig.get(\"id\")\n",
    "            if not fid or fid not in inline_fig_ids:\n",
    "                _add_fig_chunk(fig)\n",
    "\n",
    "    floats = article.find(\"floats-group\")\n",
    "    if floats is not None:\n",
    "        for fig in floats.findall(\"fig\"):\n",
    "            fid = fig.get(\"id\")\n",
    "            if not fid or fid not in inline_fig_ids:\n",
    "                _add_fig_chunk(fig)\n",
    "\n",
    "    # --- back matter (keep your code) ---\n",
    "\n",
    "    # normalize\n",
    "    out = []\n",
    "    for c in chunks:\n",
    "        t = norm_ws(c.get(\"text\", \"\"))\n",
    "        if t:\n",
    "            cc = dict(c); cc[\"text\"] = t\n",
    "            out.append(cc)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b87c91ae-ac6a-4aca-b0d6-d78f576ba2d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] abstract :: Abstract\n",
      "Alcoholic liver disease (ALD) is a complicated disease which can lead to hepatocellular carcinoma; however, there is a lack of satisfactory therapeutics. Dehydroeburicoic acid (DEA) (1), a triterpenoid isolated from Antrodia cinnamomea, has been reported to act against ALD, but its mechanisms of action are still not clear. In this study, we report for the first time the use of DEA (1) as a dual inhibitor of the Keap1Nrf2 proteinprotein interaction (PPI) and GSK3 in an in vitro ALD cell model. DEA (1) engages Keap1 to disrupt the Keap1Nrf2 PPI and inhibits GSK3 to restore Nrf2 activity in a Keap1-independent fashion. DEA (1) promotes Nrf2 nuclear translocation to activate downstream antioxidant genes. Importantly, DEA (1) restores the mitochondrial dysfunction induced by ethanol and generates antioxidant activity in the ALD cell model with minimal toxicity. We anticipate that DEA (1) could be a potential scaffold for the further development of clinical agents for treating ALD.\n",
      "--------------------------------------------------------------------------------\n",
      "[2] body :: 1. Introduction\n",
      "Excessive alcohol consumption leads to oxidative stress in the liver as a result of alcohol metabolism, resulting in inflammatory damage and the injury of liver cells [1]. Almost 20% of alcoholics are eventually diagnosed with alcoholic liver disease (ALD) [2]. The most common therapeutic options for ALD can be divided into opioid receptor antagonists, supplements to modulate liver metabolism, and therapeutics that regulate alcohol metabolism [3]. However, these treatments have been associated with side effects including dizziness, drug dependence, dermatitis, vomiting, and leukopenia [4]. Thus, there are currently no highly satisfactory therapeutic options for ALD, and hence there is an unmet clinical need to develop more effective and safer drugs for patients with ALD. Increasing evidence indicates that alcohol can damage mitochondria and destroy cellular homeostasis in the liver [5,6,7]. One mechanism that cells use to defend against alcoholinduced injury is through the activation of nuclear respiratory factor-2 (Nrf-2). Nrf2 orchestrates a complex, self-protective antioxidant response involving numerous signaling axes [8]. Upregulating Nrf2 has been envisioned as an effective tool for combating alcohol-induced acute liver injury [9,10]. Normally, the level of Nrf2 in the cytosol is low due to ubiquitination and proteasomal degradation regulated by a Kelch-like ECH-associated protein (Keap1) [11,12]. However, the presence of oxidants can oxidize cysteine residues in Keap1 into disulfides or conjugate them to electrophiles, resulting in the nuclear translocation of Nrf2 and the activation of antioxidant genes [13], including heme oxygenase-1 (HO1), NAD(P)H dehydrogenase [quinone] 1 (NQO1), and the mitochondrial superoxide dismutase 2 (SOD2) [14,15]. In addition, Nrf2 activation increases the expression levels of nuclear respiratory factor-1 (Nrf-1) and peroxisome proliferator-activated receptor  coactivator 1 (PGC1), which in turn modulates the expression of mitochondrial respiratory subunits and translational components [16]. The genes activated by the antioxidant responsive element (ARE) protect cells against any damage caused by reactive oxygen species (ROS) [17]. In addition to Keap1, glycogen synthase kinase 3 (GSK3) is another negative regulator of Nrf2 that acts to downregulate the antioxidant stress response [18,19]. GSK3 promotes the nuclear exclusion and degradation of Nrf2 in a Keap1-independent fashion in stressed or injured cells with active Nrf2, thereby reducing the degree of protection conferred by Nrf2 activity [20,21]. Therefore, increasing the Nrf2 antioxidant activity via targeting Keap1 or GSK3 could be a potential approach for treating ALD [22]. Most pharmacological approaches towards Nrf2 activation are focused on the inhibition of Keap1 [23,24]. Several Keap1 inhibitors have been reported in the last decade, and the vast majority of these covalently modify cysteine residues in Keap1 [25]. Covalently targeting cysteines in Keap1 may lack selectivity due to the presence of other reactive cysteine residues in the cell, resulting in adverse side effects [26,27]. However, because cysteine residues are abundant in cells, the safety and specificity of covalent drugs are a concern [28]. Recently, proteinprotein interaction (PPI) inhibitors of Keap1 and Nrf2, developed based on the X-ray structure of Keap1, have emerged as a new class of Nrf2 activators [29,30,31]. Our group has previously reported a cyclometalated iridium(III) metal complex as a Keap1Nrf2 PPI inhibitor, which is a promising therapeutic agent for acetaminopheninduced acute liver injury [29,30]. However, no Keap1Nrf2 inhibitor has yet entered the clinic for the treatment of human diseases [32,33]. Meanwhile, although a few GSK3 inhibitors have been reported for liver injury treatment, no GSK3 inhibitor to date has made it to the market [34]. Antrodia cinnamomea, a basidiomycete that is endemic to Taiwan, is an edible fungus that is a component of many traditional herbal medicines [35,36]. Either crude extracts or molecules isolated from A. cinnamomea have shown diverse biological properties, including antioxidant, anti-tumour, and anti-inflammatory activities [37,38]. Recently, research has shown that dehydroeburicoic acid (DEA) (1) from A. cinnamomea could inhibit alcoholic fatty liver disease (AFLD) by upregulating aldehyde dehydrogenase 2 family member (ALDH2) activity [39,40]. Additionally, DEA (1) also exhibited protective effects against non-alcoholic fatty liver disease (NAFLD) through activating ALDH2 and accelerating the elimination of ROS and harmful aldehydes [40]. Therefore, DEA (1) has the potential ability to treat ALD. However, its mechanisms of hepatoprotection in ALD are still not clear. Besides DEA (1), other triterpenoids have also been studied as GSK3 inhibitors and activators of Nrf2 because of their anti-inflammatory and antioxidant activities [41,42,43,44,45]. Research has shown that singlemolecule drugs with two different biological activities may avoid some side effects and improve efficacy [46]. In this study, we discovered a potent dual Keap1Nrf2 PPI and GSK3 inhibitor, DEA (1), and investigated its underlying mechanisms of action. Through in vitro fluorescence polarization (FP) screening, we identified DEA (1) as the most potent candidate for disrupting the Keap1Nrf2 PPI (EC50 = 14.1 M) from thirteen analogues isolated from A. cinnamomea. Moreover, DEA (1) inhibited GSK3 kinase activity with an EC50 value of 8.0 M. In cells, low-toxicity compound DEA (1) engaged Keap1 and GSK3 to promote Nrf2 accumulation in the nucleus, leading to the upregulation of ARE transcriptional activity and increasing the expression of downstream antioxidant factors with more potent than the reported Keap1Nrf2 PPI inhibitor ML334 in ALD model cells.\n",
      "--------------------------------------------------------------------------------\n",
      "[3] body :: 2.1. DEA (1) Inhibits the Keap1Nrf2 PPI and GSK3 Activity\n",
      "The fluorescence polarization assay is a commonly used high-throughput assay to detect PPI inhibitors in solution. We used labeled Nrf2 peptides, bearing the highaffinity ETGE motif, which is recognized by Keap1, as fluorescent tracers to monitor for inhibition of the Keap1Nrf2 interaction. ML334 is the first noncovalent small molecule inhibitor of Keap1Nrf2 interaction reported and is distinct from other Nrf2 inducers [47]. In this paper, ML334 was used as a positive control. Compounds 113 from A. cinnamomea and ML334 were screened against the Keap1Nrf2 interaction by FP (A). The preliminary screening results showed that DEA (1) was the most potent compound, which disrupted the Keap1Nrf2 interaction by 62% inhibition at 50 M (cf. ML334: 60% inhibition). The NMR spectrum of DEA (1) is presented in Supplementary Figure S1. The potency of DEA (1) was further investigated in a dose-response assay, revealing an EC50 value of 14.1  0.1 M against Keap1Nrf2 PPI (B). We also explored the inhibitory effect of DEA (1) against GSK3, which negatively regulates Nrf2 in a fashion independent of Keap1. 10 M of DEA (1) reduced GSK3 activity by 58.9%, making it more potent than ML334 at the same concentration (11.3% inhibition) (C). Moreover, a dose-response assay revealed an EC50 value of 8.0  0.7 M against GSK3 activity, indicating that DEA (1) is an effective GSK3 inhibitor (D). In the nucleus, Nrf2 binds to the ARE to activate antioxidant gene transcription [14]. Thus, we tested the transcriptional activity of ARE after treatment of LO2 cells with DEA (1) and ML334 (10 M) using the dual luciferase assay kit (E). The activity of ARE was significantly increased by 54% in the treatment of DEA (1) compared to 27% in the ML334 group. The results showed that DEA (1) notably activates the transcriptional activity of ARE compared to ML334 in cellulo.\n",
      "--------------------------------------------------------------------------------\n",
      "[4] body :: 2.2. DEA (1) Exhibits Low Cytotoxicity In Cellulo\n",
      "The cytotoxicity of compound DEA (1) was detected in both human normal cell lines LO2 and HEK 293T, and the hepatocellular carcinoma cell line HepG2. After treatment with different concentrations of DEA (1), no significant cytotoxicity was observed (FH). These data indicate that DEA (1) could be potentially safe for treating ALD in vivo.\n",
      "--------------------------------------------------------------------------------\n",
      "[5] body :: 2.3. DEA (1) as a Dual Inhibitor of Keap1 and GSK3\n",
      "The ability of DEA (1) to disrupt the Keap1Nrf2 binding in cellulo was monitored through a CoIP experiment using human normal liver cell line LO2 cells (A). After treatment with DEA (1) (10 M) or ML334 (10 M) for 8 h, there was a 45% reduction of Keap1 co-precipitated with Nrf2 for DEA (1) as compared to 12% inhibition for ML334, indicating that DEA (1) could more effectively inhibit the Keap1Nrf2 interaction compared to ML334 in living cells (B). We investigated the ability of DEA (1) to target Keap1, Nrf2, and GSK3 in the cellular environment using CETSA. LO2 cell lysates were incubated with 10 M of DEA (1) at room temperature for 30 min. Then, aliquots were heated individually at different temperatures and the protein in the soluble fraction was quantified by WB. Compound DEA (1) significantly stabilized Keap1 (Tm: ca. 4.5 C) and GSK3 (Tm: ca. 3.9 C), while having no observable effect on the thermal stability of Nrf2 and actin (C,D). This result suggests that DEA (1) can bind with Keap1 and GSK3, even within the complicated environment of cell lysates.\n",
      "--------------------------------------------------------------------------------\n",
      "[6] body :: 2.4. DEA (1) Targets Keap1 and GSK3 to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins\n",
      "To verify whether Keap1 or GSK3 are direct targets of DEA (1), knockdown assays were performed. NQO1 and HO1 are target antioxidant genes of Nrf2 that increase cytoprotection against oxidative stress [48] As expected, Keap1 knockdown in LO2 cells induced a noticeable rise of the expression levels of both Nrf2 and its gene targets, NQO1 and HO1 (A,B). However, treating Keap1 knockdown cells with DEA (1) led to no further rises in the NQO1 and HO1 level, in contrast to the control cells, where DEA (1) treatment produced noticeable increases of NQO1 and HO1. These results suggest that DEA (1) activates NQO1 and HO1 through targeting Keap1. Similarly, GSK3 knockdown induced Nrf2 and downstream antioxidant protein expression but without influencing Keap1 protein level (C,D). In addition, DEA (1) had a diminished effect on NQO1 and HO1 level in GSK3 knockdown cells as compared to the control cells. Finally, we knocked down GSK3 and Keap1 at the same time to verify the dual binding mechanism of DEA (1) on GSK3 and Keap1 (E,F). The results showed that treating the GSK3 and Keap1 knockdown cells with DEA (1) did not cause a further significant induction of NQO1 and HO1 expression. Evidence for the hypothesis that DEA (1) acts through targeting both Keap1 and GSK3 comes from considering the Nrf2 level. In either the Keap1 knockdown or GSK3 knockdown cells, DEA (1) treatment resulted in slight increases of Nrf2 expression relative to DMSO. However, in the double Keap1 and GSK3 knockdown cells, treatment with DEA (1) did not lead to increases of Nrf2 expression. These results suggest that DEA (1) treatment mimicked the effects of double knockdown in increasing Nrf2 expression. Therefore, DEA (1) exerts its antioxidant effects by targeting both Keap1 and GSK3. Inhibition of the Keap1Nrf2 interaction is known to increase Nrf2 accumulation in the nucleus [47]. Meanwhile, GSK3 is also a negative regulator of Nrf2 accumulation by acting independently of Keap1 [49]. Therefore, we evaluated the Nrf2 level in the cytoplasm and nucleus of LO2 cells via WB after the treatment of cells with DEA (1) (10 M) for 8 h. As shown in A, DEA (1) significantly increased Nrf2 translocation into the nucleus in a dose-dependent manner. GSK3 knockdown also phenocopied compound DEA (1) treatment at inducing Nrf2 nuclear accumulation without influencing Keap1 level (B). In summary, DEA (1) disrupts the Keap1Nrf2 PPI and inhibits GSK3 to increase the translocation of liberated Nrf2 into the nucleus, thereby presumably increasing ARE transcriptional activity.\n",
      "--------------------------------------------------------------------------------\n",
      "[7] body :: 2.5. DEA (1) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells\n",
      "Nuclear Nrf2 heterodimerizes with musculoaponeurotic fibrosarcoma (Maf) to recognize ARE and promote the transcription of antioxidant genes, including Nrf1, HO1, and NQO1 [14]. Therefore, we evaluated the ability of DEA (1) to increase Nrf1, NQO1, and HO1 levels in ALD model cells, which were prepared by treating LO2 cells with 0.3% EtOH for 8 h. 0.3% EtOH was reported to induce oxidative stress in living cells [50,51,52]. In addition, a preliminary time-dependent assay was performed, which identified an optimal incubation time length of 8 h (Supplementary Figure S2). As shown in A, DEA (1) (30 M) induced Nrf1 (B), NQO1 (C) and HO1 (D) protein levels by 1.2, 3.8 and 1.8-fold, respectively, in ALD model cells, making it more potent than ML334 (30 M), which increased Nrf1, NQO1, and HO1 protein levels by 1.0, 1.1, and 1.5-fold, respectively. Nrf2 directly controls the expression of the transcriptional factor PGC1, a regulator of mitochondrial biogenesis, while it also directly regulates the expression of mitochondrial antioxidant enzymes, such as superoxide dismutase 2 (SOD2) [53,54]. Hence, the effect of DEA (1) on mRNA levels of SOD2, HO1, NQO1, Nrf1, and PGC1 in 0.3% EtOHtreated LO2 cells was also evaluated. 30 M of DEA (1) rescued SOD2 mRNA level in ALD model cells by 2-fold, compared to 1.5-fold with ML334 (E). DEA (1) (30 M) also increased NQO1 (F) and HO1 (G) mRNA levels by 2.0 and 2.5-fold, respectively, making it more potent than ML334 which increased NQO1 and HO1 mRNA levels by 1.8 and 1.4-fold, respectively. In addition, the mRNA levels of Nrf1 and PGC1 were also increased 2-fold by compound DEA (1) (30 M) (H,I). The results also show a slight increase of NQO1, HO1, and PGC1 levels by 0.3% EtOH treatment alone, which has been previously observed in other studies [55]. However, the subsequent increases of antioxidant genes were much greater upon further treatment of DEA (1). Taken together, these results indicate that DEA (1) can activate Nrf2 downstream antioxidant genes in EtOHtreated human liver cells, a cellular model of ALD.\n",
      "--------------------------------------------------------------------------------\n",
      "[8] body :: 2.6. DEA (1) Repairs the EtOH Induced Mitochondrial Dysfunction and Improves the Antioxidant Activity in LO2 Cells\n",
      "Inflammation and oxidative stress increase ROS production, and the activated Nrf2 was able to maintain low intracellular ROS level to protect the cell from oxidant injury [17]. Moreover, growing evidence shows that mitochondrial and oxidant injury induced by ethanol consumption play key roles in alcoholinduced liver injury [56]. Alcoholinduced ROS causes mitochondrial membrane depolarization and mitochondrial permeability transition (MPT), leading to hepatic apoptosis and necrosis [54]. Therefore, we detected the effect of DEA (1) and ML334 on ROS level in 0.3% EtOHtreated LO2 cells (A). The results indicated that both 1 and ML334 could significantly decrease ROS level induced by EtOH. The MPTP assay showed that DEA (1) (30 M) can restore MPTP function (40%) with a higher activity as compared to ML334 (34%) (B,C). ATP is an indicator of mitochondrial function and ATP level are significantly reduced in ethanolfed Nrf2/ mice [57]. DEA (1) (30 M) rescued 50% of ATP in LO2 cells after EtOH treatment, making it more potent than ML334, which rescued 20% of the ATP level under the same conditions (D). Finally, DEA (1) (30 M) induced a higher increase of antioxidant activity (4-fold) compared to ML334 (2-fold) under the same conditions (E).\n",
      "--------------------------------------------------------------------------------\n",
      "[9] body :: 3. Discussion\n",
      "ALD is a serious chronic liver disease caused by oxidative stress and alcohol metabolism, which leads to massive global deaths [1]. Unfortunately, until now, there are no effective therapeutic drugs approved by the FDA for the treatment of ALD [58]. The pathogenesis of ALD is poorly characterized and research on the mechanism is still not clear. A number of reports have shown that the Keap1Nrf2 pathway plays an important role in activating ARE antioxidant signaling [14]. Nrf2 is regulated via Keap1-independent pathways, including mitogen-activated protein kinaseErk and PI3KAkt pathways [59]. Among these regulatory pathways, GSK3 has emerged as a convergent point [60,61]. Because Nrf2 degradation and nuclear exclusion are regulated by GSK3, it is critical in switching off the selfprotective antioxidant stress response after injury [59,62]. Hence, targeting Keap1 and GSK3 to activate Nrf2 are potential therapeutic strategies for ALD. Natural products provide diverse scaffolds with high bioactivity and low toxicity, and could be developed as promising candidates in drug development [63]. A. cinnamomea is a traditional Chinese herb that is a component of various medicines [64]. Triterpenoids are the major constituent in A. cinnamomea, and several studies have demonstrated their antioxidative and hepatoprotective effects [65,66]. However, the hepatoprotective mechanism of these compounds has not been extensively explored. DEA (1) has previously shown promising hepatoprotective activity in some studies [67], including in mouse models [39,67,68,69]. In recent years, many Keap1Nrf2 PPI inhibitors and GSK3 inhibitors have been reported for various indications, such as Alzheimers disease, cancers, Parkinsons disease, kidney diseases, Acetaminophen (APAP)induced liver injury, and ferroptosis [30,70,71,72,73,74]. However, few have been studied in the context of ALD. In our present study, we discovered DEA (1) as a potent Keap1Nrf2 PPI inhibitor from A. cinnamomea extracts. We explored the detailed mechanism of hepatoprotective action of DEA (1) in ALD, which acted via targeting both the Keap1Nrf2 PPI and GSK3 (). To our best knowledge, the dual inhibition mechanism of DEA (1) in ALD has not been reported before. In in vitro assays, DEA (1) inhibited the Keap1Nrf2 PPI (EC50 = 14.1 M) as well as GSK3 kinase activity (EC50 = 8.0  0.7 M) with greater potency than the known Keap1Nrf2 inhibitor, ML334. Cellular experiments confirmed the ability of DEA (1) to engage Keap1 and GSK3 in cellulo to activate Nrf2 and stimulate the expression of AREcontrolled genes. The specificity of DEA (1) was further confirmed using Keap1 and GSK3 knockdown experiments. DEA (1) treatment phenocopied dual knockdown of Keap1 and GSK3, suggesting that DEA (1) acted simultaneously through both pathways to exert its biological effects. Moreover, DEA (1) increased Nrf2 nuclear accumulation, mimicking GSK3 knockdown, without influencing Keap1 level. Finally, we showed that DEA (1) could restore MPTP function and reduce ROS level induced by EtOH in ALD model cells. DEA (1) displayed low cytotoxicity in both normal and cancerous liver cells, indicating its potential safety for treating ALD in vivo. In summary, DEA (1) is a promising scaffold for the further development of dual inhibitors of the Keap1Nrf2 PPI and GSK3 as therapeutic agents against ALD. The structural modification of compound DEA (1) is in progress and animal studies will be performed in due course.\n",
      "--------------------------------------------------------------------------------\n",
      "[10] body :: 4.1. Cell Lines and Culture\n",
      "The human normal liver cell line (LO2), human embryonic kidney 293 T cell line, and HepG2 cell line were purchased from American Type Culture Collection (Manassas, VA, USA). Cells were cultured in Dulbeccos modified Eagles medium (DMEM), with 10% FBS, 100 units/mL penicillin, 100 g/mL streptomycin. Fetal bovine serum (FBS), DMEM, penicillin, and streptomycin were purchased from Gibco BRL (Gaithersburg, MD, USA).\n",
      "--------------------------------------------------------------------------------\n",
      "[11] body :: 4.2. Chemical\n",
      "Compounds 113 (purity > 98% by HPLC analysis) were isolated from A. cinnamomea in a previous study [75]. New 1H and 13C nuclear magnetic resonance (NMR) were recorded for DEA (1) to verify its identity for this study (Supplementary Figure S1). Spectral data were consistent with our previous report [75].\n",
      "--------------------------------------------------------------------------------\n",
      "[12] body :: 4.3. Fluorescence Polarization Assay\n",
      "The ability of compounds 113 and ML334 to inhibit Keap1Nrf2 peptide binding was evaluated using a FP assay according to the manufacturers instruction (BPS Bioscience, San Diego, CA, USA). ML334 was a noncovalent small molecule inhibitor of the Keap1Nrf2 and purchased from MedChemExpress (Princeton, NJ, USA). The tested compounds were dissolved in the provided buffer at the indicated concentrations. Then, the required volumes of Nrf2 peptide, Bovine Serum Albumin (BSA), Keap1 protein, and assay buffer were added to each well, followed by incubation for 30 min and the fluorescent polarization value was recorded.\n",
      "--------------------------------------------------------------------------------\n",
      "[13] body :: 4.4. Western Blot (WB) and CoImmunoprecipitation (CoIP)\n",
      "The protocol was reported in a previous study [76]. In brief, LO2 cells were seeded at a density of 2  106 cells in a sixwell plate. LO2 lysate is harvested after incubation with 10 M of compound DEA (1) or ML334 for 8 h. A 10% SDSPAGE gel was used to separate the LO2 cell lysates, which were transferred to a PVDF membrane (BioRad, Hercules, CA, USA). The membranes were incubated with antibodies antiKeap1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 8047S), Nrf2 (1:1000, Proteintech, Wuhan, China; Cat# 16396-1-AP), GSK3 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 9832S), HO1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 5853S), NQO1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 3187S), actin (1:1000, Absin, Shanghai, China; Cat#: abs137975), or anti-lamin B (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat # 12586S) overnight. The membranes were washed with wash buffer three times. A secondary antibody (1:1000) was added to the membranes. After incubated for 2 h, the proteins bands were detected using enhanced chemiluminescent Plus reagents (GE Healthcare, Boston, MA, USA) and analyzed by Image Lab. The antibodies were purchased from Abcam (Waltham, MA, USA). CoIP between Keap1 and Nrf2 was performed following the protocol from Life Technologies. Briefly, lysates from 2  106 LO2 cells were incubated with antiNrf2 or Rabbit mAb IgG (Abcam, Waltham, MA, USA; Cat#: ab205718) overnight. Then, samples were incubated with 10 L Dynabeads (Life technologies, Foster, CA, USA) for 6 h. The Dynabeads were washed and analyzed by WB with the Keap1 antibody.\n",
      "--------------------------------------------------------------------------------\n",
      "[14] body :: 4.5. Real-Time Quantitative Polymerase Chain Reaction (RTqPCR)\n",
      "LO2 cells were incubated with 10 M or 30 M compound DEA (1) for 8 h and the total RNA was extracted using the NucleoSpin RNA Plus kit (Takara, Tokyo, Japan). We checked the RNA quality during the RTqPCR operation and ensured that all the A260/280 value of samples were around 2, which is consistent with the literature [77]. We also performed an RT negative control and found no PCR amplification. Then, cDNA was synthesized using the PrimeScript RT Reagent Kit (Takara, Tokyo, Japan). The RTqPCR conditions were as previously reported, with minor modifications [77]. DNA denaturation temperature: 95 C (30 s); annealing temperature: 58 C (30 s); extension temperature: 72 C (45 s), number of cycles: 40. RTqPCR was performed and analyzed as described previously [78]. The related primers are listed in Supplementary Table S1.\n",
      "--------------------------------------------------------------------------------\n",
      "[15] body :: 4.6. MTT Assay\n",
      "The LO2, HUVEC and HEK 293 T cell lines were seeded at 1  104 cells per well in 96well plates and incubated with DMEM overnight at 37 C in a humidified CO2 incubator to ensure attachment. The medium was then replaced with a medium that contained the different concentrations of compound for 48 h at 37 C. The final concentration of DMSO was 0.1% or lower in all cases (including controls). The medium with 100 L of MTT (3-(4,5-dimethylthiazol-2-yl)-2,5-tetrazolium bromide) reagent (1 mg/mL) was then replaced. After 4 h incubation, it was replaced with 100 L of DMSO and measured using a microplate reader at 570 nm.\n",
      "--------------------------------------------------------------------------------\n",
      "[16] body :: 4.7. Detection of ROS and Antioxidant Activity\n",
      "LO2 cells at a density of 2  106 cells/well in a six-well plate were treated with different concentrations of DEA (1) or 10 M ML334 for 8 h. ROS level and antioxidant activity were measured using the Reactive Oxygen Species Assay kit and the Total Antioxidant Capacity Assay Kit with FRAP, respectively, according to the manufacturers instructions (Beyotime, Shanghai, China).\n",
      "--------------------------------------------------------------------------------\n",
      "[17] body :: 4.8. Nuclear and Cytoplasmic Extraction\n",
      "LO2 cells were incubated with 10 M or 30 M of DEA (1) for 8 h. The nuclear and cytoplasmic extraction followed the protocol from the manufacturer (Epigentek, New York, NY, USA, EpiQuik Nuclear Extraction Kit I, Catalog # OP-0002), Then, protein expression was analyzed by WB with the indicated antibodies.\n",
      "--------------------------------------------------------------------------------\n",
      "[18] body :: 4.9. GSK3 Kinase Assay\n",
      "The ability of DEA (1) and ML334 to inhibit GSK3 activity was evaluated using luminescent kinase assay according to the manufacturers instruction (Promega Corporation, Madison, WI, USA). Briefly, the 384 well white plate was mixed with 1 L of inhibitor, 2 L of enzyme, and 2 L of substrate/ATP mix and incubated at room temperature for 60 min. After 60 min, 5 L of ADPGlo reagent was added and incubated at room temperature for 40 min. Then 10 L of kinase detection reagent was added, incubated at room temperature for 30 min and the luminescence value was recorded (integration time 0.51 s).\n",
      "--------------------------------------------------------------------------------\n",
      "[19] body :: 4.10. Dual Luciferase Reporter Gene Assay\n",
      "LO2 cells were seeded in a sixwell plate with 80% confluence in a DMEM medium for 24 h. The AREluciferase vector or a control plasmid with TurboFect transfection reagent (Thermo Fisher Scientific, Waltham, MA, USA) were transfected into LO2 cells according to the manufacturers instruction, and cells were incubated for 48 h. Then, LO2 cells were incubated with 10 M compound DEA (1), 10 M ML334 or DMSO for 4 h. The ARE activity was detected through the manufacturers instruction (Beyotime Dual Luciferase Reporter Gene Assay Kit, Shanghai, China).\n",
      "--------------------------------------------------------------------------------\n",
      "[20] body :: 4.11. Cellular Thermal Shift Assay (CETSA)\n",
      "LO2 cell lysates were incubated with 10 M compound DEA (1) and DMSO at room temperature for 30 min. The lysates are divided and heated individually at the indicated temperature. The supernatants of the heated lysates were collected after centrifugation, and then analyzed by WB with the Keap1, Nrf2, and actin antibodies.\n",
      "--------------------------------------------------------------------------------\n",
      "[21] body :: 4.12. Mitochondrial Permeability Transition Pore (MPTP) Assay\n",
      "MPTP kit was used for evaluation of mitochondrial permeability transition according to the manufacturers instructions (Beyotime, Shanghai, China). A density of 1  104 cells in a sixwell plate, EtOHinduced LO2 cells were treated with 10 M of ML334, 10 or 30 M compound DEA (1), DMSO for 8 h, and then pretreated with calcein or calcein and CoCl2. Cells were stained by calcein acetoxymethyl ester (Calcein AM) and MPT function was detected by fluorescence spectroscopy.\n",
      "--------------------------------------------------------------------------------\n",
      "[22] body :: 4.13. SiRNA Gene Knockdown\n",
      "The sequences of Keap1 siRNA are 5-GGCCUUUGGCAUCAUGAACTT-3 (sense) and 5-GUUCAUGAUGCCAAAGGCCTG-3 (antisense) [79]. GSK3 siRNA is 5-GCAUUUAUCGUUAACCUAA-3 (sense) and 5-UUAGGUUAACGA UAAAUGC-3 (antisense) [80]. Keap1 and GSK3 siRNA was transfected with Lipo3000 reagent (Thermo Fisher Scientific, Waltham, MA, USA) into 1  106 LO2 cells following the manufacturers instruction. After incubation for 48 h, Keap1 and GSK3 protein expression and functional experiments were performed.\n",
      "--------------------------------------------------------------------------------\n",
      "[23] body :: 4.14. Statistical Analysis\n",
      "Statistical significance was determined using the Students t-test for experiments comparing two groups. The data is homogeneous and follows a normal distribution before proceeding to the statistical analysis. Comparisons among groups were analysed using analysis of variance (ANOVA). All statistical tests were done using the GraphPad Prism version 8.0 software (GraphPad Software Inc. San Diego, CA, USA). All values were expressed as the mean  standard deviation (SD). p < 0.05 means statistical significance.\n",
      "--------------------------------------------------------------------------------\n",
      "[24] body :: 5. Conclusions\n",
      "Recent studies on ALD are scarce compared to research on other liver diseases, and there are still no highly satisfactory therapeutic options for ALD. To our knowledge, no dual inhibitor of the Keap1Nrf2 PPI and GSK3 to treat alcohol-induced liver injury has been reported. This paper has revealed that DEA (1), one of the active triterpenes isolated from A. cinnamomea, could be a potential lead compound for treating ALD. Compound DEA (1) inhibited the Keap1Nrf2 interaction and GSK3 activity both in vitro and in cellulo, and displayed low cytotoxicity at the concentrations required for inducing antioxidant activity. Moreover, DEA (1) could engage Keap1 and GSK3 in liver cells and increase Nrf2 nucleus translocation, thereby activating ARE transcriptional activity and upregulating antioxidant genes and mitochondrial biogenesis regulators. Significantly, compound DEA (1) displayed more potent hepatoprotective activity when compared with the classic Keap1Nrf2 PPI inhibitor ML334. Therefore, the structure of DEA (1) may be optimized to expand the diversity, improve the activity, and be used in animal research. To our knowledge, this paper firstly reports DEA (1) acts as the dual inhibitor of Keap1Nrf2 PPI and GSK3 with the potential for protecting against ALD. We anticipate that natural product DEA (1) can be considered as a potential scaffold for the development of clinical agents for treating ALD.\n",
      "--------------------------------------------------------------------------------\n",
      "[25] body :: Figure 1\n",
      "[[FIG::pharmaceuticals-16-00014-f001]]\n",
      "--------------------------------------------------------------------------------\n",
      "[26] body :: Figure 2\n",
      "[[FIG::pharmaceuticals-16-00014-f002]]\n",
      "--------------------------------------------------------------------------------\n",
      "[27] body :: Figure 3\n",
      "[[FIG::pharmaceuticals-16-00014-f003]]\n",
      "--------------------------------------------------------------------------------\n",
      "[28] body :: Figure 4\n",
      "[[FIG::pharmaceuticals-16-00014-f004]]\n",
      "--------------------------------------------------------------------------------\n",
      "[29] body :: Figure 5\n",
      "[[FIG::pharmaceuticals-16-00014-f005]]\n",
      "--------------------------------------------------------------------------------\n",
      "[30] body :: Figure 6\n",
      "[[FIG::pharmaceuticals-16-00014-f006]]\n",
      "--------------------------------------------------------------------------------\n",
      "[31] body :: Figure 7\n",
      "[[FIG::pharmaceuticals-16-00014-f007]]\n",
      "--------------------------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    # ----- helpers for preview -----\n",
    "    def preview(chunks: List[Dict], n: int = 120):\n",
    "        for i, ch in enumerate(chunks, 1):\n",
    "            title = ch.get(\"title\") or ch.get(\"section\") or \"(untitled)\"\n",
    "            print(f\"[{i}] {ch['type']} :: {title}\")\n",
    "            print(ch[\"text\"][:] + (\"\" if len(ch[\"text\"]) > n else \"\"))\n",
    "            print(\"-\" * 80)\n",
    "\n",
    "    \n",
    "    root = ET.fromstring(fulltext_xml)\n",
    "    article = root if root.tag == \"article\" else first(root, \"article\")\n",
    "    if article is None:\n",
    "        raise ValueError(\"No <article> element found.\")\n",
    "\n",
    "    \n",
    "    sections = jats_to_chunks(fulltext_xml)\n",
    "\n",
    "    preview(sections)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "85b50fed-004f-4bd0-89c2-6dbcb36f2ea8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sections"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6b99a84-4436-4ccf-9136-70345a07c764",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6bc0c1a9-d923-497c-9c11-5506e8e913b5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60d99c70-dbf2-4a1a-b24e-faa25e9df1e2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8ef7492-3617-400d-9813-46ad850febb0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "7eb2c68d-fdd3-44d4-9f4c-2ac6bebd6f46",
   "metadata": {},
   "source": [
    "# Zimanje link do tileshop na slikite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7937f44d-523a-45c1-a812-a45619de4c4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import xml.etree.ElementTree as ET\n",
    "from urllib.parse import urlencode\n",
    "\n",
    "XLINK = \"{http://www.w3.org/1999/xlink}\"\n",
    "\n",
    "WS_RE = re.compile(r\"\\s+\")\n",
    "def norm_ws(s: str) -> str:\n",
    "    return WS_RE.sub(\" \", (s or \"\")).strip()\n",
    "\n",
    "\n",
    "SKIP_TAGS = {\"ext-link\", \"uri\", \"inline-formula\", \"disp-formula\", \"table-wrap\"}\n",
    "\n",
    "def extract_text(node: ET.Element) -> str:\n",
    "    parts = []\n",
    "    def walk(el):\n",
    "        if el.tag in SKIP_TAGS:\n",
    "            if el.tail:\n",
    "                parts.append(el.tail)\n",
    "            return\n",
    "        if el.text:\n",
    "            parts.append(el.text)\n",
    "        for child in list(el):\n",
    "            walk(child)\n",
    "        if el.tail:\n",
    "            parts.append(el.tail)\n",
    "    walk(node)\n",
    "    return norm_ws(\"\".join(parts))\n",
    "\n",
    "def pmcid_number(pmcid: str) -> str:\n",
    "    return pmcid.upper().replace(\"PMC\", \"\")\n",
    "\n",
    "def tileshop_inline_url(pmcid: str, filename: str, p: str = \"PMC3\",\n",
    "                        title: str = \"Click on image to zoom\") -> str:\n",
    "    q = {\"title\": title, \"p\": p, \"id\": f\"{pmcid_number(pmcid)}_{filename}\"}\n",
    "    return \"https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?\" + urlencode(q)\n",
    "\n",
    "def _first_graphic(fig: ET.Element) -> ET.Element | None:\n",
    "    g = fig.find(\".//graphic\")\n",
    "    if g is not None:\n",
    "        return g\n",
    "    alt = fig.find(\".//alternatives/graphic\")\n",
    "    return alt\n",
    "\n",
    "def figure_links_from_jats(xml_text: str, pmcid: str):\n",
    "    \"\"\"Return label, FULL caption text, and a best URL for each <fig>.\"\"\"\n",
    "    root = ET.fromstring(xml_text)\n",
    "    article = root if root.tag == \"article\" else root.find(\"article\")\n",
    "    if article is None:\n",
    "        return []\n",
    "\n",
    "    out = []\n",
    "    for fig in article.findall(\".//fig\"):\n",
    "        label = (fig.findtext(\"label\") or \"\").strip()\n",
    "\n",
    "        \n",
    "        cap_el = fig.find(\"caption\")\n",
    "        if cap_el is not None:\n",
    "            caption = extract_text(cap_el)           \n",
    "        else:\n",
    "            caption = \"\"                             \n",
    "\n",
    "       \n",
    "        g = _first_graphic(fig)\n",
    "        if g is None:\n",
    "            continue\n",
    "        href = g.get(f\"{XLINK}href\")\n",
    "        if not href:\n",
    "            continue\n",
    "\n",
    "        if href.startswith((\"http://\", \"https://\", \"ftp://\")):\n",
    "            url = href                 \n",
    "        else:\n",
    "            url = tileshop_inline_url(pmcid, href)  \n",
    "\n",
    "        out.append({\n",
    "            \"label\": label,\n",
    "            \"caption\": caption,       \n",
    "            \"url\": url,\n",
    "            \"href\": href\n",
    "        })\n",
    "    return out\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "85450ca4-5807-4115-aadc-d03634ae1d94",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'label': 'Figure 1',\n",
       "  'caption': 'Screening of compounds 113 and identification of DEA (1) as a potent Keap1Nrf2 inhibitor in vitro with low toxicity. (A) The binding activity of Keap1Nrf2 was measured via an FP assay with the compounds added at 50 M. (B) Dose-response effect of compound DEA (1) on Keap1Nrf2 binding. (C) Compound DEA (1) inhibited GSK3 activity in vitro. (D) Dose-response effect of compound DEA (1) on GSK3 activity. (E) Compound DEA (1) enhances ARE activity in LO2 cells determined by the dual luciferase assay. (FH) The cytotoxicity of DEA (1) and ML334 on LO2, HEK 293T, and HepG2 cell lines for 48 h. Cells were treated with 0100 M of DEA (1) and ML334 for 48 h and cytotoxicity was detected using the MTT assay. Data are represented as mean  SD. * p < 0.05, ** p < 0.01, *** p < 0.001 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. DEA: Dehydroeburicoic acid; FP: fluorescence polarization; DMSO: Dimethyl sulfoxide; GSK3: Glycogen synthase kinase 3; IgG: Immunoglobulin G; Keap1: Kelch-like erythroid cellderived protein with CNC homologyassociated protein 1; Nrf2: Nuclear respiratory factor-2; MTT: (3-(4,5-dimethylthiazol-2-yl)-2,5-diphenyltetrazolium bromide) tetrazolium reduction.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g001.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g001.jpg'},\n",
       " {'label': 'Figure 2',\n",
       "  'caption': 'Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3. (A) LO2 cells were treated with 10 M DEA (1) and positive control ML334 for 8 h. Interactions between Keap1 and Nrf2 in LO2 cells were examined by WB and CoIP. (B) Quantification analysis of Keap1 in WB. Error bars represent the standard deviations of the results from three independent experiments. (C) LO2 lysates were treated with 10 M DEA (1) or DMSO and incubated at room temperature for 30 minutes. Then, CETSA was performed to assess Keap1, Nrf2, GSK3, and actin thermal stability. (D) Quantification analysis of Keap1, Nrf2, GSK3, and actin in WB. Data are represented as mean  SD. * p < 0.05, ** p < 0.01 vs. DMSO group (Students t-test). Error bars represent the standard deviations of the results from three independent experiments. AntiNrf2: antiNrf2 antibody; WB: Western blot; CoIP: coimmunoprecipitation.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g002.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g002.jpg'},\n",
       " {'label': 'Figure 3',\n",
       "  'caption': 'Compound DEA (1) regulates the antioxidant effects, at least in part, by targeting Keap1 and GSK3. (A) Keap1 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. Keap1, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (B) Quantification analysis of Keap1, Nrf2, HO1, and NQO1 levels in Keap1 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, * p < 0.05 vs. DMSO. Error bars represent the standard deviations of the results from three independent experiments. (C) GSK3 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. GSK3, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (D) Quantification analysis of GSK3, Nrf2, HO1, and NQO1 levels in GSK3 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, * p < 0.05, ** p < 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. (E) Keap1 and GSK3 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. GSK3, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (F) Quantification analysis of GSK3, Keap1, Nrf2, HO1, and NQO1 levels in Keap1 and GSK3 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, ** p < 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA Keap1: SiKeap1; SiRNA GSK3: SiGSK3; SiGSK3 + SiKeap1: SiRNA GSK3 + SiRNA Keap1.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g003.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g003.jpg'},\n",
       " {'label': 'Figure 4',\n",
       "  'caption': 'Compound DEA (1) increases Nrf2 translocation into the nucleus. (A) LO2 cells were treated with 10 or 30 M compound DEA (1) for 8 h and then analyzed by WB to detect the level of Nrf2 in the nucleus and cytoplasm. actin and lamin B were used as internal controls. (B) GSK3 knockdown increased nuclear Nrf2 level. Keap1 and Nrf2 were blotted to control for the total protein levels. The data represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA GSK3: SiGSK3.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g004.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g004.jpg'},\n",
       " {'label': 'Figure 5',\n",
       "  'caption': 'Compound DEA (1) induces antioxidant factor expression in the ALD cell model. (A) Effects of 10 M, 30 M DEA (1), and 30 M ML334 on Nrf1, NQO1, HO1, Keap1, and Nrf2 protein levels in LO2 cells after 8 h treatment. (BD) Quantification analysis of Nrf1 (B), NQO1 (C), and HO1 (D) in WB. (EI) Effects of DEA (1) and ML334 on (E) SOD2, (F) NQO1, (G) HO1, (H) Nrf1, (I) PGC1 mRNA levels in 0.3% EtOHinduced LO2 cells. Data are represented as mean  SD. * p < 0.05, *** p < 0.001 vs. control group. The data represent the standard deviations of the results from three independent experiments. EtOH: Ethyl alcohol. Nrf1: Nuclear respiratory factor1; NQO1: NAD(P)H dehydrogenase [quinone] 1; PGC1: Peroxisome proliferatoractivated receptor  coactivator 1; SOD 2: Superoxide dismutase 2.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g005.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g005.jpg'},\n",
       " {'label': 'Figure 6',\n",
       "  'caption': 'Compound DEA (1) protects mitochondrial dysfunction from 0.3% EtOHinduced injury. (A) 0.3% EtOHinduced LO2 cells were treated with 30 M ML334, 10 or 30 M compound DEA (1) for 8 h. Intracellular ROS level were detected by DCFHDA probe and measured using the confocal laser scanning microscopy. (B) EtOHinduced LO2 cells were treated with 10 M ML334, 10 or 30 M compound DEA (1), DMSO for 8 h, and MPTP function was detected by fluorescence spectroscopy. (C) Quantification analysis of fluorescence in MPTP assay. (D) ATP level were detected using an ATP assay kit in 0.3% EtOHinduced LO2 cells. (E) Antioxidant activity of DEA (1) was detected in 0.3% EtOHinduced LO2 cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01, vs. control group. The data represent the standard deviations of the results from three independent experiments. ATP: Adenosine triphosphate; DCFHDA: Dichlorodihydrofluorescein diacetate; MPTP: Mitochondrial Permeability Transition Pore Assay; ROS: Reactive oxygen species.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g006.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g006.jpg'},\n",
       " {'label': 'Figure 7',\n",
       "  'caption': 'Hepatoprotective mechanism of the dual inhibitor DEA (1) against EtOHinduced cell injury.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g007.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g007.jpg'}]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "figs = figure_links_from_jats(fulltext_xml,pmcid)\n",
    "figs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11b46e30-ce29-4460-85c1-5225d72246bb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f02ffb55-5586-4b4c-acc2-ee7698ef972f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "badab799-7c02-4688-a774-02608dbce1fb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46e653ec-0711-41ae-868a-c08bdaa0a57d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7cc82264-4d33-4b86-8780-d332c2e83e6f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e531fab-f70f-4349-9f80-b3b849c20d47",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "b6529345-0831-4e07-840a-d38cdecb8f88",
   "metadata": {},
   "source": [
    "# Scrapeanje od tileshop stranata link do slikata i siminnjanje nejzino"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "349a5722-c5eb-4b3d-a151-2dbd40c78478",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\n",
    "def make_driver(headless: bool = True) -> webdriver.Chrome:\n",
    "    opts = Options()\n",
    "    if headless:\n",
    "        opts.add_argument(\"--headless=new\")\n",
    "    opts.add_argument(\"--disable-gpu\")\n",
    "    opts.add_argument(\"--no-sandbox\")\n",
    "    opts.add_argument(\"--window-size=1280,1000\")\n",
    "    # be a tiny bit more like a real browser\n",
    "    opts.add_argument(\"--user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 \"\n",
    "                      \"(KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36\")\n",
    "    return webdriver.Chrome(options=opts)\n",
    "\n",
    "\n",
    "SEL_IMG = 'img[title=\"Click to change focus to this area of image.\"]'\n",
    "FALLBACK_SEL = \"img#tileshop, img.tileshop\"\n",
    "\n",
    "SAFE = re.compile(r\"[^A-Za-z0-9._-]+\")\n",
    "def safe_name(s: str) -> str:\n",
    "    return SAFE.sub(\"_\", s).strip(\"_\")\n",
    "\n",
    "def resolve_img_src(driver: webdriver.Chrome, inline_url: str, timeout: int = 20) -> str:\n",
    "    driver.get(inline_url)\n",
    "    wait = WebDriverWait(driver, timeout)\n",
    "    try:\n",
    "        el = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, SEL_IMG)))\n",
    "    except Exception:\n",
    "        el = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, FALLBACK_SEL)))\n",
    "    src = el.get_attribute(\"src\") \n",
    "    return urljoin(inline_url, src)\n",
    "\n",
    "def selenium_cookies_to_jar(driver: webdriver.Chrome, domain_hint: str) -> requests.cookies.RequestsCookieJar:\n",
    "    jar = requests.cookies.RequestsCookieJar()\n",
    "   \n",
    "    for c in driver.get_cookies():\n",
    "        dom = c.get(\"domain\") or \"\"\n",
    "        if domain_hint in dom:\n",
    "            jar.set(c[\"name\"], c[\"value\"], domain=c.get(\"domain\"), path=c.get(\"path\", \"/\"))\n",
    "    return jar\n",
    "\n",
    "def looks_like_html(data: bytes) -> bool:\n",
    "    head = data[:64].lstrip().lower()\n",
    "    return head.startswith(b\"<!doctype\") or head.startswith(b\"<html\") or b\"<body\" in head\n",
    "\n",
    "def ensure_image_bytes(raw: bytes) -> Image.Image:\n",
    "    \"\"\"Open with PIL to guarantee it's a real image.\"\"\"\n",
    "    im = Image.open(io.BytesIO(raw))\n",
    "    im.load()\n",
    "    return im\n",
    "\n",
    "def pick_ext_from_content_type(ct: str) -> str:\n",
    "    if not ct:\n",
    "        return \".png\"\n",
    "    ct = ct.lower().split(\";\")[0].strip()\n",
    "    return {\n",
    "        \"image/jpeg\": \".jpg\",\n",
    "        \"image/jpg\": \".jpg\",\n",
    "        \"image/png\": \".png\",\n",
    "        \"image/webp\": \".webp\",\n",
    "        \"image/tiff\": \".tif\",\n",
    "        \"image/gif\": \".gif\",\n",
    "        \"image/bmp\": \".bmp\",\n",
    "    }.get(ct, \".png\")\n",
    "\n",
    "def download_image_with_headers(url: str, referer: str, cookies=None, timeout=60) -> (bytes, str):\n",
    "    headers = {\n",
    "        \"User-Agent\": (\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 \"\n",
    "                       \"(KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36\"),\n",
    "        \"Referer\": referer,\n",
    "        \"Accept\": \"image/avif,image/webp,image/apng,image/*,*/*;q=0.8\",\n",
    "        \"Accept-Language\": \"en-US,en;q=0.9\",\n",
    "        \"Cache-Control\": \"no-cache\",\n",
    "        \"Pragma\": \"no-cache\",\n",
    "    }\n",
    "    r = requests.get(url, headers=headers, cookies=cookies, timeout=timeout)\n",
    "    r.raise_for_status()\n",
    "    return r.content, r.headers.get(\"Content-Type\", \"\")\n",
    "\n",
    "def save_as_png_norm(raw: bytes, outpath: Path):\n",
    "    \"\"\"Normalize weird encodings to RGB PNG for consistency.\"\"\"\n",
    "    im = ensure_image_bytes(raw)\n",
    "    if im.mode not in (\"RGB\", \"RGBA\"):\n",
    "        im = im.convert(\"RGB\")\n",
    "    outpath.parent.mkdir(parents=True, exist_ok=True)\n",
    "    im.save(outpath, format=\"PNG\", optimize=True)\n",
    "\n",
    "\n",
    "def download_all_figures(\n",
    "    fig_list,\n",
    "    pmcid: str,\n",
    "    out_dir: str = r\"D:\\Users\\User\\VNPRagProekt\\figures\",\n",
    "    headless: bool = True,\n",
    "    retries: int = 3,           \n",
    "    sleep: float = 1.0,\n",
    "    figure_retries: int = 3      \n",
    "):\n",
    "    out_dir = Path(out_dir)\n",
    "    out_dir.mkdir(parents=True, exist_ok=True)\n",
    "    driver = make_driver(headless=headless)\n",
    "\n",
    "    try:\n",
    "        for fig in fig_list:\n",
    "            label = fig.get(\"label\", \"(figure)\")\n",
    "            final_error = None\n",
    "\n",
    "            for ftry in range(1, figure_retries + 1):\n",
    "                try:\n",
    "                    inline_url = fig.get(\"url\") or fig.get(\"image_url\")\n",
    "                    if not inline_url:\n",
    "                        raise RuntimeError(\"No inline tileshop URL present for this figure\")\n",
    "\n",
    "                   \n",
    "                    img_url = resolve_img_src(driver, inline_url)\n",
    "                    fig[\"image_url\"] = img_url\n",
    "\n",
    "                    \n",
    "                    href = fig.get(\"href\") or f\"{label}.jpg\"\n",
    "                    base = safe_name(Path(href).stem)\n",
    "                    fname = f\"{pmcid}_{base}.png\"\n",
    "                    outpath = out_dir / fname\n",
    "\n",
    "                    \n",
    "                    cookies = selenium_cookies_to_jar(driver, domain_hint=\"ncbi.nlm.nih.gov\")\n",
    "\n",
    "                    \n",
    "                    last_err = None\n",
    "                    for attempt in range(1, retries + 1):\n",
    "                        try:\n",
    "                            raw, ctype = download_image_with_headers(\n",
    "                                img_url, referer=inline_url, cookies=cookies, timeout=60\n",
    "                            )\n",
    "                            if ((\"image\" not in (ctype or \"\").lower()) or looks_like_html(raw)):\n",
    "                                raise RuntimeError(f\"Non-image response (Content-Type={ctype!r})\")\n",
    "                            ensure_image_bytes(raw)  # validate\n",
    "                            save_as_png_norm(raw, outpath)\n",
    "                            fig[\"image_path\"] = str(outpath.resolve())\n",
    "                            print(f\" {label}  {outpath.name}\")\n",
    "                            last_err = None\n",
    "                            break\n",
    "                        except (requests.RequestException, UnidentifiedImageError, RuntimeError) as e:\n",
    "                            last_err = e\n",
    "                            if attempt < retries:\n",
    "                                time.sleep(sleep * attempt)\n",
    "                                continue\n",
    "                            else:\n",
    "                                raise\n",
    "\n",
    "                   \n",
    "                    final_error = None\n",
    "                    break\n",
    "\n",
    "                except Exception as e:\n",
    "                    final_error = e\n",
    "                    \n",
    "                    if ftry < figure_retries:\n",
    "                        time.sleep(sleep * ftry)\n",
    "                        continue\n",
    "                    else:\n",
    "                       \n",
    "                        fig.pop(\"image_path\", None)\n",
    "                        fig[\"error\"] = str(final_error)\n",
    "                        print(f\" {label} failed after {figure_retries} attempts: {final_error}\")\n",
    "\n",
    "    finally:\n",
    "        driver.quit()\n",
    "\n",
    "    return fig_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "15271acc-b5d3-42c2-9362-f90ec7f8020d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Figure 1  PMC9866905_pharmaceuticals-16-00014-g001.png\n",
      " Figure 2  PMC9866905_pharmaceuticals-16-00014-g002.png\n",
      " Figure 3  PMC9866905_pharmaceuticals-16-00014-g003.png\n",
      " Figure 4  PMC9866905_pharmaceuticals-16-00014-g004.png\n",
      " Figure 5  PMC9866905_pharmaceuticals-16-00014-g005.png\n",
      " Figure 6  PMC9866905_pharmaceuticals-16-00014-g006.png\n",
      " Figure 7  PMC9866905_pharmaceuticals-16-00014-g007.png\n"
     ]
    }
   ],
   "source": [
    "results = download_all_figures(figs, pmcid, out_dir=r\"D:\\Users\\User\\VNPRagProekt\\figures\", headless=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a92edab6-2e34-4922-b783-5ed678c4751e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d12c9a0-9599-4ffa-9823-411eda31519d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "167e1bdf-95a9-42cd-8dc5-0e4f49d44355",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'label': 'Figure 1',\n",
       "  'caption': 'Screening of compounds 113 and identification of DEA (1) as a potent Keap1Nrf2 inhibitor in vitro with low toxicity. (A) The binding activity of Keap1Nrf2 was measured via an FP assay with the compounds added at 50 M. (B) Dose-response effect of compound DEA (1) on Keap1Nrf2 binding. (C) Compound DEA (1) inhibited GSK3 activity in vitro. (D) Dose-response effect of compound DEA (1) on GSK3 activity. (E) Compound DEA (1) enhances ARE activity in LO2 cells determined by the dual luciferase assay. (FH) The cytotoxicity of DEA (1) and ML334 on LO2, HEK 293T, and HepG2 cell lines for 48 h. Cells were treated with 0100 M of DEA (1) and ML334 for 48 h and cytotoxicity was detected using the MTT assay. Data are represented as mean  SD. * p < 0.05, ** p < 0.01, *** p < 0.001 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. DEA: Dehydroeburicoic acid; FP: fluorescence polarization; DMSO: Dimethyl sulfoxide; GSK3: Glycogen synthase kinase 3; IgG: Immunoglobulin G; Keap1: Kelch-like erythroid cellderived protein with CNC homologyassociated protein 1; Nrf2: Nuclear respiratory factor-2; MTT: (3-(4,5-dimethylthiazol-2-yl)-2,5-diphenyltetrazolium bromide) tetrazolium reduction.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g001.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g001.jpg',\n",
       "  'image_url': 'https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=86761&s=158&r=1&c=1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g001.png'},\n",
       " {'label': 'Figure 2',\n",
       "  'caption': 'Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3. (A) LO2 cells were treated with 10 M DEA (1) and positive control ML334 for 8 h. Interactions between Keap1 and Nrf2 in LO2 cells were examined by WB and CoIP. (B) Quantification analysis of Keap1 in WB. Error bars represent the standard deviations of the results from three independent experiments. (C) LO2 lysates were treated with 10 M DEA (1) or DMSO and incubated at room temperature for 30 minutes. Then, CETSA was performed to assess Keap1, Nrf2, GSK3, and actin thermal stability. (D) Quantification analysis of Keap1, Nrf2, GSK3, and actin in WB. Data are represented as mean  SD. * p < 0.05, ** p < 0.01 vs. DMSO group (Students t-test). Error bars represent the standard deviations of the results from three independent experiments. AntiNrf2: antiNrf2 antibody; WB: Western blot; CoIP: coimmunoprecipitation.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g002.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g002.jpg',\n",
       "  'image_url': 'https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=86766&s=158&r=1&c=1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g002.png'},\n",
       " {'label': 'Figure 3',\n",
       "  'caption': 'Compound DEA (1) regulates the antioxidant effects, at least in part, by targeting Keap1 and GSK3. (A) Keap1 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. Keap1, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (B) Quantification analysis of Keap1, Nrf2, HO1, and NQO1 levels in Keap1 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, * p < 0.05 vs. DMSO. Error bars represent the standard deviations of the results from three independent experiments. (C) GSK3 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. GSK3, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (D) Quantification analysis of GSK3, Nrf2, HO1, and NQO1 levels in GSK3 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, * p < 0.05, ** p < 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. (E) Keap1 and GSK3 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. GSK3, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (F) Quantification analysis of GSK3, Keap1, Nrf2, HO1, and NQO1 levels in Keap1 and GSK3 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, ** p < 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA Keap1: SiKeap1; SiRNA GSK3: SiGSK3; SiGSK3 + SiKeap1: SiRNA GSK3 + SiRNA Keap1.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g003.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g003.jpg',\n",
       "  'image_url': 'https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=86772&s=158&r=1&c=1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g003.png'},\n",
       " {'label': 'Figure 4',\n",
       "  'caption': 'Compound DEA (1) increases Nrf2 translocation into the nucleus. (A) LO2 cells were treated with 10 or 30 M compound DEA (1) for 8 h and then analyzed by WB to detect the level of Nrf2 in the nucleus and cytoplasm. actin and lamin B were used as internal controls. (B) GSK3 knockdown increased nuclear Nrf2 level. Keap1 and Nrf2 were blotted to control for the total protein levels. The data represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA GSK3: SiGSK3.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g004.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g004.jpg',\n",
       "  'image_url': 'https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=86777&s=158&r=1&c=1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g004.png'},\n",
       " {'label': 'Figure 5',\n",
       "  'caption': 'Compound DEA (1) induces antioxidant factor expression in the ALD cell model. (A) Effects of 10 M, 30 M DEA (1), and 30 M ML334 on Nrf1, NQO1, HO1, Keap1, and Nrf2 protein levels in LO2 cells after 8 h treatment. (BD) Quantification analysis of Nrf1 (B), NQO1 (C), and HO1 (D) in WB. (EI) Effects of DEA (1) and ML334 on (E) SOD2, (F) NQO1, (G) HO1, (H) Nrf1, (I) PGC1 mRNA levels in 0.3% EtOHinduced LO2 cells. Data are represented as mean  SD. * p < 0.05, *** p < 0.001 vs. control group. The data represent the standard deviations of the results from three independent experiments. EtOH: Ethyl alcohol. Nrf1: Nuclear respiratory factor1; NQO1: NAD(P)H dehydrogenase [quinone] 1; PGC1: Peroxisome proliferatoractivated receptor  coactivator 1; SOD 2: Superoxide dismutase 2.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g005.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g005.jpg',\n",
       "  'image_url': 'https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=86782&s=158&r=1&c=1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g005.png'},\n",
       " {'label': 'Figure 6',\n",
       "  'caption': 'Compound DEA (1) protects mitochondrial dysfunction from 0.3% EtOHinduced injury. (A) 0.3% EtOHinduced LO2 cells were treated with 30 M ML334, 10 or 30 M compound DEA (1) for 8 h. Intracellular ROS level were detected by DCFHDA probe and measured using the confocal laser scanning microscopy. (B) EtOHinduced LO2 cells were treated with 10 M ML334, 10 or 30 M compound DEA (1), DMSO for 8 h, and MPTP function was detected by fluorescence spectroscopy. (C) Quantification analysis of fluorescence in MPTP assay. (D) ATP level were detected using an ATP assay kit in 0.3% EtOHinduced LO2 cells. (E) Antioxidant activity of DEA (1) was detected in 0.3% EtOHinduced LO2 cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01, vs. control group. The data represent the standard deviations of the results from three independent experiments. ATP: Adenosine triphosphate; DCFHDA: Dichlorodihydrofluorescein diacetate; MPTP: Mitochondrial Permeability Transition Pore Assay; ROS: Reactive oxygen species.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g006.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g006.jpg',\n",
       "  'image_url': 'https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=86787&s=158&r=1&c=1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g006.png'},\n",
       " {'label': 'Figure 7',\n",
       "  'caption': 'Hepatoprotective mechanism of the dual inhibitor DEA (1) against EtOHinduced cell injury.',\n",
       "  'url': 'https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click+on+image+to+zoom&p=PMC3&id=9866905_pharmaceuticals-16-00014-g007.jpg',\n",
       "  'href': 'pharmaceuticals-16-00014-g007.jpg',\n",
       "  'image_url': 'https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=86793&s=158&r=1&c=1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g007.png'}]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "figs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87cefe5f-8837-4ae5-9b18-bf914cff492a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "7eea5a8c-ac2d-48e5-ab49-18a2c3810b2b",
   "metadata": {},
   "source": [
    "# Model za sliki"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6929baef-4631-40d5-80da-da47ab0e6741",
   "metadata": {},
   "outputs": [],
   "source": [
    "import base64, json, requests\n",
    "\n",
    "OLLAMA_BASE = \"http://127.0.0.1:11434\"\n",
    "MODEL = \"qwen2.5vl:3b\" \n",
    "\n",
    "SYSTEM_PROMPT = (\n",
    "    \"You are a bioscience figure explainer for a RAG pipeline. \"\n",
    "    \"Write a summary that integrates evidence across all panels. \"\n",
    "    \"Be objective and specific. Do not copy the caption. No citations. \"\n",
    "    \"Return STRICT JSON only with the requested fieldsno extra keys, no prose.\"\n",
    ")\n",
    "\n",
    "USER_TEMPLATE = (\n",
    "    \"Paper figure {LBL}. Use the official caption ONLY as context; do not copy it.\\n\\n\"\n",
    "    \"Caption:\\n{CAP}\\n\\n\"\n",
    "    \"TASK: Synthesize all panels (A,B,...) into one short summary.\\n\\n\"\n",
    "    \"Return JSON with EXACTLY these fields:\\n\"\n",
    "    \"{{\\n\"\n",
    "    '  \"summary\": \"120 wordsmain scientific takeaway across all panels\",\\n'\n",
    "    '  \"key_terms\": [\"term1\", \"term2\", \"...\"],\\n'\n",
    "    '  \"caveats\": []\\n'\n",
    "    \"}}\\n\"\n",
    "    \"Output JSON only.\"\n",
    ")\n",
    "\n",
    "\n",
    "def b64_image(path: str) -> str:\n",
    "    with open(path, \"rb\") as f:\n",
    "        return base64.b64encode(f.read()).decode(\"utf-8\")\n",
    "\n",
    "def describe_figure(fig: dict, temperature: float = 0.2, num_predict: int = 320) -> dict:\n",
    "    user_prompt = USER_TEMPLATE.format(LBL=fig[\"label\"], CAP=fig[\"caption\"])\n",
    "    img_b64 = b64_image(fig[\"image_path\"])\n",
    "\n",
    "    payload = {\n",
    "        \"model\": MODEL,\n",
    "        \"stream\": False,\n",
    "        \"options\": {\"temperature\": temperature, \"num_predict\": num_predict},\n",
    "        \"messages\": [\n",
    "            {\"role\": \"system\", \"content\": SYSTEM_PROMPT},\n",
    "            {\"role\": \"user\", \"content\": user_prompt, \"images\": [img_b64]},\n",
    "        ],\n",
    "    }\n",
    "\n",
    "    r = requests.post(f\"{OLLAMA_BASE}/api/chat\", json=payload, timeout=300)\n",
    "    try:\n",
    "        r.raise_for_status()\n",
    "    except requests.HTTPError as e:\n",
    "        print(\"HTTP error from Ollama:\", r.status_code, r.text[:500])\n",
    "        raise e\n",
    "\n",
    "    text = (r.json().get(\"message\", {}) or {}).get(\"content\", \"\").strip()\n",
    "\n",
    "\n",
    "    try:\n",
    "        resp = json.loads(text)\n",
    "    except json.JSONDecodeError:\n",
    "        return {\"raw\": text}\n",
    "\n",
    "\n",
    "    resp.setdefault(\"summary\", \"\")\n",
    "    # resp.setdefault(\"rationale\", \"\")\n",
    "    resp.setdefault(\"key_terms\", [])\n",
    "    resp.setdefault(\"caveats\", [])\n",
    "    return resp\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "32c93c59-78ed-4b2a-b808-dad6368bb687",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_descriptions = []\n",
    "for fig in figs:\n",
    "    out = {\n",
    "        \"label\": fig[\"label\"],\n",
    "        \"image_path\": fig[\"image_path\"],\n",
    "        \"caption\": fig[\"caption\"],\n",
    "        \"llm_description\": describe_figure(fig),\n",
    "    }\n",
    "    figure_descriptions.append(out)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "421f1262-a490-48c5-afc1-674c72413f8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "if len(figure_descriptions) != 0:\n",
    "    figure_descriptions[0]['llm_description']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c1fe48cc-9305-4cda-b637-719e98a6ad28",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'label': 'Figure 1',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g001.png',\n",
       "  'caption': 'Screening of compounds 113 and identification of DEA (1) as a potent Keap1Nrf2 inhibitor in vitro with low toxicity. (A) The binding activity of Keap1Nrf2 was measured via an FP assay with the compounds added at 50 M. (B) Dose-response effect of compound DEA (1) on Keap1Nrf2 binding. (C) Compound DEA (1) inhibited GSK3 activity in vitro. (D) Dose-response effect of compound DEA (1) on GSK3 activity. (E) Compound DEA (1) enhances ARE activity in LO2 cells determined by the dual luciferase assay. (FH) The cytotoxicity of DEA (1) and ML334 on LO2, HEK 293T, and HepG2 cell lines for 48 h. Cells were treated with 0100 M of DEA (1) and ML334 for 48 h and cytotoxicity was detected using the MTT assay. Data are represented as mean  SD. * p < 0.05, ** p < 0.01, *** p < 0.001 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. DEA: Dehydroeburicoic acid; FP: fluorescence polarization; DMSO: Dimethyl sulfoxide; GSK3: Glycogen synthase kinase 3; IgG: Immunoglobulin G; Keap1: Kelch-like erythroid cellderived protein with CNC homologyassociated protein 1; Nrf2: Nuclear respiratory factor-2; MTT: (3-(4,5-dimethylthiazol-2-yl)-2,5-diphenyltetrazolium bromide) tetrazolium reduction.',\n",
       "  'llm_description': {'summary': 'Compound DEA (1) potently inhibits Keap1-Nrf2 binding and GSK3 activity, enhancing ARE activity in cells, while showing low cytotoxicity.',\n",
       "   'key_terms': ['Dehydroeburicoic acid',\n",
       "    'Keap1-Nrf2',\n",
       "    'GSK3',\n",
       "    'ARE activity',\n",
       "    'cytotoxicity'],\n",
       "   'caveats': []}},\n",
       " {'label': 'Figure 2',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g002.png',\n",
       "  'caption': 'Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3. (A) LO2 cells were treated with 10 M DEA (1) and positive control ML334 for 8 h. Interactions between Keap1 and Nrf2 in LO2 cells were examined by WB and CoIP. (B) Quantification analysis of Keap1 in WB. Error bars represent the standard deviations of the results from three independent experiments. (C) LO2 lysates were treated with 10 M DEA (1) or DMSO and incubated at room temperature for 30 minutes. Then, CETSA was performed to assess Keap1, Nrf2, GSK3, and actin thermal stability. (D) Quantification analysis of Keap1, Nrf2, GSK3, and actin in WB. Data are represented as mean  SD. * p < 0.05, ** p < 0.01 vs. DMSO group (Students t-test). Error bars represent the standard deviations of the results from three independent experiments. AntiNrf2: antiNrf2 antibody; WB: Western blot; CoIP: coimmunoprecipitation.',\n",
       "  'llm_description': {'summary': 'Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3, enhancing its stability and activity in LO2 cells.',\n",
       "   'key_terms': ['Compound DEA (1)',\n",
       "    'Nrf2',\n",
       "    'Keap1',\n",
       "    'GSK3',\n",
       "    'LO2 cells',\n",
       "    'Thermal stability'],\n",
       "   'caveats': []}},\n",
       " {'label': 'Figure 3',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g003.png',\n",
       "  'caption': 'Compound DEA (1) regulates the antioxidant effects, at least in part, by targeting Keap1 and GSK3. (A) Keap1 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. Keap1, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (B) Quantification analysis of Keap1, Nrf2, HO1, and NQO1 levels in Keap1 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, * p < 0.05 vs. DMSO. Error bars represent the standard deviations of the results from three independent experiments. (C) GSK3 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. GSK3, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (D) Quantification analysis of GSK3, Nrf2, HO1, and NQO1 levels in GSK3 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, * p < 0.05, ** p < 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. (E) Keap1 and GSK3 knockdown decreased the antioxidant effect of DEA (1) in LO2 cells. GSK3, Nrf2, HO1, and NQO1 were blotted to control for total protein levels. (F) Quantification analysis of GSK3, Keap1, Nrf2, HO1, and NQO1 levels in Keap1 and GSK3 knockdown cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01 vs. control group, ** p < 0.01 vs. DMSO group. Error bars represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA Keap1: SiKeap1; SiRNA GSK3: SiGSK3; SiGSK3 + SiKeap1: SiRNA GSK3 + SiRNA Keap1.',\n",
       "  'llm_description': {'summary': 'Compound DEA (1) regulates antioxidant effects by targeting Keap1 and GSK3, as evidenced by reduced expression of antioxidant proteins in cells with knockdown of either Keap1 or GSK3.',\n",
       "   'key_terms': ['Compound DEA (1)',\n",
       "    'Keap1',\n",
       "    'GSK3',\n",
       "    'Antioxidant proteins',\n",
       "    'Antioxidant effect'],\n",
       "   'caveats': []}},\n",
       " {'label': 'Figure 4',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g004.png',\n",
       "  'caption': 'Compound DEA (1) increases Nrf2 translocation into the nucleus. (A) LO2 cells were treated with 10 or 30 M compound DEA (1) for 8 h and then analyzed by WB to detect the level of Nrf2 in the nucleus and cytoplasm. actin and lamin B were used as internal controls. (B) GSK3 knockdown increased nuclear Nrf2 level. Keap1 and Nrf2 were blotted to control for the total protein levels. The data represent the standard deviations of the results from three independent experiments. SiRNA Control: SiCon; SiRNA GSK3: SiGSK3.',\n",
       "  'llm_description': {'summary': 'Compound DEA (1) increases Nrf2 translocation into the nucleus, while GSK3 knockdown enhances nuclear Nrf2 levels.',\n",
       "   'key_terms': ['Nrf2', 'DEA (1)', 'GSK3', 'Nuclear translocation', 'Keap1'],\n",
       "   'caveats': []}},\n",
       " {'label': 'Figure 5',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g005.png',\n",
       "  'caption': 'Compound DEA (1) induces antioxidant factor expression in the ALD cell model. (A) Effects of 10 M, 30 M DEA (1), and 30 M ML334 on Nrf1, NQO1, HO1, Keap1, and Nrf2 protein levels in LO2 cells after 8 h treatment. (BD) Quantification analysis of Nrf1 (B), NQO1 (C), and HO1 (D) in WB. (EI) Effects of DEA (1) and ML334 on (E) SOD2, (F) NQO1, (G) HO1, (H) Nrf1, (I) PGC1 mRNA levels in 0.3% EtOHinduced LO2 cells. Data are represented as mean  SD. * p < 0.05, *** p < 0.001 vs. control group. The data represent the standard deviations of the results from three independent experiments. EtOH: Ethyl alcohol. Nrf1: Nuclear respiratory factor1; NQO1: NAD(P)H dehydrogenase [quinone] 1; PGC1: Peroxisome proliferatoractivated receptor  coactivator 1; SOD 2: Superoxide dismutase 2.',\n",
       "  'llm_description': {'summary': 'Compound DEA (1) and ML334 induce antioxidant factor expression in ALD cells, enhancing Nrf1, NQO1, HO-1, Keap1, and Nrf2 protein levels and mRNA expression.',\n",
       "   'key_terms': ['antioxidant factor',\n",
       "    'Nrf1',\n",
       "    'NQO1',\n",
       "    'HO-1',\n",
       "    'Keap1',\n",
       "    'Nrf2',\n",
       "    'SOD2',\n",
       "    'PGC-1',\n",
       "    'ALD cells',\n",
       "    'DEA (1)',\n",
       "    'ML334'],\n",
       "   'caveats': []}},\n",
       " {'label': 'Figure 6',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g006.png',\n",
       "  'caption': 'Compound DEA (1) protects mitochondrial dysfunction from 0.3% EtOHinduced injury. (A) 0.3% EtOHinduced LO2 cells were treated with 30 M ML334, 10 or 30 M compound DEA (1) for 8 h. Intracellular ROS level were detected by DCFHDA probe and measured using the confocal laser scanning microscopy. (B) EtOHinduced LO2 cells were treated with 10 M ML334, 10 or 30 M compound DEA (1), DMSO for 8 h, and MPTP function was detected by fluorescence spectroscopy. (C) Quantification analysis of fluorescence in MPTP assay. (D) ATP level were detected using an ATP assay kit in 0.3% EtOHinduced LO2 cells. (E) Antioxidant activity of DEA (1) was detected in 0.3% EtOHinduced LO2 cells. Data are represented as mean  SD. # p < 0.05, ## p < 0.01, vs. control group. The data represent the standard deviations of the results from three independent experiments. ATP: Adenosine triphosphate; DCFHDA: Dichlorodihydrofluorescein diacetate; MPTP: Mitochondrial Permeability Transition Pore Assay; ROS: Reactive oxygen species.',\n",
       "  'llm_description': {'summary': 'Compound DEA (1) protects mitochondria from 0.3% EtOH-induced injury by reducing ROS levels, enhancing ATP levels, and increasing antioxidant activity, suggesting its potential as a mitochondrial protective agent.',\n",
       "   'key_terms': ['mitochondrial dysfunction',\n",
       "    'ROS',\n",
       "    'ATP',\n",
       "    'antioxidant activity'],\n",
       "   'caveats': []}},\n",
       " {'label': 'Figure 7',\n",
       "  'image_path': 'D:\\\\Users\\\\User\\\\VNPRagProekt\\\\figures\\\\PMC9866905_pharmaceuticals-16-00014-g007.png',\n",
       "  'caption': 'Hepatoprotective mechanism of the dual inhibitor DEA (1) against EtOHinduced cell injury.',\n",
       "  'llm_description': {'summary': 'The dual inhibitor DEA (1) protects hepatocytes from EtOH-induced injury by inhibiting the degradation of MPTP and enhancing antioxidant activity.',\n",
       "   'key_terms': ['DEA (1)', 'EtOH', 'MPTP', 'Antioxidant activity'],\n",
       "   'caveats': []}}]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "figure_descriptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0baaaac-09e8-4da0-8514-6f442055fe2b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "4fd0c1fc-83a0-4881-894f-68503e9cb90b",
   "metadata": {},
   "source": [
    "# Spojutanje na dictionaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "b5cc58d3-5d4a-4e9e-9158-784733f55452",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "desc_by_label = {d[\"label\"]: d for d in figure_descriptions}\n",
    "\n",
    "\n",
    "merged = {}\n",
    "\n",
    "for entry in sections:\n",
    "    title = entry.get(\"title\")\n",
    "    if title and title.startswith(\"Figure \"):\n",
    "        desc = desc_by_label.get(title)\n",
    "        if desc:\n",
    "            merged[title] = {\n",
    "                **entry, \n",
    "                \"label\": desc[\"label\"],\n",
    "                \"image_path\": desc[\"image_path\"],\n",
    "                \"caption\": desc[\"caption\"],\n",
    "                \"text\": desc[\"llm_description\"][\"summary\"], \n",
    "            }\n",
    "        else:\n",
    "            merged[title] = {**entry} \n",
    "    else:\n",
    "        merged[entry.get(\"title\", entry.get(\"id\", \"\"))] = {**entry}\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "ca7e6857-3404-4589-906d-0087e0237490",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['Abstract', '1. Introduction', '2.1. DEA (1) Inhibits the Keap1Nrf2 PPI and GSK3 Activity', '2.2. DEA (1) Exhibits Low Cytotoxicity In Cellulo', '2.3. DEA (1) as a Dual Inhibitor of Keap1 and GSK3', '2.4. DEA (1) Targets Keap1 and GSK3 to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins', '2.5. DEA (1) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells', '2.6. DEA (1) Repairs the EtOH Induced Mitochondrial Dysfunction and Improves the Antioxidant Activity in LO2 Cells', '3. Discussion', '4.1. Cell Lines and Culture', '4.2. Chemical', '4.3. Fluorescence Polarization Assay', '4.4. Western Blot (WB) and CoImmunoprecipitation (CoIP)', '4.5. Real-Time Quantitative Polymerase Chain Reaction (RTqPCR)', '4.6. MTT Assay', '4.7. Detection of ROS and Antioxidant Activity', '4.8. Nuclear and Cytoplasmic Extraction', '4.9. GSK3 Kinase Assay', '4.10. Dual Luciferase Reporter Gene Assay', '4.11. Cellular Thermal Shift Assay (CETSA)', '4.12. Mitochondrial Permeability Transition Pore (MPTP) Assay', '4.13. SiRNA Gene Knockdown', '4.14. Statistical Analysis', '5. Conclusions', 'Figure 1', 'Figure 2', 'Figure 3', 'Figure 4', 'Figure 5', 'Figure 6', 'Figure 7'])"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d9ddf94e-6f63-4416-87b7-41591c5119ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "71a34a21-20ee-4141-8550-1f82518f0db0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "============================================================\n",
      "TITLE: Abstract\n",
      "------------------------------------------------------------\n",
      "Alcoholic liver disease (ALD) is a complicated disease which can lead to hepatocellular carcinoma; however, there is a lack of satisfactory therapeutics. Dehydroeburicoic acid (DEA) (1), a triterpenoid isolated from Antrodia cinnamomea, has been reported to act against ALD, but its mechanisms of action are still not clear. In this study, we report for the first time the use of DEA (1) as a dual inhibitor of the Keap1Nrf2 proteinprotein interaction (PPI) and GSK3 in an in vitro ALD cell model. DEA (1) engages Keap1 to disrupt the Keap1Nrf2 PPI and inhibits GSK3 to restore Nrf2 activity in a Keap1-independent fashion. DEA (1) promotes Nrf2 nuclear translocation to activate downstream antioxidant genes. Importantly, DEA (1) restores the mitochondrial dysfunction induced by ethanol and generates antioxidant activity in the ALD cell model with minimal toxicity. We anticipate that DEA (1) could be a potential scaffold for the further development of clinical agents for treating ALD.\n",
      "============================================================\n",
      "TITLE: 1. Introduction\n",
      "------------------------------------------------------------\n",
      "Excessive alcohol consumption leads to oxidative stress in the liver as a result of alcohol metabolism, resulting in inflammatory damage and the injury of liver cells [1]. Almost 20% of alcoholics are eventually diagnosed with alcoholic liver disease (ALD) [2]. The most common therapeutic options for ALD can be divided into opioid receptor antagonists, supplements to modulate liver metabolism, and therapeutics that regulate alcohol metabolism [3]. However, these treatments have been associated with side effects including dizziness, drug dependence, dermatitis, vomiting, and leukopenia [4]. Thus, there are currently no highly satisfactory therapeutic options for ALD, and hence there is an unmet clinical need to develop more effective and safer drugs for patients with ALD. Increasing evidence indicates that alcohol can damage mitochondria and destroy cellular homeostasis in the liver [5,6,7]. One mechanism that cells use to defend against alcoholinduced injury is through the activation of nuclear respiratory factor-2 (Nrf-2). Nrf2 orchestrates a complex, self-protective antioxidant response involving numerous signaling axes [8]. Upregulating Nrf2 has been envisioned as an effective tool for combating alcohol-induced acute liver injury [9,10]. Normally, the level of Nrf2 in the cytosol is low due to ubiquitination and proteasomal degradation regulated by a Kelch-like ECH-associated protein (Keap1) [11,12]. However, the presence of oxidants can oxidize cysteine residues in Keap1 into disulfides or conjugate them to electrophiles, resulting in the nuclear translocation of Nrf2 and the activation of antioxidant genes [13], including heme oxygenase-1 (HO1), NAD(P)H dehydrogenase [quinone] 1 (NQO1), and the mitochondrial superoxide dismutase 2 (SOD2) [14,15]. In addition, Nrf2 activation increases the expression levels of nuclear respiratory factor-1 (Nrf-1) and peroxisome proliferator-activated receptor  coactivator 1 (PGC1), which in turn modulates the expression of mitochondrial respiratory subunits and translational components [16]. The genes activated by the antioxidant responsive element (ARE) protect cells against any damage caused by reactive oxygen species (ROS) [17]. In addition to Keap1, glycogen synthase kinase 3 (GSK3) is another negative regulator of Nrf2 that acts to downregulate the antioxidant stress response [18,19]. GSK3 promotes the nuclear exclusion and degradation of Nrf2 in a Keap1-independent fashion in stressed or injured cells with active Nrf2, thereby reducing the degree of protection conferred by Nrf2 activity [20,21]. Therefore, increasing the Nrf2 antioxidant activity via targeting Keap1 or GSK3 could be a potential approach for treating ALD [22]. Most pharmacological approaches towards Nrf2 activation are focused on the inhibition of Keap1 [23,24]. Several Keap1 inhibitors have been reported in the last decade, and the vast majority of these covalently modify cysteine residues in Keap1 [25]. Covalently targeting cysteines in Keap1 may lack selectivity due to the presence of other reactive cysteine residues in the cell, resulting in adverse side effects [26,27]. However, because cysteine residues are abundant in cells, the safety and specificity of covalent drugs are a concern [28]. Recently, proteinprotein interaction (PPI) inhibitors of Keap1 and Nrf2, developed based on the X-ray structure of Keap1, have emerged as a new class of Nrf2 activators [29,30,31]. Our group has previously reported a cyclometalated iridium(III) metal complex as a Keap1Nrf2 PPI inhibitor, which is a promising therapeutic agent for acetaminopheninduced acute liver injury [29,30]. However, no Keap1Nrf2 inhibitor has yet entered the clinic for the treatment of human diseases [32,33]. Meanwhile, although a few GSK3 inhibitors have been reported for liver injury treatment, no GSK3 inhibitor to date has made it to the market [34]. Antrodia cinnamomea, a basidiomycete that is endemic to Taiwan, is an edible fungus that is a component of many traditional herbal medicines [35,36]. Either crude extracts or molecules isolated from A. cinnamomea have shown diverse biological properties, including antioxidant, anti-tumour, and anti-inflammatory activities [37,38]. Recently, research has shown that dehydroeburicoic acid (DEA) (1) from A. cinnamomea could inhibit alcoholic fatty liver disease (AFLD) by upregulating aldehyde dehydrogenase 2 family member (ALDH2) activity [39,40]. Additionally, DEA (1) also exhibited protective effects against non-alcoholic fatty liver disease (NAFLD) through activating ALDH2 and accelerating the elimination of ROS and harmful aldehydes [40]. Therefore, DEA (1) has the potential ability to treat ALD. However, its mechanisms of hepatoprotection in ALD are still not clear. Besides DEA (1), other triterpenoids have also been studied as GSK3 inhibitors and activators of Nrf2 because of their anti-inflammatory and antioxidant activities [41,42,43,44,45]. Research has shown that singlemolecule drugs with two different biological activities may avoid some side effects and improve efficacy [46]. In this study, we discovered a potent dual Keap1Nrf2 PPI and GSK3 inhibitor, DEA (1), and investigated its underlying mechanisms of action. Through in vitro fluorescence polarization (FP) screening, we identified DEA (1) as the most potent candidate for disrupting the Keap1Nrf2 PPI (EC50 = 14.1 M) from thirteen analogues isolated from A. cinnamomea. Moreover, DEA (1) inhibited GSK3 kinase activity with an EC50 value of 8.0 M. In cells, low-toxicity compound DEA (1) engaged Keap1 and GSK3 to promote Nrf2 accumulation in the nucleus, leading to the upregulation of ARE transcriptional activity and increasing the expression of downstream antioxidant factors with more potent than the reported Keap1Nrf2 PPI inhibitor ML334 in ALD model cells.\n",
      "============================================================\n",
      "TITLE: 2.1. DEA (1) Inhibits the Keap1Nrf2 PPI and GSK3 Activity\n",
      "------------------------------------------------------------\n",
      "The fluorescence polarization assay is a commonly used high-throughput assay to detect PPI inhibitors in solution. We used labeled Nrf2 peptides, bearing the highaffinity ETGE motif, which is recognized by Keap1, as fluorescent tracers to monitor for inhibition of the Keap1Nrf2 interaction. ML334 is the first noncovalent small molecule inhibitor of Keap1Nrf2 interaction reported and is distinct from other Nrf2 inducers [47]. In this paper, ML334 was used as a positive control. Compounds 113 from A. cinnamomea and ML334 were screened against the Keap1Nrf2 interaction by FP (A). The preliminary screening results showed that DEA (1) was the most potent compound, which disrupted the Keap1Nrf2 interaction by 62% inhibition at 50 M (cf. ML334: 60% inhibition). The NMR spectrum of DEA (1) is presented in Supplementary Figure S1. The potency of DEA (1) was further investigated in a dose-response assay, revealing an EC50 value of 14.1  0.1 M against Keap1Nrf2 PPI (B). We also explored the inhibitory effect of DEA (1) against GSK3, which negatively regulates Nrf2 in a fashion independent of Keap1. 10 M of DEA (1) reduced GSK3 activity by 58.9%, making it more potent than ML334 at the same concentration (11.3% inhibition) (C). Moreover, a dose-response assay revealed an EC50 value of 8.0  0.7 M against GSK3 activity, indicating that DEA (1) is an effective GSK3 inhibitor (D). In the nucleus, Nrf2 binds to the ARE to activate antioxidant gene transcription [14]. Thus, we tested the transcriptional activity of ARE after treatment of LO2 cells with DEA (1) and ML334 (10 M) using the dual luciferase assay kit (E). The activity of ARE was significantly increased by 54% in the treatment of DEA (1) compared to 27% in the ML334 group. The results showed that DEA (1) notably activates the transcriptional activity of ARE compared to ML334 in cellulo.\n",
      "============================================================\n",
      "TITLE: 2.2. DEA (1) Exhibits Low Cytotoxicity In Cellulo\n",
      "------------------------------------------------------------\n",
      "The cytotoxicity of compound DEA (1) was detected in both human normal cell lines LO2 and HEK 293T, and the hepatocellular carcinoma cell line HepG2. After treatment with different concentrations of DEA (1), no significant cytotoxicity was observed (FH). These data indicate that DEA (1) could be potentially safe for treating ALD in vivo.\n",
      "============================================================\n",
      "TITLE: 2.3. DEA (1) as a Dual Inhibitor of Keap1 and GSK3\n",
      "------------------------------------------------------------\n",
      "The ability of DEA (1) to disrupt the Keap1Nrf2 binding in cellulo was monitored through a CoIP experiment using human normal liver cell line LO2 cells (A). After treatment with DEA (1) (10 M) or ML334 (10 M) for 8 h, there was a 45% reduction of Keap1 co-precipitated with Nrf2 for DEA (1) as compared to 12% inhibition for ML334, indicating that DEA (1) could more effectively inhibit the Keap1Nrf2 interaction compared to ML334 in living cells (B). We investigated the ability of DEA (1) to target Keap1, Nrf2, and GSK3 in the cellular environment using CETSA. LO2 cell lysates were incubated with 10 M of DEA (1) at room temperature for 30 min. Then, aliquots were heated individually at different temperatures and the protein in the soluble fraction was quantified by WB. Compound DEA (1) significantly stabilized Keap1 (Tm: ca. 4.5 C) and GSK3 (Tm: ca. 3.9 C), while having no observable effect on the thermal stability of Nrf2 and actin (C,D). This result suggests that DEA (1) can bind with Keap1 and GSK3, even within the complicated environment of cell lysates.\n",
      "============================================================\n",
      "TITLE: 2.4. DEA (1) Targets Keap1 and GSK3 to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins\n",
      "------------------------------------------------------------\n",
      "To verify whether Keap1 or GSK3 are direct targets of DEA (1), knockdown assays were performed. NQO1 and HO1 are target antioxidant genes of Nrf2 that increase cytoprotection against oxidative stress [48] As expected, Keap1 knockdown in LO2 cells induced a noticeable rise of the expression levels of both Nrf2 and its gene targets, NQO1 and HO1 (A,B). However, treating Keap1 knockdown cells with DEA (1) led to no further rises in the NQO1 and HO1 level, in contrast to the control cells, where DEA (1) treatment produced noticeable increases of NQO1 and HO1. These results suggest that DEA (1) activates NQO1 and HO1 through targeting Keap1. Similarly, GSK3 knockdown induced Nrf2 and downstream antioxidant protein expression but without influencing Keap1 protein level (C,D). In addition, DEA (1) had a diminished effect on NQO1 and HO1 level in GSK3 knockdown cells as compared to the control cells. Finally, we knocked down GSK3 and Keap1 at the same time to verify the dual binding mechanism of DEA (1) on GSK3 and Keap1 (E,F). The results showed that treating the GSK3 and Keap1 knockdown cells with DEA (1) did not cause a further significant induction of NQO1 and HO1 expression. Evidence for the hypothesis that DEA (1) acts through targeting both Keap1 and GSK3 comes from considering the Nrf2 level. In either the Keap1 knockdown or GSK3 knockdown cells, DEA (1) treatment resulted in slight increases of Nrf2 expression relative to DMSO. However, in the double Keap1 and GSK3 knockdown cells, treatment with DEA (1) did not lead to increases of Nrf2 expression. These results suggest that DEA (1) treatment mimicked the effects of double knockdown in increasing Nrf2 expression. Therefore, DEA (1) exerts its antioxidant effects by targeting both Keap1 and GSK3. Inhibition of the Keap1Nrf2 interaction is known to increase Nrf2 accumulation in the nucleus [47]. Meanwhile, GSK3 is also a negative regulator of Nrf2 accumulation by acting independently of Keap1 [49]. Therefore, we evaluated the Nrf2 level in the cytoplasm and nucleus of LO2 cells via WB after the treatment of cells with DEA (1) (10 M) for 8 h. As shown in A, DEA (1) significantly increased Nrf2 translocation into the nucleus in a dose-dependent manner. GSK3 knockdown also phenocopied compound DEA (1) treatment at inducing Nrf2 nuclear accumulation without influencing Keap1 level (B). In summary, DEA (1) disrupts the Keap1Nrf2 PPI and inhibits GSK3 to increase the translocation of liberated Nrf2 into the nucleus, thereby presumably increasing ARE transcriptional activity.\n",
      "============================================================\n",
      "TITLE: 2.5. DEA (1) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells\n",
      "------------------------------------------------------------\n",
      "Nuclear Nrf2 heterodimerizes with musculoaponeurotic fibrosarcoma (Maf) to recognize ARE and promote the transcription of antioxidant genes, including Nrf1, HO1, and NQO1 [14]. Therefore, we evaluated the ability of DEA (1) to increase Nrf1, NQO1, and HO1 levels in ALD model cells, which were prepared by treating LO2 cells with 0.3% EtOH for 8 h. 0.3% EtOH was reported to induce oxidative stress in living cells [50,51,52]. In addition, a preliminary time-dependent assay was performed, which identified an optimal incubation time length of 8 h (Supplementary Figure S2). As shown in A, DEA (1) (30 M) induced Nrf1 (B), NQO1 (C) and HO1 (D) protein levels by 1.2, 3.8 and 1.8-fold, respectively, in ALD model cells, making it more potent than ML334 (30 M), which increased Nrf1, NQO1, and HO1 protein levels by 1.0, 1.1, and 1.5-fold, respectively. Nrf2 directly controls the expression of the transcriptional factor PGC1, a regulator of mitochondrial biogenesis, while it also directly regulates the expression of mitochondrial antioxidant enzymes, such as superoxide dismutase 2 (SOD2) [53,54]. Hence, the effect of DEA (1) on mRNA levels of SOD2, HO1, NQO1, Nrf1, and PGC1 in 0.3% EtOHtreated LO2 cells was also evaluated. 30 M of DEA (1) rescued SOD2 mRNA level in ALD model cells by 2-fold, compared to 1.5-fold with ML334 (E). DEA (1) (30 M) also increased NQO1 (F) and HO1 (G) mRNA levels by 2.0 and 2.5-fold, respectively, making it more potent than ML334 which increased NQO1 and HO1 mRNA levels by 1.8 and 1.4-fold, respectively. In addition, the mRNA levels of Nrf1 and PGC1 were also increased 2-fold by compound DEA (1) (30 M) (H,I). The results also show a slight increase of NQO1, HO1, and PGC1 levels by 0.3% EtOH treatment alone, which has been previously observed in other studies [55]. However, the subsequent increases of antioxidant genes were much greater upon further treatment of DEA (1). Taken together, these results indicate that DEA (1) can activate Nrf2 downstream antioxidant genes in EtOHtreated human liver cells, a cellular model of ALD.\n",
      "============================================================\n",
      "TITLE: 2.6. DEA (1) Repairs the EtOH Induced Mitochondrial Dysfunction and Improves the Antioxidant Activity in LO2 Cells\n",
      "------------------------------------------------------------\n",
      "Inflammation and oxidative stress increase ROS production, and the activated Nrf2 was able to maintain low intracellular ROS level to protect the cell from oxidant injury [17]. Moreover, growing evidence shows that mitochondrial and oxidant injury induced by ethanol consumption play key roles in alcoholinduced liver injury [56]. Alcoholinduced ROS causes mitochondrial membrane depolarization and mitochondrial permeability transition (MPT), leading to hepatic apoptosis and necrosis [54]. Therefore, we detected the effect of DEA (1) and ML334 on ROS level in 0.3% EtOHtreated LO2 cells (A). The results indicated that both 1 and ML334 could significantly decrease ROS level induced by EtOH. The MPTP assay showed that DEA (1) (30 M) can restore MPTP function (40%) with a higher activity as compared to ML334 (34%) (B,C). ATP is an indicator of mitochondrial function and ATP level are significantly reduced in ethanolfed Nrf2/ mice [57]. DEA (1) (30 M) rescued 50% of ATP in LO2 cells after EtOH treatment, making it more potent than ML334, which rescued 20% of the ATP level under the same conditions (D). Finally, DEA (1) (30 M) induced a higher increase of antioxidant activity (4-fold) compared to ML334 (2-fold) under the same conditions (E).\n",
      "============================================================\n",
      "TITLE: 3. Discussion\n",
      "------------------------------------------------------------\n",
      "ALD is a serious chronic liver disease caused by oxidative stress and alcohol metabolism, which leads to massive global deaths [1]. Unfortunately, until now, there are no effective therapeutic drugs approved by the FDA for the treatment of ALD [58]. The pathogenesis of ALD is poorly characterized and research on the mechanism is still not clear. A number of reports have shown that the Keap1Nrf2 pathway plays an important role in activating ARE antioxidant signaling [14]. Nrf2 is regulated via Keap1-independent pathways, including mitogen-activated protein kinaseErk and PI3KAkt pathways [59]. Among these regulatory pathways, GSK3 has emerged as a convergent point [60,61]. Because Nrf2 degradation and nuclear exclusion are regulated by GSK3, it is critical in switching off the selfprotective antioxidant stress response after injury [59,62]. Hence, targeting Keap1 and GSK3 to activate Nrf2 are potential therapeutic strategies for ALD. Natural products provide diverse scaffolds with high bioactivity and low toxicity, and could be developed as promising candidates in drug development [63]. A. cinnamomea is a traditional Chinese herb that is a component of various medicines [64]. Triterpenoids are the major constituent in A. cinnamomea, and several studies have demonstrated their antioxidative and hepatoprotective effects [65,66]. However, the hepatoprotective mechanism of these compounds has not been extensively explored. DEA (1) has previously shown promising hepatoprotective activity in some studies [67], including in mouse models [39,67,68,69]. In recent years, many Keap1Nrf2 PPI inhibitors and GSK3 inhibitors have been reported for various indications, such as Alzheimers disease, cancers, Parkinsons disease, kidney diseases, Acetaminophen (APAP)induced liver injury, and ferroptosis [30,70,71,72,73,74]. However, few have been studied in the context of ALD. In our present study, we discovered DEA (1) as a potent Keap1Nrf2 PPI inhibitor from A. cinnamomea extracts. We explored the detailed mechanism of hepatoprotective action of DEA (1) in ALD, which acted via targeting both the Keap1Nrf2 PPI and GSK3 (). To our best knowledge, the dual inhibition mechanism of DEA (1) in ALD has not been reported before. In in vitro assays, DEA (1) inhibited the Keap1Nrf2 PPI (EC50 = 14.1 M) as well as GSK3 kinase activity (EC50 = 8.0  0.7 M) with greater potency than the known Keap1Nrf2 inhibitor, ML334. Cellular experiments confirmed the ability of DEA (1) to engage Keap1 and GSK3 in cellulo to activate Nrf2 and stimulate the expression of AREcontrolled genes. The specificity of DEA (1) was further confirmed using Keap1 and GSK3 knockdown experiments. DEA (1) treatment phenocopied dual knockdown of Keap1 and GSK3, suggesting that DEA (1) acted simultaneously through both pathways to exert its biological effects. Moreover, DEA (1) increased Nrf2 nuclear accumulation, mimicking GSK3 knockdown, without influencing Keap1 level. Finally, we showed that DEA (1) could restore MPTP function and reduce ROS level induced by EtOH in ALD model cells. DEA (1) displayed low cytotoxicity in both normal and cancerous liver cells, indicating its potential safety for treating ALD in vivo. In summary, DEA (1) is a promising scaffold for the further development of dual inhibitors of the Keap1Nrf2 PPI and GSK3 as therapeutic agents against ALD. The structural modification of compound DEA (1) is in progress and animal studies will be performed in due course.\n",
      "============================================================\n",
      "TITLE: 4.1. Cell Lines and Culture\n",
      "------------------------------------------------------------\n",
      "The human normal liver cell line (LO2), human embryonic kidney 293 T cell line, and HepG2 cell line were purchased from American Type Culture Collection (Manassas, VA, USA). Cells were cultured in Dulbeccos modified Eagles medium (DMEM), with 10% FBS, 100 units/mL penicillin, 100 g/mL streptomycin. Fetal bovine serum (FBS), DMEM, penicillin, and streptomycin were purchased from Gibco BRL (Gaithersburg, MD, USA).\n",
      "============================================================\n",
      "TITLE: 4.2. Chemical\n",
      "------------------------------------------------------------\n",
      "Compounds 113 (purity > 98% by HPLC analysis) were isolated from A. cinnamomea in a previous study [75]. New 1H and 13C nuclear magnetic resonance (NMR) were recorded for DEA (1) to verify its identity for this study (Supplementary Figure S1). Spectral data were consistent with our previous report [75].\n",
      "============================================================\n",
      "TITLE: 4.3. Fluorescence Polarization Assay\n",
      "------------------------------------------------------------\n",
      "The ability of compounds 113 and ML334 to inhibit Keap1Nrf2 peptide binding was evaluated using a FP assay according to the manufacturers instruction (BPS Bioscience, San Diego, CA, USA). ML334 was a noncovalent small molecule inhibitor of the Keap1Nrf2 and purchased from MedChemExpress (Princeton, NJ, USA). The tested compounds were dissolved in the provided buffer at the indicated concentrations. Then, the required volumes of Nrf2 peptide, Bovine Serum Albumin (BSA), Keap1 protein, and assay buffer were added to each well, followed by incubation for 30 min and the fluorescent polarization value was recorded.\n",
      "============================================================\n",
      "TITLE: 4.4. Western Blot (WB) and CoImmunoprecipitation (CoIP)\n",
      "------------------------------------------------------------\n",
      "The protocol was reported in a previous study [76]. In brief, LO2 cells were seeded at a density of 2  106 cells in a sixwell plate. LO2 lysate is harvested after incubation with 10 M of compound DEA (1) or ML334 for 8 h. A 10% SDSPAGE gel was used to separate the LO2 cell lysates, which were transferred to a PVDF membrane (BioRad, Hercules, CA, USA). The membranes were incubated with antibodies antiKeap1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 8047S), Nrf2 (1:1000, Proteintech, Wuhan, China; Cat# 16396-1-AP), GSK3 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 9832S), HO1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 5853S), NQO1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 3187S), actin (1:1000, Absin, Shanghai, China; Cat#: abs137975), or anti-lamin B (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat # 12586S) overnight. The membranes were washed with wash buffer three times. A secondary antibody (1:1000) was added to the membranes. After incubated for 2 h, the proteins bands were detected using enhanced chemiluminescent Plus reagents (GE Healthcare, Boston, MA, USA) and analyzed by Image Lab. The antibodies were purchased from Abcam (Waltham, MA, USA). CoIP between Keap1 and Nrf2 was performed following the protocol from Life Technologies. Briefly, lysates from 2  106 LO2 cells were incubated with antiNrf2 or Rabbit mAb IgG (Abcam, Waltham, MA, USA; Cat#: ab205718) overnight. Then, samples were incubated with 10 L Dynabeads (Life technologies, Foster, CA, USA) for 6 h. The Dynabeads were washed and analyzed by WB with the Keap1 antibody.\n",
      "============================================================\n",
      "TITLE: 4.5. Real-Time Quantitative Polymerase Chain Reaction (RTqPCR)\n",
      "------------------------------------------------------------\n",
      "LO2 cells were incubated with 10 M or 30 M compound DEA (1) for 8 h and the total RNA was extracted using the NucleoSpin RNA Plus kit (Takara, Tokyo, Japan). We checked the RNA quality during the RTqPCR operation and ensured that all the A260/280 value of samples were around 2, which is consistent with the literature [77]. We also performed an RT negative control and found no PCR amplification. Then, cDNA was synthesized using the PrimeScript RT Reagent Kit (Takara, Tokyo, Japan). The RTqPCR conditions were as previously reported, with minor modifications [77]. DNA denaturation temperature: 95 C (30 s); annealing temperature: 58 C (30 s); extension temperature: 72 C (45 s), number of cycles: 40. RTqPCR was performed and analyzed as described previously [78]. The related primers are listed in Supplementary Table S1.\n",
      "============================================================\n",
      "TITLE: 4.6. MTT Assay\n",
      "------------------------------------------------------------\n",
      "The LO2, HUVEC and HEK 293 T cell lines were seeded at 1  104 cells per well in 96well plates and incubated with DMEM overnight at 37 C in a humidified CO2 incubator to ensure attachment. The medium was then replaced with a medium that contained the different concentrations of compound for 48 h at 37 C. The final concentration of DMSO was 0.1% or lower in all cases (including controls). The medium with 100 L of MTT (3-(4,5-dimethylthiazol-2-yl)-2,5-tetrazolium bromide) reagent (1 mg/mL) was then replaced. After 4 h incubation, it was replaced with 100 L of DMSO and measured using a microplate reader at 570 nm.\n",
      "============================================================\n",
      "TITLE: 4.7. Detection of ROS and Antioxidant Activity\n",
      "------------------------------------------------------------\n",
      "LO2 cells at a density of 2  106 cells/well in a six-well plate were treated with different concentrations of DEA (1) or 10 M ML334 for 8 h. ROS level and antioxidant activity were measured using the Reactive Oxygen Species Assay kit and the Total Antioxidant Capacity Assay Kit with FRAP, respectively, according to the manufacturers instructions (Beyotime, Shanghai, China).\n",
      "============================================================\n",
      "TITLE: 4.8. Nuclear and Cytoplasmic Extraction\n",
      "------------------------------------------------------------\n",
      "LO2 cells were incubated with 10 M or 30 M of DEA (1) for 8 h. The nuclear and cytoplasmic extraction followed the protocol from the manufacturer (Epigentek, New York, NY, USA, EpiQuik Nuclear Extraction Kit I, Catalog # OP-0002), Then, protein expression was analyzed by WB with the indicated antibodies.\n",
      "============================================================\n",
      "TITLE: 4.9. GSK3 Kinase Assay\n",
      "------------------------------------------------------------\n",
      "The ability of DEA (1) and ML334 to inhibit GSK3 activity was evaluated using luminescent kinase assay according to the manufacturers instruction (Promega Corporation, Madison, WI, USA). Briefly, the 384 well white plate was mixed with 1 L of inhibitor, 2 L of enzyme, and 2 L of substrate/ATP mix and incubated at room temperature for 60 min. After 60 min, 5 L of ADPGlo reagent was added and incubated at room temperature for 40 min. Then 10 L of kinase detection reagent was added, incubated at room temperature for 30 min and the luminescence value was recorded (integration time 0.51 s).\n",
      "============================================================\n",
      "TITLE: 4.10. Dual Luciferase Reporter Gene Assay\n",
      "------------------------------------------------------------\n",
      "LO2 cells were seeded in a sixwell plate with 80% confluence in a DMEM medium for 24 h. The AREluciferase vector or a control plasmid with TurboFect transfection reagent (Thermo Fisher Scientific, Waltham, MA, USA) were transfected into LO2 cells according to the manufacturers instruction, and cells were incubated for 48 h. Then, LO2 cells were incubated with 10 M compound DEA (1), 10 M ML334 or DMSO for 4 h. The ARE activity was detected through the manufacturers instruction (Beyotime Dual Luciferase Reporter Gene Assay Kit, Shanghai, China).\n",
      "============================================================\n",
      "TITLE: 4.11. Cellular Thermal Shift Assay (CETSA)\n",
      "------------------------------------------------------------\n",
      "LO2 cell lysates were incubated with 10 M compound DEA (1) and DMSO at room temperature for 30 min. The lysates are divided and heated individually at the indicated temperature. The supernatants of the heated lysates were collected after centrifugation, and then analyzed by WB with the Keap1, Nrf2, and actin antibodies.\n",
      "============================================================\n",
      "TITLE: 4.12. Mitochondrial Permeability Transition Pore (MPTP) Assay\n",
      "------------------------------------------------------------\n",
      "MPTP kit was used for evaluation of mitochondrial permeability transition according to the manufacturers instructions (Beyotime, Shanghai, China). A density of 1  104 cells in a sixwell plate, EtOHinduced LO2 cells were treated with 10 M of ML334, 10 or 30 M compound DEA (1), DMSO for 8 h, and then pretreated with calcein or calcein and CoCl2. Cells were stained by calcein acetoxymethyl ester (Calcein AM) and MPT function was detected by fluorescence spectroscopy.\n",
      "============================================================\n",
      "TITLE: 4.13. SiRNA Gene Knockdown\n",
      "------------------------------------------------------------\n",
      "The sequences of Keap1 siRNA are 5-GGCCUUUGGCAUCAUGAACTT-3 (sense) and 5-GUUCAUGAUGCCAAAGGCCTG-3 (antisense) [79]. GSK3 siRNA is 5-GCAUUUAUCGUUAACCUAA-3 (sense) and 5-UUAGGUUAACGA UAAAUGC-3 (antisense) [80]. Keap1 and GSK3 siRNA was transfected with Lipo3000 reagent (Thermo Fisher Scientific, Waltham, MA, USA) into 1  106 LO2 cells following the manufacturers instruction. After incubation for 48 h, Keap1 and GSK3 protein expression and functional experiments were performed.\n",
      "============================================================\n",
      "TITLE: 4.14. Statistical Analysis\n",
      "------------------------------------------------------------\n",
      "Statistical significance was determined using the Students t-test for experiments comparing two groups. The data is homogeneous and follows a normal distribution before proceeding to the statistical analysis. Comparisons among groups were analysed using analysis of variance (ANOVA). All statistical tests were done using the GraphPad Prism version 8.0 software (GraphPad Software Inc. San Diego, CA, USA). All values were expressed as the mean  standard deviation (SD). p < 0.05 means statistical significance.\n",
      "============================================================\n",
      "TITLE: 5. Conclusions\n",
      "------------------------------------------------------------\n",
      "Recent studies on ALD are scarce compared to research on other liver diseases, and there are still no highly satisfactory therapeutic options for ALD. To our knowledge, no dual inhibitor of the Keap1Nrf2 PPI and GSK3 to treat alcohol-induced liver injury has been reported. This paper has revealed that DEA (1), one of the active triterpenes isolated from A. cinnamomea, could be a potential lead compound for treating ALD. Compound DEA (1) inhibited the Keap1Nrf2 interaction and GSK3 activity both in vitro and in cellulo, and displayed low cytotoxicity at the concentrations required for inducing antioxidant activity. Moreover, DEA (1) could engage Keap1 and GSK3 in liver cells and increase Nrf2 nucleus translocation, thereby activating ARE transcriptional activity and upregulating antioxidant genes and mitochondrial biogenesis regulators. Significantly, compound DEA (1) displayed more potent hepatoprotective activity when compared with the classic Keap1Nrf2 PPI inhibitor ML334. Therefore, the structure of DEA (1) may be optimized to expand the diversity, improve the activity, and be used in animal research. To our knowledge, this paper firstly reports DEA (1) acts as the dual inhibitor of Keap1Nrf2 PPI and GSK3 with the potential for protecting against ALD. We anticipate that natural product DEA (1) can be considered as a potential scaffold for the development of clinical agents for treating ALD.\n",
      "============================================================\n",
      "TITLE: Figure 1\n",
      "------------------------------------------------------------\n",
      "Compound DEA (1) potently inhibits Keap1-Nrf2 binding and GSK3 activity, enhancing ARE activity in cells, while showing low cytotoxicity.\n",
      "============================================================\n",
      "TITLE: Figure 2\n",
      "------------------------------------------------------------\n",
      "Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3, enhancing its stability and activity in LO2 cells.\n",
      "============================================================\n",
      "TITLE: Figure 3\n",
      "------------------------------------------------------------\n",
      "Compound DEA (1) regulates antioxidant effects by targeting Keap1 and GSK3, as evidenced by reduced expression of antioxidant proteins in cells with knockdown of either Keap1 or GSK3.\n",
      "============================================================\n",
      "TITLE: Figure 4\n",
      "------------------------------------------------------------\n",
      "Compound DEA (1) increases Nrf2 translocation into the nucleus, while GSK3 knockdown enhances nuclear Nrf2 levels.\n",
      "============================================================\n",
      "TITLE: Figure 5\n",
      "------------------------------------------------------------\n",
      "Compound DEA (1) and ML334 induce antioxidant factor expression in ALD cells, enhancing Nrf1, NQO1, HO-1, Keap1, and Nrf2 protein levels and mRNA expression.\n",
      "============================================================\n",
      "TITLE: Figure 6\n",
      "------------------------------------------------------------\n",
      "Compound DEA (1) protects mitochondria from 0.3% EtOH-induced injury by reducing ROS levels, enhancing ATP levels, and increasing antioxidant activity, suggesting its potential as a mitochondrial protective agent.\n",
      "============================================================\n",
      "TITLE: Figure 7\n",
      "------------------------------------------------------------\n",
      "The dual inhibitor DEA (1) protects hepatocytes from EtOH-induced injury by inhibiting the degradation of MPTP and enhancing antioxidant activity.\n"
     ]
    }
   ],
   "source": [
    "for title, entry in merged.items():\n",
    "    print(\"=\" * 60)\n",
    "    print(f\"TITLE: {title}\")\n",
    "    print(\"-\" * 60)\n",
    "    \n",
    "    print(entry.get(\"text\"))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c402ac94-c2b6-482f-a02e-a14766634b1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7907f6ab-15be-4eea-b9aa-9b50207df496",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "63f5004c-e665-404c-b8f0-59f1c79f4cac",
   "metadata": {},
   "source": [
    "# Semantic Chunking"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "657badc2-9a48-4cc6-8953-e86bd2f52479",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\User\\AppData\\Local\\Temp\\ipykernel_19152\\2763678214.py:1: LangChainDeprecationWarning: The class `HuggingFaceEmbeddings` was deprecated in LangChain 0.2.2 and will be removed in 1.0. An updated version of the class exists in the :class:`~langchain-huggingface package and should be used instead. To use it run `pip install -U :class:`~langchain-huggingface` and import as `from :class:`~langchain_huggingface import HuggingFaceEmbeddings``.\n",
      "  emb = HuggingFaceEmbeddings(model_name=\"NeuML/pubmedbert-base-embeddings\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "============================================================\n",
      "Chunk 1 | section: Abstract | sub_idx: 0\n",
      "Token count: 116\n",
      "Character count: 500\n",
      "Alcoholic liver disease (ALD) is a complicated disease which can lead to hepatocellular carcinoma; however, there is a lack of satisfactory therapeutics. Dehydr\n",
      "============================================================\n",
      "Chunk 2 | section: Abstract | sub_idx: 1\n",
      "Token count: 93\n",
      "Character count: 494\n",
      "DEA (1) engages Keap1 to disrupt the Keap1Nrf2 PPI and inhibits GSK3 to restore Nrf2 activity in a Keap1-independent fashion. DEA (1) promotes Nrf2 nuclear tr\n",
      "============================================================\n",
      "Chunk 3 | section: 1. Introduction | sub_idx: 0\n",
      "Token count: 745\n",
      "Character count: 3761\n",
      "Excessive alcohol consumption leads to oxidative stress in the liver as a result of alcohol metabolism, resulting in inflammatory damage and the injury of liver\n",
      "============================================================\n",
      "Chunk 4 | section: 1. Introduction | sub_idx: 1\n",
      "Token count: 69\n",
      "Character count: 298\n",
      "Meanwhile, although a few GSK3 inhibitors have been reported for liver injury treatment, no GSK3 inhibitor to date has made it to the market [34]. Antrodia ci\n",
      "============================================================\n",
      "Chunk 5 | section: 1. Introduction | sub_idx: 2\n",
      "Token count: 376\n",
      "Character count: 1809\n",
      "Either crude extracts or molecules isolated from A. cinnamomea have shown diverse biological properties, including antioxidant, anti-tumour, and anti-inflammato\n",
      "============================================================\n",
      "Chunk 6 | section: 2.1. DEA (1) Inhibits the Keap1Nrf2 PPI and GSK3 Activity | sub_idx: 0\n",
      "Token count: 99\n",
      "Character count: 485\n",
      "The fluorescence polarization assay is a commonly used high-throughput assay to detect PPI inhibitors in solution. We used labeled Nrf2 peptides, bearing the hi\n",
      "============================================================\n",
      "Chunk 7 | section: 2.1. DEA (1) Inhibits the Keap1Nrf2 PPI and GSK3 Activity | sub_idx: 1\n",
      "Token count: 338\n",
      "Character count: 1394\n",
      "Compounds 113 from A. cinnamomea and ML334 were screened against the Keap1Nrf2 interaction by FP (A). The preliminary screening results showed that DEA (1) wa\n",
      "============================================================\n",
      "Chunk 8 | section: 2.2. DEA (1) Exhibits Low Cytotoxicity In Cellulo | sub_idx: 0\n",
      "Token count: 54\n",
      "Character count: 255\n",
      "The cytotoxicity of compound DEA (1) was detected in both human normal cell lines LO2 and HEK 293T, and the hepatocellular carcinoma cell line HepG2. After trea\n",
      "============================================================\n",
      "Chunk 9 | section: 2.2. DEA (1) Exhibits Low Cytotoxicity In Cellulo | sub_idx: 1\n",
      "Token count: 20\n",
      "Character count: 84\n",
      "These data indicate that DEA (1) could be potentially safe for treating ALD in vivo.\n",
      "============================================================\n",
      "Chunk 10 | section: 2.3. DEA (1) as a Dual Inhibitor of Keap1 and GSK3 | sub_idx: 0\n",
      "Token count: 200\n",
      "Character count: 841\n",
      "The ability of DEA (1) to disrupt the Keap1Nrf2 binding in cellulo was monitored through a CoIP experiment using human normal liver cell line LO2 cells (A). A\n",
      "============================================================\n",
      "Chunk 11 | section: 2.3. DEA (1) as a Dual Inhibitor of Keap1 and GSK3 | sub_idx: 1\n",
      "Token count: 66\n",
      "Character count: 244\n",
      "4.5 C) and GSK3 (Tm: ca. 3.9 C), while having no observable effect on the thermal stability of Nrf2 and actin (C,D). This result suggests that DEA (1) can\n",
      "============================================================\n",
      "Chunk 12 | section: 2.4. DEA (1) Targets Keap1 and GSK3 to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins | sub_idx: 0\n",
      "Token count: 394\n",
      "Character count: 1895\n",
      "To verify whether Keap1 or GSK3 are direct targets of DEA (1), knockdown assays were performed. NQO1 and HO1 are target antioxidant genes of Nrf2 that increas\n",
      "============================================================\n",
      "Chunk 13 | section: 2.4. DEA (1) Targets Keap1 and GSK3 to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins | sub_idx: 1\n",
      "Token count: 142\n",
      "Character count: 691\n",
      "Meanwhile, GSK3 is also a negative regulator of Nrf2 accumulation by acting independently of Keap1 [49]. Therefore, we evaluated the Nrf2 level in the cytoplas\n",
      "============================================================\n",
      "Chunk 14 | section: 2.5. DEA (1) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells | sub_idx: 0\n",
      "Token count: 501\n",
      "Character count: 1829\n",
      "Nuclear Nrf2 heterodimerizes with musculoaponeurotic fibrosarcoma (Maf) to recognize ARE and promote the transcription of antioxidant genes, including Nrf1, HO\n",
      "============================================================\n",
      "Chunk 15 | section: 2.5. DEA (1) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells | sub_idx: 1\n",
      "Token count: 53\n",
      "Character count: 266\n",
      "However, the subsequent increases of antioxidant genes were much greater upon further treatment of DEA (1). Taken together, these results indicate that DEA (1) \n",
      "============================================================\n",
      "Chunk 16 | section: 2.6. DEA (1) Repairs the EtOH Induced Mitochondrial Dysfunction and Improves the Antioxidant Activity in LO2 Cells | sub_idx: 0\n",
      "Token count: 119\n",
      "Character count: 597\n",
      "Inflammation and oxidative stress increase ROS production, and the activated Nrf2 was able to maintain low intracellular ROS level to protect the cell from oxid\n",
      "============================================================\n",
      "Chunk 17 | section: 2.6. DEA (1) Repairs the EtOH Induced Mitochondrial Dysfunction and Improves the Antioxidant Activity in LO2 Cells | sub_idx: 1\n",
      "Token count: 174\n",
      "Character count: 663\n",
      "The results indicated that both 1 and ML334 could significantly decrease ROS level induced by EtOH. The MPTP assay showed that DEA (1) (30 M) can restore MPTP \n",
      "============================================================\n",
      "Chunk 18 | section: 3. Discussion | sub_idx: 0\n",
      "Token count: 211\n",
      "Character count: 1108\n",
      "ALD is a serious chronic liver disease caused by oxidative stress and alcohol metabolism, which leads to massive global deaths [1]. Unfortunately, until now, th\n",
      "============================================================\n",
      "Chunk 19 | section: 3. Discussion | sub_idx: 1\n",
      "Token count: 69\n",
      "Character count: 338\n",
      "A. cinnamomea is a traditional Chinese herb that is a component of various medicines [64]. Triterpenoids are the major constituent in A. cinnamomea, and several\n",
      "============================================================\n",
      "Chunk 20 | section: 3. Discussion | sub_idx: 2\n",
      "Token count: 451\n",
      "Character count: 2061\n",
      "DEA (1) has previously shown promising hepatoprotective activity in some studies [67], including in mouse models [39,67,68,69]. In recent years, many Keap1Nrf2\n",
      "============================================================\n",
      "Chunk 21 | section: 4.1. Cell Lines and Culture | sub_idx: 0\n",
      "Token count: 73\n",
      "Character count: 302\n",
      "The human normal liver cell line (LO2), human embryonic kidney 293 T cell line, and HepG2 cell line were purchased from American Type Culture Collection (Manass\n",
      "============================================================\n",
      "Chunk 22 | section: 4.1. Cell Lines and Culture | sub_idx: 1\n",
      "Token count: 31\n",
      "Character count: 115\n",
      "Fetal bovine serum (FBS), DMEM, penicillin, and streptomycin were purchased from Gibco BRL (Gaithersburg, MD, USA).\n",
      "============================================================\n",
      "Chunk 23 | section: 4.2. Chemical | sub_idx: 0\n",
      "Token count: 20\n",
      "Character count: 68\n",
      "Compounds 113 (purity > 98% by HPLC analysis) were isolated from A.\n",
      "============================================================\n",
      "Chunk 24 | section: 4.2. Chemical | sub_idx: 1\n",
      "Token count: 55\n",
      "Character count: 236\n",
      "cinnamomea in a previous study [75]. New 1H and 13C nuclear magnetic resonance (NMR) were recorded for DEA (1) to verify its identity for this study (Supplement\n",
      "============================================================\n",
      "Chunk 25 | section: 4.3. Fluorescence Polarization Assay | sub_idx: 0\n",
      "Token count: 91\n",
      "Character count: 406\n",
      "The ability of compounds 113 and ML334 to inhibit Keap1Nrf2 peptide binding was evaluated using a FP assay according to the manufacturers instruction (BPS Bi\n",
      "============================================================\n",
      "Chunk 26 | section: 4.3. Fluorescence Polarization Assay | sub_idx: 1\n",
      "Token count: 44\n",
      "Character count: 215\n",
      "Then, the required volumes of Nrf2 peptide, Bovine Serum Albumin (BSA), Keap1 protein, and assay buffer were added to each well, followed by incubation for 30 m\n",
      "============================================================\n",
      "Chunk 27 | section: 4.4. Western Blot (WB) and CoImmunoprecipitation (CoIP) | sub_idx: 0\n",
      "Token count: 285\n",
      "Character count: 967\n",
      "The protocol was reported in a previous study [76]. In brief, LO2 cells were seeded at a density of 2  106 cells in a sixwell plate. LO2 lysate is harvested a\n",
      "============================================================\n",
      "Chunk 28 | section: 4.4. Western Blot (WB) and CoImmunoprecipitation (CoIP) | sub_idx: 1\n",
      "Token count: 157\n",
      "Character count: 679\n",
      "A secondary antibody (1:1000) was added to the membranes. After incubated for 2 h, the proteins bands were detected using enhanced chemiluminescent Plus reagent\n",
      "============================================================\n",
      "Chunk 29 | section: 4.5. Real-Time Quantitative Polymerase Chain Reaction (RTqPCR) | sub_idx: 0\n",
      "Token count: 81\n",
      "Character count: 328\n",
      "LO2 cells were incubated with 10 M or 30 M compound DEA (1) for 8 h and the total RNA was extracted using the NucleoSpin RNA Plus kit (Takara, Tokyo, Japan).\n",
      "============================================================\n",
      "Chunk 30 | section: 4.5. Real-Time Quantitative Polymerase Chain Reaction (RTqPCR) | sub_idx: 1\n",
      "Token count: 114\n",
      "Character count: 507\n",
      "We also performed an RT negative control and found no PCR amplification. Then, cDNA was synthesized using the PrimeScript RT Reagent Kit (Takara, Tokyo, Japan)\n",
      "============================================================\n",
      "Chunk 31 | section: 4.6. MTT Assay | sub_idx: 0\n",
      "Token count: 66\n",
      "Character count: 308\n",
      "The LO2, HUVEC and HEK 293 T cell lines were seeded at 1  104 cells per well in 96well plates and incubated with DMEM overnight at 37 C in a humidified CO2 i\n",
      "============================================================\n",
      "Chunk 32 | section: 4.6. MTT Assay | sub_idx: 1\n",
      "Token count: 85\n",
      "Character count: 314\n",
      "The final concentration of DMSO was 0.1% or lower in all cases (including controls). The medium with 100 L of MTT (3-(4,5-dimethylthiazol-2-yl)-2,5-tetrazolium\n",
      "============================================================\n",
      "Chunk 33 | section: 4.7. Detection of ROS and Antioxidant Activity | sub_idx: 0\n",
      "Token count: 82\n",
      "Character count: 379\n",
      "LO2 cells at a density of 2  106 cells/well in a six-well plate were treated with different concentrations of DEA (1) or 10 M ML334 for 8 h. ROS level and ant\n",
      "============================================================\n",
      "Chunk 34 | section: 4.8. Nuclear and Cytoplasmic Extraction | sub_idx: 0\n",
      "Token count: 73\n",
      "Character count: 308\n",
      "LO2 cells were incubated with 10 M or 30 M of DEA (1) for 8 h. The nuclear and cytoplasmic extraction followed the protocol from the manufacturer (Epigentek, \n",
      "============================================================\n",
      "Chunk 35 | section: 4.9. GSK3 Kinase Assay | sub_idx: 0\n",
      "Token count: 78\n",
      "Character count: 348\n",
      "The ability of DEA (1) and ML334 to inhibit GSK3 activity was evaluated using luminescent kinase assay according to the manufacturers instruction (Promega Cor\n",
      "============================================================\n",
      "Chunk 36 | section: 4.9. GSK3 Kinase Assay | sub_idx: 1\n",
      "Token count: 59\n",
      "Character count: 253\n",
      "After 60 min, 5 L of ADPGlo reagent was added and incubated at room temperature for 40 min. Then 10 L of kinase detection reagent was added, incubated at ro\n",
      "============================================================\n",
      "Chunk 37 | section: 4.10. Dual Luciferase Reporter Gene Assay | sub_idx: 0\n",
      "Token count: 73\n",
      "Character count: 328\n",
      "LO2 cells were seeded in a sixwell plate with 80% confluence in a DMEM medium for 24 h. The AREluciferase vector or a control plasmid with TurboFect transfect\n",
      "============================================================\n",
      "Chunk 38 | section: 4.10. Dual Luciferase Reporter Gene Assay | sub_idx: 1\n",
      "Token count: 54\n",
      "Character count: 226\n",
      "Then, LO2 cells were incubated with 10 M compound DEA (1), 10 M ML334 or DMSO for 4 h. The ARE activity was detected through the manufacturers instruction (B\n",
      "============================================================\n",
      "Chunk 39 | section: 4.11. Cellular Thermal Shift Assay (CETSA) | sub_idx: 0\n",
      "Token count: 37\n",
      "Character count: 178\n",
      "LO2 cell lysates were incubated with 10 M compound DEA (1) and DMSO at room temperature for 30 min. The lysates are divided and heated individually at the indi\n",
      "============================================================\n",
      "Chunk 40 | section: 4.11. Cellular Thermal Shift Assay (CETSA) | sub_idx: 1\n",
      "Token count: 30\n",
      "Character count: 145\n",
      "The supernatants of the heated lysates were collected after centrifugation, and then analyzed by WB with the Keap1, Nrf2, and actin antibodies.\n",
      "============================================================\n",
      "Chunk 41 | section: 4.12. Mitochondrial Permeability Transition Pore (MPTP) Assay | sub_idx: 0\n",
      "Token count: 83\n",
      "Character count: 351\n",
      "MPTP kit was used for evaluation of mitochondrial permeability transition according to the manufacturers instructions (Beyotime, Shanghai, China). A density of\n",
      "============================================================\n",
      "Chunk 42 | section: 4.12. Mitochondrial Permeability Transition Pore (MPTP) Assay | sub_idx: 1\n",
      "Token count: 24\n",
      "Character count: 122\n",
      "Cells were stained by calcein acetoxymethyl ester (Calcein AM) and MPT function was detected by fluorescence spectroscopy.\n",
      "============================================================\n",
      "Chunk 43 | section: 4.13. SiRNA Gene Knockdown | sub_idx: 0\n",
      "Token count: 128\n",
      "Character count: 386\n",
      "The sequences of Keap1 siRNA are 5-GGCCUUUGGCAUCAUGAACTT-3 (sense) and 5-GUUCAUGAUGCCAAAGGCCTG-3 (antisense) [79]. GSK3 siRNA is 5-GCAUUUAUCGUUAACCUAA-3 \n",
      "============================================================\n",
      "Chunk 44 | section: 4.13. SiRNA Gene Knockdown | sub_idx: 1\n",
      "Token count: 19\n",
      "Character count: 104\n",
      "After incubation for 48 h, Keap1 and GSK3 protein expression and functional experiments were performed.\n",
      "============================================================\n",
      "Chunk 45 | section: 4.14. Statistical Analysis | sub_idx: 0\n",
      "Token count: 76\n",
      "Character count: 407\n",
      "Statistical significance was determined using the Students t-test for experiments comparing two groups. The data is homogeneous and follows a normal distributi\n",
      "============================================================\n",
      "Chunk 46 | section: 4.14. Statistical Analysis | sub_idx: 1\n",
      "Token count: 25\n",
      "Character count: 105\n",
      "All values were expressed as the mean  standard deviation (SD). p < 0.05 means statistical significance.\n",
      "============================================================\n",
      "Chunk 47 | section: 5. Conclusions | sub_idx: 0\n",
      "Token count: 54\n",
      "Character count: 275\n",
      "Recent studies on ALD are scarce compared to research on other liver diseases, and there are still no highly satisfactory therapeutic options for ALD. To our kn\n",
      "============================================================\n",
      "Chunk 48 | section: 5. Conclusions | sub_idx: 1\n",
      "Token count: 220\n",
      "Character count: 1148\n",
      "This paper has revealed that DEA (1), one of the active triterpenes isolated from A. cinnamomea, could be a potential lead compound for treating ALD. Compound D\n",
      "============================================================\n",
      "Chunk 49 | section: Figure 1 | sub_idx: 0\n",
      "Token count: 28\n",
      "Character count: 138\n",
      "Compound DEA (1) potently inhibits Keap1-Nrf2 binding and GSK3 activity, enhancing ARE activity in cells, while showing low cytotoxicity.\n",
      "============================================================\n",
      "Chunk 50 | section: Figure 2 | sub_idx: 0\n",
      "Token count: 26\n",
      "Character count: 114\n",
      "Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3, enhancing its stability and activity in LO2 cells.\n",
      "============================================================\n",
      "Chunk 51 | section: Figure 3 | sub_idx: 0\n",
      "Token count: 34\n",
      "Character count: 185\n",
      "Compound DEA (1) regulates antioxidant effects by targeting Keap1 and GSK3, as evidenced by reduced expression of antioxidant proteins in cells with knockdown \n",
      "============================================================\n",
      "Chunk 52 | section: Figure 4 | sub_idx: 0\n",
      "Token count: 22\n",
      "Character count: 115\n",
      "Compound DEA (1) increases Nrf2 translocation into the nucleus, while GSK3 knockdown enhances nuclear Nrf2 levels.\n",
      "============================================================\n",
      "Chunk 53 | section: Figure 5 | sub_idx: 0\n",
      "Token count: 39\n",
      "Character count: 157\n",
      "Compound DEA (1) and ML334 induce antioxidant factor expression in ALD cells, enhancing Nrf1, NQO1, HO-1, Keap1, and Nrf2 protein levels and mRNA expression.\n",
      "============================================================\n",
      "Chunk 54 | section: Figure 6 | sub_idx: 0\n",
      "Token count: 41\n",
      "Character count: 213\n",
      "Compound DEA (1) protects mitochondria from 0.3% EtOH-induced injury by reducing ROS levels, enhancing ATP levels, and increasing antioxidant activity, suggesti\n",
      "============================================================\n",
      "Chunk 55 | section: Figure 7 | sub_idx: 0\n",
      "Token count: 27\n",
      "Character count: 146\n",
      "The dual inhibitor DEA (1) protects hepatocytes from EtOH-induced injury by inhibiting the degradation of MPTP and enhancing antioxidant activity.\n"
     ]
    }
   ],
   "source": [
    "emb = HuggingFaceEmbeddings(model_name=\"NeuML/pubmedbert-base-embeddings\")\n",
    "tokenizer = AutoTokenizer.from_pretrained(\"NeuML/pubmedbert-base-embeddings\")\n",
    "\n",
    "\n",
    "data = merged \n",
    "\n",
    "\n",
    "sections = []\n",
    "for title, node in data.items():\n",
    "    text = node.get(\"text\", \"\")\n",
    "   \n",
    "    sections.append(\n",
    "        Document(\n",
    "            page_content=text.strip(),\n",
    "            metadata={\n",
    "                \"section_title\": title,\n",
    "                \"section_id\": node.get(\"id\"),\n",
    "                \"section_type\": node.get(\"type\"),\n",
    "                \"is_parent\": True,      # marks the full-section parent\n",
    "            },\n",
    "        )\n",
    "    )\n",
    "\n",
    "\n",
    "text_splitter = SemanticChunker(emb)\n",
    "\n",
    "\n",
    "all_chunks = []\n",
    "for parent_doc in sections:\n",
    "   \n",
    "    sub_chunks = text_splitter.split_documents([parent_doc])\n",
    "\n",
    "    \n",
    "    for idx, ch in enumerate(sub_chunks):\n",
    "        ch.metadata.update({\n",
    "            \"section_title\": parent_doc.metadata[\"section_title\"],\n",
    "            \"section_id\": parent_doc.metadata[\"section_id\"],\n",
    "            \"section_type\": parent_doc.metadata[\"section_type\"],\n",
    "            \n",
    "            \"subchunk_index\": idx,\n",
    "           \n",
    "        })\n",
    "    all_chunks.extend(sub_chunks)\n",
    "\n",
    "\n",
    "for i, d in enumerate(all_chunks):\n",
    "    tokens = tokenizer.encode(d.page_content, truncation=False)\n",
    "    print(\"=\" * 60)\n",
    "    print(f\"Chunk {i+1} | section: {d.metadata.get('section_title')}\"\n",
    "          \n",
    "          f\" | sub_idx: {d.metadata.get('subchunk_index')}\")\n",
    "    print(f\"Token count: {len(tokens)}\")\n",
    "    print(f\"Character count: {len(d.page_content)}\")\n",
    "    \n",
    "    preview = (d.page_content[:160] + \"\") if len(d.page_content) > 160 else d.page_content\n",
    "    print(preview)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "902c78cb-617c-4d69-9e2b-cf83242a4d3b",
   "metadata": {},
   "source": [
    "# Normalizacija na chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ab93eba7-e1a8-4096-8c1e-c96bb0c0d38e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "============================================================\n",
      "Abstract | idx 0/28\n",
      "207 tokens | 995 chars\n",
      "Alcoholic liver disease (ALD) is a complicated disease which can lead to hepatocellular carcinoma; however, there is a lack of satisfactory therapeutics. Dehydroeburicoic acid (DEA) (1), a triterpenoid isolated from Antrodia cinnamomea, has been reported to act against ALD, but its mechanisms of action are still not clear. In this study, we report for the first time the use of DEA (1) as a dual inhibitor of the Keap1Nrf2 proteinprotein interaction (PPI) and GSK3 in an in vitro ALD cell model. DEA (1) engages Keap1 to disrupt the Keap1Nrf2 PPI and inhibits GSK3 to restore Nrf2 activity in a Keap1-independent fashion. DEA (1) promotes Nrf2 nuclear translocation to activate downstream antioxidant genes. Importantly, DEA (1) restores the mitochondrial dysfunction induced by ethanol and generates antioxidant activity in the ALD cell model with minimal toxicity. We anticipate that DEA (1) could be a potential scaffold for the further development of clinical agents for treating ALD.\n",
      "============================================================\n",
      "1. Introduction | idx 1/28\n",
      "356 tokens | 1783 chars\n",
      "Excessive alcohol consumption leads to oxidative stress in the liver as a result of alcohol metabolism, resulting in inflammatory damage and the injury of liver cells [1]. Almost 20% of alcoholics are eventually diagnosed with alcoholic liver disease (ALD) [2]. The most common therapeutic options for ALD can be divided into opioid receptor antagonists, supplements to modulate liver metabolism, and therapeutics that regulate alcohol metabolism [3]. However, these treatments have been associated with side effects including dizziness, drug dependence, dermatitis, vomiting, and leukopenia [4]. Thus, there are currently no highly satisfactory therapeutic options for ALD, and hence there is an unmet clinical need to develop more effective and safer drugs for patients with ALD. Increasing evidence indicates that alcohol can damage mitochondria and destroy cellular homeostasis in the liver [5,6,7]. One mechanism that cells use to defend against alcoholinduced injury is through the activation of nuclear respiratory factor-2 (Nrf-2). Nrf2 orchestrates a complex, self-protective antioxidant response involving numerous signaling axes [8]. Upregulating Nrf2 has been envisioned as an effective tool for combating alcohol-induced acute liver injury [9,10]. Normally, the level of Nrf2 in the cytosol is low due to ubiquitination and proteasomal degradation regulated by a Kelch-like ECH-associated protein (Keap1) [11,12]. However, the presence of oxidants can oxidize cysteine residues in Keap1 into disulfides or conjugate them to electrophiles, resulting in the nuclear translocation of Nrf2 and the activation of antioxidant genes [13], including heme oxygenase-1 (HO1), NAD(P)H dehydrogenase [quinone] 1 (NQO1), and the mitochondrial superoxide dismutase 2 (SOD2) [14,15].\n",
      "============================================================\n",
      "1. Introduction | idx 2/28\n",
      "322 tokens | 1670 chars\n",
      "In addition, Nrf2 activation increases the expression levels of nuclear respiratory factor-1 (Nrf-1) and peroxisome proliferator-activated receptor  coactivator 1 (PGC1), which in turn modulates the expression of mitochondrial respiratory subunits and translational components [16]. The genes activated by the antioxidant responsive element (ARE) protect cells against any damage caused by reactive oxygen species (ROS) [17]. In addition to Keap1, glycogen synthase kinase 3 (GSK3) is another negative regulator of Nrf2 that acts to downregulate the antioxidant stress response [18,19]. GSK3 promotes the nuclear exclusion and degradation of Nrf2 in a Keap1-independent fashion in stressed or injured cells with active Nrf2, thereby reducing the degree of protection conferred by Nrf2 activity [20,21]. Therefore, increasing the Nrf2 antioxidant activity via targeting Keap1 or GSK3 could be a potential approach for treating ALD [22]. Most pharmacological approaches towards Nrf2 activation are focused on the inhibition of Keap1 [23,24]. Several Keap1 inhibitors have been reported in the last decade, and the vast majority of these covalently modify cysteine residues in Keap1 [25]. Covalently targeting cysteines in Keap1 may lack selectivity due to the presence of other reactive cysteine residues in the cell, resulting in adverse side effects [26,27]. However, because cysteine residues are abundant in cells, the safety and specificity of covalent drugs are a concern [28]. Recently, proteinprotein interaction (PPI) inhibitors of Keap1 and Nrf2, developed based on the X-ray structure of Keap1, have emerged as a new class of Nrf2 activators [29,30,31].\n",
      "============================================================\n",
      "1. Introduction | idx 3/28\n",
      "138 tokens | 605 chars\n",
      "Our group has previously reported a cyclometalated iridium(III) metal complex as a Keap1Nrf2 PPI inhibitor, which is a promising therapeutic agent for acetaminopheninduced acute liver injury [29,30]. However, no Keap1Nrf2 inhibitor has yet entered the clinic for the treatment of human diseases [32,33]. Meanwhile, although a few GSK3 inhibitors have been reported for liver injury treatment, no GSK3 inhibitor to date has made it to the market [34]. Antrodia cinnamomea, a basidiomycete that is endemic to Taiwan, is an edible fungus that is a component of many traditional herbal medicines [35,36].\n",
      "============================================================\n",
      "1. Introduction | idx 4/28\n",
      "376 tokens | 1809 chars\n",
      "Either crude extracts or molecules isolated from A. cinnamomea have shown diverse biological properties, including antioxidant, anti-tumour, and anti-inflammatory activities [37,38]. Recently, research has shown that dehydroeburicoic acid (DEA) (1) from A. cinnamomea could inhibit alcoholic fatty liver disease (AFLD) by upregulating aldehyde dehydrogenase 2 family member (ALDH2) activity [39,40]. Additionally, DEA (1) also exhibited protective effects against non-alcoholic fatty liver disease (NAFLD) through activating ALDH2 and accelerating the elimination of ROS and harmful aldehydes [40]. Therefore, DEA (1) has the potential ability to treat ALD. However, its mechanisms of hepatoprotection in ALD are still not clear. Besides DEA (1), other triterpenoids have also been studied as GSK3 inhibitors and activators of Nrf2 because of their anti-inflammatory and antioxidant activities [41,42,43,44,45]. Research has shown that singlemolecule drugs with two different biological activities may avoid some side effects and improve efficacy [46]. In this study, we discovered a potent dual Keap1Nrf2 PPI and GSK3 inhibitor, DEA (1), and investigated its underlying mechanisms of action. Through in vitro fluorescence polarization (FP) screening, we identified DEA (1) as the most potent candidate for disrupting the Keap1Nrf2 PPI (EC50 = 14.1 M) from thirteen analogues isolated from A. cinnamomea. Moreover, DEA (1) inhibited GSK3 kinase activity with an EC50 value of 8.0 M. In cells, low-toxicity compound DEA (1) engaged Keap1 and GSK3 to promote Nrf2 accumulation in the nucleus, leading to the upregulation of ARE transcriptional activity and increasing the expression of downstream antioxidant factors with more potent than the reported Keap1Nrf2 PPI inhibitor ML334 in ALD model cells.\n",
      "============================================================\n",
      "2.1. DEA (1) Inhibits the Keap1Nrf2 PPI and GSK3 Activity | idx 5/28\n",
      "99 tokens | 485 chars\n",
      "The fluorescence polarization assay is a commonly used high-throughput assay to detect PPI inhibitors in solution. We used labeled Nrf2 peptides, bearing the highaffinity ETGE motif, which is recognized by Keap1, as fluorescent tracers to monitor for inhibition of the Keap1Nrf2 interaction. ML334 is the first noncovalent small molecule inhibitor of Keap1Nrf2 interaction reported and is distinct from other Nrf2 inducers [47]. In this paper, ML334 was used as a positive control.\n",
      "============================================================\n",
      "2.1. DEA (1) Inhibits the Keap1Nrf2 PPI and GSK3 Activity | idx 6/28\n",
      "338 tokens | 1394 chars\n",
      "Compounds 113 from A. cinnamomea and ML334 were screened against the Keap1Nrf2 interaction by FP (A). The preliminary screening results showed that DEA (1) was the most potent compound, which disrupted the Keap1Nrf2 interaction by 62% inhibition at 50 M (cf. ML334: 60% inhibition). The NMR spectrum of DEA (1) is presented in Supplementary Figure S1. The potency of DEA (1) was further investigated in a dose-response assay, revealing an EC50 value of 14.1  0.1 M against Keap1Nrf2 PPI (B). We also explored the inhibitory effect of DEA (1) against GSK3, which negatively regulates Nrf2 in a fashion independent of Keap1. 10 M of DEA (1) reduced GSK3 activity by 58.9%, making it more potent than ML334 at the same concentration (11.3% inhibition) (C). Moreover, a dose-response assay revealed an EC50 value of 8.0  0.7 M against GSK3 activity, indicating that DEA (1) is an effective GSK3 inhibitor (D). In the nucleus, Nrf2 binds to the ARE to activate antioxidant gene transcription [14]. Thus, we tested the transcriptional activity of ARE after treatment of LO2 cells with DEA (1) and ML334 (10 M) using the dual luciferase assay kit (E). The activity of ARE was significantly increased by 54% in the treatment of DEA (1) compared to 27% in the ML334 group. The results showed that DEA (1) notably activates the transcriptional activity of ARE compared to ML334 in cellulo.\n",
      "============================================================\n",
      "2.2. DEA (1) Exhibits Low Cytotoxicity In Cellulo | idx 7/28\n",
      "72 tokens | 340 chars\n",
      "The cytotoxicity of compound DEA (1) was detected in both human normal cell lines LO2 and HEK 293T, and the hepatocellular carcinoma cell line HepG2. After treatment with different concentrations of DEA (1), no significant cytotoxicity was observed (FH). These data indicate that DEA (1) could be potentially safe for treating ALD in vivo.\n",
      "============================================================\n",
      "2.3. DEA (1) as a Dual Inhibitor of Keap1 and GSK3 | idx 8/28\n",
      "264 tokens | 1086 chars\n",
      "The ability of DEA (1) to disrupt the Keap1Nrf2 binding in cellulo was monitored through a CoIP experiment using human normal liver cell line LO2 cells (A). After treatment with DEA (1) (10 M) or ML334 (10 M) for 8 h, there was a 45% reduction of Keap1 co-precipitated with Nrf2 for DEA (1) as compared to 12% inhibition for ML334, indicating that DEA (1) could more effectively inhibit the Keap1Nrf2 interaction compared to ML334 in living cells (B). We investigated the ability of DEA (1) to target Keap1, Nrf2, and GSK3 in the cellular environment using CETSA. LO2 cell lysates were incubated with 10 M of DEA (1) at room temperature for 30 min. Then, aliquots were heated individually at different temperatures and the protein in the soluble fraction was quantified by WB. Compound DEA (1) significantly stabilized Keap1 (Tm: ca. 4.5 C) and GSK3 (Tm: ca. 3.9 C), while having no observable effect on the thermal stability of Nrf2 and actin (C,D). This result suggests that DEA (1) can bind with Keap1 and GSK3, even within the complicated environment of cell lysates.\n",
      "============================================================\n",
      "2.4. DEA (1) Targets Keap1 and GSK3 to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins | idx 9/28\n",
      "357 tokens | 1709 chars\n",
      "To verify whether Keap1 or GSK3 are direct targets of DEA (1), knockdown assays were performed. NQO1 and HO1 are target antioxidant genes of Nrf2 that increase cytoprotection against oxidative stress [48] As expected, Keap1 knockdown in LO2 cells induced a noticeable rise of the expression levels of both Nrf2 and its gene targets, NQO1 and HO1 (A,B). However, treating Keap1 knockdown cells with DEA (1) led to no further rises in the NQO1 and HO1 level, in contrast to the control cells, where DEA (1) treatment produced noticeable increases of NQO1 and HO1. These results suggest that DEA (1) activates NQO1 and HO1 through targeting Keap1. Similarly, GSK3 knockdown induced Nrf2 and downstream antioxidant protein expression but without influencing Keap1 protein level (C,D). In addition, DEA (1) had a diminished effect on NQO1 and HO1 level in GSK3 knockdown cells as compared to the control cells. Finally, we knocked down GSK3 and Keap1 at the same time to verify the dual binding mechanism of DEA (1) on GSK3 and Keap1 (E,F). The results showed that treating the GSK3 and Keap1 knockdown cells with DEA (1) did not cause a further significant induction of NQO1 and HO1 expression. Evidence for the hypothesis that DEA (1) acts through targeting both Keap1 and GSK3 comes from considering the Nrf2 level. In either the Keap1 knockdown or GSK3 knockdown cells, DEA (1) treatment resulted in slight increases of Nrf2 expression relative to DMSO. However, in the double Keap1 and GSK3 knockdown cells, treatment with DEA (1) did not lead to increases of Nrf2 expression. These results suggest that DEA (1) treatment mimicked the effects of double knockdown in increasing Nrf2 expression.\n",
      "============================================================\n",
      "2.4. DEA (1) Targets Keap1 and GSK3 to Induce Nrf2 Accumulation in the Nucleus and the Expression of Downstream Antioxidant Proteins | idx 10/28\n",
      "179 tokens | 877 chars\n",
      "Therefore, DEA (1) exerts its antioxidant effects by targeting both Keap1 and GSK3. Inhibition of the Keap1Nrf2 interaction is known to increase Nrf2 accumulation in the nucleus [47]. Meanwhile, GSK3 is also a negative regulator of Nrf2 accumulation by acting independently of Keap1 [49]. Therefore, we evaluated the Nrf2 level in the cytoplasm and nucleus of LO2 cells via WB after the treatment of cells with DEA (1) (10 M) for 8 h. As shown in A, DEA (1) significantly increased Nrf2 translocation into the nucleus in a dose-dependent manner. GSK3 knockdown also phenocopied compound DEA (1) treatment at inducing Nrf2 nuclear accumulation without influencing Keap1 level (B). In summary, DEA (1) disrupts the Keap1Nrf2 PPI and inhibits GSK3 to increase the translocation of liberated Nrf2 into the nucleus, thereby presumably increasing ARE transcriptional activity.\n",
      "============================================================\n",
      "2.5. DEA (1) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells | idx 11/28\n",
      "360 tokens | 1348 chars\n",
      "Nuclear Nrf2 heterodimerizes with musculoaponeurotic fibrosarcoma (Maf) to recognize ARE and promote the transcription of antioxidant genes, including Nrf1, HO1, and NQO1 [14]. Therefore, we evaluated the ability of DEA (1) to increase Nrf1, NQO1, and HO1 levels in ALD model cells, which were prepared by treating LO2 cells with 0.3% EtOH for 8 h. 0.3% EtOH was reported to induce oxidative stress in living cells [50,51,52]. In addition, a preliminary time-dependent assay was performed, which identified an optimal incubation time length of 8 h (Supplementary Figure S2). As shown in A, DEA (1) (30 M) induced Nrf1 (B), NQO1 (C) and HO1 (D) protein levels by 1.2, 3.8 and 1.8-fold, respectively, in ALD model cells, making it more potent than ML334 (30 M), which increased Nrf1, NQO1, and HO1 protein levels by 1.0, 1.1, and 1.5-fold, respectively. Nrf2 directly controls the expression of the transcriptional factor PGC1, a regulator of mitochondrial biogenesis, while it also directly regulates the expression of mitochondrial antioxidant enzymes, such as superoxide dismutase 2 (SOD2) [53,54]. Hence, the effect of DEA (1) on mRNA levels of SOD2, HO1, NQO1, Nrf1, and PGC1 in 0.3% EtOHtreated LO2 cells was also evaluated. 30 M of DEA (1) rescued SOD2 mRNA level in ALD model cells by 2-fold, compared to 1.5-fold with ML334 (E).\n",
      "============================================================\n",
      "2.5. DEA (1) Activates Nrf2 Downstream Antioxidant Genes in ALD Model Cells | idx 12/28\n",
      "194 tokens | 747 chars\n",
      "DEA (1) (30 M) also increased NQO1 (F) and HO1 (G) mRNA levels by 2.0 and 2.5-fold, respectively, making it more potent than ML334 which increased NQO1 and HO1 mRNA levels by 1.8 and 1.4-fold, respectively. In addition, the mRNA levels of Nrf1 and PGC1 were also increased 2-fold by compound DEA (1) (30 M) (H,I). The results also show a slight increase of NQO1, HO1, and PGC1 levels by 0.3% EtOH treatment alone, which has been previously observed in other studies [55]. However, the subsequent increases of antioxidant genes were much greater upon further treatment of DEA (1). Taken together, these results indicate that DEA (1) can activate Nrf2 downstream antioxidant genes in EtOHtreated human liver cells, a cellular model of ALD.\n",
      "============================================================\n",
      "2.6. DEA (1) Repairs the EtOH Induced Mitochondrial Dysfunction and Improves the Antioxidant Activity in LO2 Cells | idx 13/28\n",
      "291 tokens | 1261 chars\n",
      "Inflammation and oxidative stress increase ROS production, and the activated Nrf2 was able to maintain low intracellular ROS level to protect the cell from oxidant injury [17]. Moreover, growing evidence shows that mitochondrial and oxidant injury induced by ethanol consumption play key roles in alcoholinduced liver injury [56]. Alcoholinduced ROS causes mitochondrial membrane depolarization and mitochondrial permeability transition (MPT), leading to hepatic apoptosis and necrosis [54]. Therefore, we detected the effect of DEA (1) and ML334 on ROS level in 0.3% EtOHtreated LO2 cells (A). The results indicated that both 1 and ML334 could significantly decrease ROS level induced by EtOH. The MPTP assay showed that DEA (1) (30 M) can restore MPTP function (40%) with a higher activity as compared to ML334 (34%) (B,C). ATP is an indicator of mitochondrial function and ATP level are significantly reduced in ethanolfed Nrf2/ mice [57]. DEA (1) (30 M) rescued 50% of ATP in LO2 cells after EtOH treatment, making it more potent than ML334, which rescued 20% of the ATP level under the same conditions (D). Finally, DEA (1) (30 M) induced a higher increase of antioxidant activity (4-fold) compared to ML334 (2-fold) under the same conditions (E).\n",
      "============================================================\n",
      "3. Discussion | idx 14/28\n",
      "278 tokens | 1447 chars\n",
      "ALD is a serious chronic liver disease caused by oxidative stress and alcohol metabolism, which leads to massive global deaths [1]. Unfortunately, until now, there are no effective therapeutic drugs approved by the FDA for the treatment of ALD [58]. The pathogenesis of ALD is poorly characterized and research on the mechanism is still not clear. A number of reports have shown that the Keap1Nrf2 pathway plays an important role in activating ARE antioxidant signaling [14]. Nrf2 is regulated via Keap1-independent pathways, including mitogen-activated protein kinaseErk and PI3KAkt pathways [59]. Among these regulatory pathways, GSK3 has emerged as a convergent point [60,61]. Because Nrf2 degradation and nuclear exclusion are regulated by GSK3, it is critical in switching off the selfprotective antioxidant stress response after injury [59,62]. Hence, targeting Keap1 and GSK3 to activate Nrf2 are potential therapeutic strategies for ALD. Natural products provide diverse scaffolds with high bioactivity and low toxicity, and could be developed as promising candidates in drug development [63]. A. cinnamomea is a traditional Chinese herb that is a component of various medicines [64]. Triterpenoids are the major constituent in A. cinnamomea, and several studies have demonstrated their antioxidative and hepatoprotective effects [65,66]. However, the hepatoprotective mechanism of these compounds has not been extensively explored.\n",
      "============================================================\n",
      "3. Discussion | idx 15/28\n",
      "347 tokens | 1536 chars\n",
      "DEA (1) has previously shown promising hepatoprotective activity in some studies [67], including in mouse models [39,67,68,69]. In recent years, many Keap1Nrf2 PPI inhibitors and GSK3 inhibitors have been reported for various indications, such as Alzheimers disease, cancers, Parkinsons disease, kidney diseases, Acetaminophen (APAP)induced liver injury, and ferroptosis [30,70,71,72,73,74]. However, few have been studied in the context of ALD. In our present study, we discovered DEA (1) as a potent Keap1Nrf2 PPI inhibitor from A. cinnamomea extracts. We explored the detailed mechanism of hepatoprotective action of DEA (1) in ALD, which acted via targeting both the Keap1Nrf2 PPI and GSK3 (). To our best knowledge, the dual inhibition mechanism of DEA (1) in ALD has not been reported before. In in vitro assays, DEA (1) inhibited the Keap1Nrf2 PPI (EC50 = 14.1 M) as well as GSK3 kinase activity (EC50 = 8.0  0.7 M) with greater potency than the known Keap1Nrf2 inhibitor, ML334. Cellular experiments confirmed the ability of DEA (1) to engage Keap1 and GSK3 in cellulo to activate Nrf2 and stimulate the expression of AREcontrolled genes. The specificity of DEA (1) was further confirmed using Keap1 and GSK3 knockdown experiments. DEA (1) treatment phenocopied dual knockdown of Keap1 and GSK3, suggesting that DEA (1) acted simultaneously through both pathways to exert its biological effects. Moreover, DEA (1) increased Nrf2 nuclear accumulation, mimicking GSK3 knockdown, without influencing Keap1 level.\n",
      "============================================================\n",
      "3. Discussion + 4.1. Cell Lines and Culture + 4.2. Chemical | idx 16/28\n",
      "277 tokens | 1249 chars\n",
      "Finally, we showed that DEA (1) could restore MPTP function and reduce ROS level induced by EtOH in ALD model cells. DEA (1) displayed low cytotoxicity in both normal and cancerous liver cells, indicating its potential safety for treating ALD in vivo. In summary, DEA (1) is a promising scaffold for the further development of dual inhibitors of the Keap1Nrf2 PPI and GSK3 as therapeutic agents against ALD. The structural modification of compound DEA (1) is in progress and animal studies will be performed in due course. The human normal liver cell line (LO2), human embryonic kidney 293 T cell line, and HepG2 cell line were purchased from American Type Culture Collection (Manassas, VA, USA). Cells were cultured in Dulbeccos modified Eagles medium (DMEM), with 10% FBS, 100 units/mL penicillin, 100 g/mL streptomycin. Fetal bovine serum (FBS), DMEM, penicillin, and streptomycin were purchased from Gibco BRL (Gaithersburg, MD, USA). Compounds 113 (purity > 98% by HPLC analysis) were isolated from A. cinnamomea in a previous study [75]. New 1H and 13C nuclear magnetic resonance (NMR) were recorded for DEA (1) to verify its identity for this study (Supplementary Figure S1). Spectral data were consistent with our previous report [75].\n",
      "============================================================\n",
      "4.3. Fluorescence Polarization Assay | idx 17/28\n",
      "133 tokens | 622 chars\n",
      "The ability of compounds 113 and ML334 to inhibit Keap1Nrf2 peptide binding was evaluated using a FP assay according to the manufacturers instruction (BPS Bioscience, San Diego, CA, USA). ML334 was a noncovalent small molecule inhibitor of the Keap1Nrf2 and purchased from MedChemExpress (Princeton, NJ, USA). The tested compounds were dissolved in the provided buffer at the indicated concentrations. Then, the required volumes of Nrf2 peptide, Bovine Serum Albumin (BSA), Keap1 protein, and assay buffer were added to each well, followed by incubation for 30 min and the fluorescent polarization value was recorded.\n",
      "============================================================\n",
      "4.4. Western Blot (WB) and CoImmunoprecipitation (CoIP) | idx 18/28\n",
      "285 tokens | 967 chars\n",
      "The protocol was reported in a previous study [76]. In brief, LO2 cells were seeded at a density of 2  106 cells in a sixwell plate. LO2 lysate is harvested after incubation with 10 M of compound DEA (1) or ML334 for 8 h. A 10% SDSPAGE gel was used to separate the LO2 cell lysates, which were transferred to a PVDF membrane (BioRad, Hercules, CA, USA). The membranes were incubated with antibodies antiKeap1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 8047S), Nrf2 (1:1000, Proteintech, Wuhan, China; Cat# 16396-1-AP), GSK3 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 9832S), HO1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 5853S), NQO1 (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat# 3187S), actin (1:1000, Absin, Shanghai, China; Cat#: abs137975), or anti-lamin B (1:1000, Cell Signaling Technology, Danvers, MA, USA; Cat # 12586S) overnight. The membranes were washed with wash buffer three times.\n",
      "============================================================\n",
      "4.4. Western Blot (WB) and CoImmunoprecipitation (CoIP) | idx 19/28\n",
      "157 tokens | 679 chars\n",
      "A secondary antibody (1:1000) was added to the membranes. After incubated for 2 h, the proteins bands were detected using enhanced chemiluminescent Plus reagents (GE Healthcare, Boston, MA, USA) and analyzed by Image Lab. The antibodies were purchased from Abcam (Waltham, MA, USA). CoIP between Keap1 and Nrf2 was performed following the protocol from Life Technologies. Briefly, lysates from 2  106 LO2 cells were incubated with antiNrf2 or Rabbit mAb IgG (Abcam, Waltham, MA, USA; Cat#: ab205718) overnight. Then, samples were incubated with 10 L Dynabeads (Life technologies, Foster, CA, USA) for 6 h. The Dynabeads were washed and analyzed by WB with the Keap1 antibody.\n",
      "============================================================\n",
      "4.5. Real-Time Quantitative Polymerase Chain Reaction (RTqPCR) | idx 20/28\n",
      "193 tokens | 836 chars\n",
      "LO2 cells were incubated with 10 M or 30 M compound DEA (1) for 8 h and the total RNA was extracted using the NucleoSpin RNA Plus kit (Takara, Tokyo, Japan). We checked the RNA quality during the RTqPCR operation and ensured that all the A260/280 value of samples were around 2, which is consistent with the literature [77]. We also performed an RT negative control and found no PCR amplification. Then, cDNA was synthesized using the PrimeScript RT Reagent Kit (Takara, Tokyo, Japan). The RTqPCR conditions were as previously reported, with minor modifications [77]. DNA denaturation temperature: 95 C (30 s); annealing temperature: 58 C (30 s); extension temperature: 72 C (45 s), number of cycles: 40. RTqPCR was performed and analyzed as described previously [78]. The related primers are listed in Supplementary Table S1.\n",
      "============================================================\n",
      "4.6. MTT Assay + 4.7. Detection of ROS and Antioxidant Activity + 4.8. Nuclear and Cytoplasmic Extraction | idx 21/28\n",
      "300 tokens | 1312 chars\n",
      "The LO2, HUVEC and HEK 293 T cell lines were seeded at 1  104 cells per well in 96well plates and incubated with DMEM overnight at 37 C in a humidified CO2 incubator to ensure attachment. The medium was then replaced with a medium that contained the different concentrations of compound for 48 h at 37 C. The final concentration of DMSO was 0.1% or lower in all cases (including controls). The medium with 100 L of MTT (3-(4,5-dimethylthiazol-2-yl)-2,5-tetrazolium bromide) reagent (1 mg/mL) was then replaced. After 4 h incubation, it was replaced with 100 L of DMSO and measured using a microplate reader at 570 nm. LO2 cells at a density of 2  106 cells/well in a six-well plate were treated with different concentrations of DEA (1) or 10 M ML334 for 8 h. ROS level and antioxidant activity were measured using the Reactive Oxygen Species Assay kit and the Total Antioxidant Capacity Assay Kit with FRAP, respectively, according to the manufacturers instructions (Beyotime, Shanghai, China). LO2 cells were incubated with 10 M or 30 M of DEA (1) for 8 h. The nuclear and cytoplasmic extraction followed the protocol from the manufacturer (Epigentek, New York, NY, USA, EpiQuik Nuclear Extraction Kit I, Catalog # OP-0002), Then, protein expression was analyzed by WB with the indicated antibodies.\n",
      "============================================================\n",
      "4.9. GSK3 Kinase Assay | idx 22/28\n",
      "135 tokens | 602 chars\n",
      "The ability of DEA (1) and ML334 to inhibit GSK3 activity was evaluated using luminescent kinase assay according to the manufacturers instruction (Promega Corporation, Madison, WI, USA). Briefly, the 384 well white plate was mixed with 1 L of inhibitor, 2 L of enzyme, and 2 L of substrate/ATP mix and incubated at room temperature for 60 min. After 60 min, 5 L of ADPGlo reagent was added and incubated at room temperature for 40 min. Then 10 L of kinase detection reagent was added, incubated at room temperature for 30 min and the luminescence value was recorded (integration time 0.51 s).\n",
      "============================================================\n",
      "4.10. Dual Luciferase Reporter Gene Assay + 4.11. Cellular Thermal Shift Assay (CETSA) + 4.12. Mitochondrial Permeability Transition Pore (MPTP) Assay | idx 23/28\n",
      "291 tokens | 1355 chars\n",
      "LO2 cells were seeded in a sixwell plate with 80% confluence in a DMEM medium for 24 h. The AREluciferase vector or a control plasmid with TurboFect transfection reagent (Thermo Fisher Scientific, Waltham, MA, USA) were transfected into LO2 cells according to the manufacturers instruction, and cells were incubated for 48 h. Then, LO2 cells were incubated with 10 M compound DEA (1), 10 M ML334 or DMSO for 4 h. The ARE activity was detected through the manufacturers instruction (Beyotime Dual Luciferase Reporter Gene Assay Kit, Shanghai, China). LO2 cell lysates were incubated with 10 M compound DEA (1) and DMSO at room temperature for 30 min. The lysates are divided and heated individually at the indicated temperature. The supernatants of the heated lysates were collected after centrifugation, and then analyzed by WB with the Keap1, Nrf2, and actin antibodies. MPTP kit was used for evaluation of mitochondrial permeability transition according to the manufacturers instructions (Beyotime, Shanghai, China). A density of 1  104 cells in a sixwell plate, EtOHinduced LO2 cells were treated with 10 M of ML334, 10 or 30 M compound DEA (1), DMSO for 8 h, and then pretreated with calcein or calcein and CoCl2. Cells were stained by calcein acetoxymethyl ester (Calcein AM) and MPT function was detected by fluorescence spectroscopy.\n",
      "============================================================\n",
      "4.13. SiRNA Gene Knockdown + 4.14. Statistical Analysis | idx 24/28\n",
      "242 tokens | 1005 chars\n",
      "The sequences of Keap1 siRNA are 5-GGCCUUUGGCAUCAUGAACTT-3 (sense) and 5-GUUCAUGAUGCCAAAGGCCTG-3 (antisense) [79]. GSK3 siRNA is 5-GCAUUUAUCGUUAACCUAA-3 (sense) and 5-UUAGGUUAACGA UAAAUGC-3 (antisense) [80]. Keap1 and GSK3 siRNA was transfected with Lipo3000 reagent (Thermo Fisher Scientific, Waltham, MA, USA) into 1  106 LO2 cells following the manufacturers instruction. After incubation for 48 h, Keap1 and GSK3 protein expression and functional experiments were performed. Statistical significance was determined using the Students t-test for experiments comparing two groups. The data is homogeneous and follows a normal distribution before proceeding to the statistical analysis. Comparisons among groups were analysed using analysis of variance (ANOVA). All statistical tests were done using the GraphPad Prism version 8.0 software (GraphPad Software Inc. San Diego, CA, USA). All values were expressed as the mean  standard deviation (SD). p < 0.05 means statistical significance.\n",
      "============================================================\n",
      "5. Conclusions | idx 25/28\n",
      "136 tokens | 715 chars\n",
      "Recent studies on ALD are scarce compared to research on other liver diseases, and there are still no highly satisfactory therapeutic options for ALD. To our knowledge, no dual inhibitor of the Keap1Nrf2 PPI and GSK3 to treat alcohol-induced liver injury has been reported. Compound DEA (1) potently inhibits Keap1-Nrf2 binding and GSK3 activity, enhancing ARE activity in cells, while showing low cytotoxicity. Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3, enhancing its stability and activity in LO2 cells. Compound DEA (1) regulates antioxidant effects by targeting Keap1 and GSK3, as evidenced by reduced expression of antioxidant proteins in cells with knockdown of either Keap1 or GSK3.\n",
      "============================================================\n",
      "5. Conclusions | idx 26/28\n",
      "123 tokens | 634 chars\n",
      "Compound DEA (1) increases Nrf2 translocation into the nucleus, while GSK3 knockdown enhances nuclear Nrf2 levels. Compound DEA (1) and ML334 induce antioxidant factor expression in ALD cells, enhancing Nrf1, NQO1, HO-1, Keap1, and Nrf2 protein levels and mRNA expression. Compound DEA (1) protects mitochondria from 0.3% EtOH-induced injury by reducing ROS levels, enhancing ATP levels, and increasing antioxidant activity, suggesting its potential as a mitochondrial protective agent. The dual inhibitor DEA (1) protects hepatocytes from EtOH-induced injury by inhibiting the degradation of MPTP and enhancing antioxidant activity.\n",
      "============================================================\n",
      "5. Conclusions | idx 27/28\n",
      "220 tokens | 1148 chars\n",
      "This paper has revealed that DEA (1), one of the active triterpenes isolated from A. cinnamomea, could be a potential lead compound for treating ALD. Compound DEA (1) inhibited the Keap1Nrf2 interaction and GSK3 activity both in vitro and in cellulo, and displayed low cytotoxicity at the concentrations required for inducing antioxidant activity. Moreover, DEA (1) could engage Keap1 and GSK3 in liver cells and increase Nrf2 nucleus translocation, thereby activating ARE transcriptional activity and upregulating antioxidant genes and mitochondrial biogenesis regulators. Significantly, compound DEA (1) displayed more potent hepatoprotective activity when compared with the classic Keap1Nrf2 PPI inhibitor ML334. Therefore, the structure of DEA (1) may be optimized to expand the diversity, improve the activity, and be used in animal research. To our knowledge, this paper firstly reports DEA (1) acts as the dual inhibitor of Keap1Nrf2 PPI and GSK3 with the potential for protecting against ALD. We anticipate that natural product DEA (1) can be considered as a potential scaffold for the development of clinical agents for treating ALD.\n"
     ]
    }
   ],
   "source": [
    "from typing import List, Iterable, Tuple, Dict\n",
    "import re\n",
    "MIN_TOKENS = 120\n",
    "MAX_TOKENS = 380\n",
    "ALLOW_SENTENCE_OVERFLOW = True    \n",
    "ABSORB_FIGURES_INTO_PREV = True    \n",
    "NO_OVERLAP = True                  \n",
    "\n",
    "\n",
    "SENT_BOUNDARY = re.compile(r'([.!?][\"\\')\\]]*)\\s+(?=\\S)')\n",
    "\n",
    "def sentences(text: str) -> List[str]:\n",
    "    \"\"\"\n",
    "    Split text into whole sentences. Keeps punctuation and any trailing quotes/brackets\n",
    "    with the sentence on the left. No lookbehinds => works in Python re.\n",
    "    \"\"\"\n",
    "    parts = SENT_BOUNDARY.split(text.strip())\n",
    "    if len(parts) == 1:\n",
    "        return [parts[0]] if parts and parts[0].strip() else []\n",
    "\n",
    "    out = []\n",
    "    cur = parts[0]\n",
    "    i = 1\n",
    "    while i < len(parts):\n",
    "        delim = parts[i]         \n",
    "        nxt   = parts[i+1] if i+1 < len(parts) else \"\"\n",
    "        cur = (cur + delim).strip()\n",
    "        out.append(cur)\n",
    "        cur = nxt\n",
    "        i += 2\n",
    "\n",
    "    if cur.strip():\n",
    "        out.append(cur.strip())\n",
    "    return out\n",
    "\n",
    "def tokenize_len(text: str) -> int:\n",
    "    return len(tokenizer.encode(text, truncation=False))\n",
    "\n",
    "def pack_by_tokens(sents: List[str], max_tokens: int) -> List[str]:\n",
    "    \"\"\"\n",
    "    Greedily pack whole sentences up to max_tokens.\n",
    "    If one sentence alone > max_tokens:\n",
    "      - keep it as its own (oversized) chunk if ALLOW_SENTENCE_OVERFLOW\n",
    "      - else (not recommended) you could hard-slice here.\n",
    "    \"\"\"\n",
    "    out, cur, cur_len = [], [], 0\n",
    "    for s in sents:\n",
    "        slen = tokenize_len(s)\n",
    "        if slen > max_tokens and ALLOW_SENTENCE_OVERFLOW:\n",
    "            if cur: out.append(\" \".join(cur)); cur, cur_len = [], 0\n",
    "            out.append(s)\n",
    "            continue\n",
    "        if not cur or cur_len + slen <= max_tokens:\n",
    "            cur.append(s); cur_len += slen\n",
    "        else:\n",
    "            out.append(\" \".join(cur)); cur, cur_len = [s], slen\n",
    "    if cur:\n",
    "        out.append(\" \".join(cur))\n",
    "    return out\n",
    "\n",
    "def resegment_big_chunks(texts: List[str]) -> List[str]:\n",
    "    \"\"\"Re-pack any long raw chunks by sentence so none exceed MAX_TOKENS (except single long sentences).\"\"\"\n",
    "    out = []\n",
    "    for txt in texts:\n",
    "        toks = tokenize_len(txt)\n",
    "        if toks <= MAX_TOKENS:\n",
    "            out.append(txt)\n",
    "        else:\n",
    "            out.extend(pack_by_tokens(sentences(txt), MAX_TOKENS))\n",
    "    return out\n",
    "\n",
    "def merge_small_chunks_within_section(texts: List[str]) -> List[str]:\n",
    "    \"\"\"Merge adjacent chunks within a section until each is >= MIN_TOKENS (except single long sentence case).\"\"\"\n",
    "    if not texts: return []\n",
    "    out, buf, buf_tok = [], \"\", 0\n",
    "    for t in texts:\n",
    "        t_tok = tokenize_len(t)\n",
    "        if not buf:\n",
    "            buf, buf_tok = t, t_tok\n",
    "            continue\n",
    "        \n",
    "        if buf_tok < MIN_TOKENS and (buf_tok + t_tok) <= MAX_TOKENS:\n",
    "            buf = (buf + \" \" + t).strip()\n",
    "            buf_tok += t_tok\n",
    "        else:\n",
    "            out.append(buf)\n",
    "            buf, buf_tok = t, t_tok\n",
    "    if buf:\n",
    "        out.append(buf)\n",
    "    \n",
    "    if len(out) >= 2 and tokenize_len(out[-1]) < MIN_TOKENS:\n",
    "        if tokenize_len(out[-2]) + tokenize_len(out[-1]) <= MAX_TOKENS:\n",
    "            out[-2] = (out[-2] + \" \" + out[-1]).strip()\n",
    "            out.pop()\n",
    "    return out\n",
    "\n",
    "def is_figure_like(title: str) -> bool:\n",
    "    t = (title or \"\").strip().lower()\n",
    "    return t.startswith(\"figure\") or t.startswith(\"table\")\n",
    "\n",
    "\n",
    "def normalize_chunks_by_section(docs: List[Document]) -> List[Document]:\n",
    "    \n",
    "    grouped: Dict[Tuple[str,str], List[Tuple[int,Document]]] = {}\n",
    "    first_idx: Dict[Tuple[str,str], int] = {}\n",
    "\n",
    "    for i, d in enumerate(docs):\n",
    "        key = (d.metadata.get(\"section_id\"), d.metadata.get(\"section_title\"))\n",
    "        grouped.setdefault(key, []).append((i, d))\n",
    "        first_idx.setdefault(key, i)\n",
    "\n",
    "   \n",
    "    sections_sorted = sorted(grouped.items(), key=lambda kv: first_idx[kv[0]])\n",
    "\n",
    "   \n",
    "    absorbed: Dict[Tuple[str,str], List[Tuple[int,Document]]] = {}\n",
    "    last_non_fig_key = None\n",
    "    for key, items in sections_sorted:\n",
    "        sec_id, sec_title = key\n",
    "        if ABSORB_FIGURES_INTO_PREV and is_figure_like(sec_title) and last_non_fig_key is not None:\n",
    "            absorbed.setdefault(last_non_fig_key, []).extend(items)\n",
    "        else:\n",
    "            absorbed.setdefault(key, []).extend(items)\n",
    "            if not is_figure_like(sec_title):\n",
    "                last_non_fig_key = key\n",
    "\n",
    "   \n",
    "    linear: List[Document] = []\n",
    "    for key, items in sorted(absorbed.items(), key=lambda kv: first_idx[kv[0]]):\n",
    "       \n",
    "        items_sorted = sorted(\n",
    "            items,\n",
    "            key=lambda t: (\n",
    "                t[1].metadata.get(\"subchunk_index\") is None,\n",
    "                t[1].metadata.get(\"subchunk_index\", 10**9),\n",
    "                t[0],\n",
    "            )\n",
    "        )\n",
    "        base_meta = (items_sorted[0][1].metadata or {}).copy()\n",
    "        texts = [it[1].page_content for it in items_sorted]\n",
    "        texts = resegment_big_chunks(texts)\n",
    "        texts = merge_small_chunks_within_section(texts)\n",
    "       \n",
    "        for t in texts:\n",
    "            meta = {**base_meta, \"is_parent\": False}\n",
    "            linear.append(Document(page_content=t, metadata=meta))\n",
    "\n",
    "   \n",
    "    compacted: List[Document] = []\n",
    "    for d in linear:\n",
    "        if compacted and tokenize_len(d.page_content) < MIN_TOKENS:\n",
    "            prev = compacted[-1]\n",
    "            if tokenize_len(prev.page_content) + tokenize_len(d.page_content) <= MAX_TOKENS:\n",
    "               \n",
    "                prev.page_content = (prev.page_content + \" \" + d.page_content).strip()\n",
    "                \n",
    "                pt = (prev.metadata.get(\"section_title\") or \"\").strip()\n",
    "                ct = (d.metadata.get(\"section_title\") or \"\").strip()\n",
    "                if pt and ct and pt != ct and not pt.endswith(ct) and f\" + {ct}\" not in pt:\n",
    "                    prev.metadata[\"section_title\"] = f\"{pt} + {ct}\"\n",
    "                continue\n",
    "        compacted.append(d)\n",
    "\n",
    "   \n",
    "    total = len(compacted)\n",
    "    for i, d in enumerate(compacted):\n",
    "        d.metadata[\"subchunk_index\"] = i\n",
    "        d.metadata[\"subchunk_count\"] = total\n",
    "\n",
    "    return compacted\n",
    "\n",
    "\n",
    "normalized = normalize_chunks_by_section(all_chunks)\n",
    "\n",
    "\n",
    "def report(docs: Iterable[Document]):\n",
    "    for d in docs:\n",
    "        n_tok = tokenize_len(d.page_content)\n",
    "        print(\"=\"*60)\n",
    "        print(f\"{d.metadata.get('section_title')} | idx {d.metadata.get('subchunk_index')}/{d.metadata.get('subchunk_count')}\")\n",
    "        print(f\"{n_tok} tokens | {len(d.page_content)} chars\")\n",
    "        print((d.page_content[:] + \"\") if len(d.page_content) > 220 else d.page_content)\n",
    "\n",
    "report(normalized)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3fe6557-16dd-4953-83f0-106650f2513d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "15e87575-d55f-4b89-81ed-aa73f3b2d123",
   "metadata": {},
   "source": [
    "# Embeding menager za embeding na chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "574abc35-9350-437f-8fd1-18e7ed473cfe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading embedding model: NeuML/pubmedbert-base-embeddings\n",
      "Model loaded successfully. Embedding dimension: 768\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<__main__.EmbeddingManager at 0x262bc92db10>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "class EmbeddingManager:\n",
    "    \"\"\"Handles document embedding generation using SentenceTransformer\"\"\"\n",
    "    \n",
    "    def __init__(self, model_name: str = \"NeuML/pubmedbert-base-embeddings\"):\n",
    "        \"\"\"\n",
    "        Initialize the embedding manager\n",
    "        \n",
    "        Args:\n",
    "            model_name: HuggingFace model name for sentence embeddings\n",
    "        \"\"\"\n",
    "        self.model_name = model_name\n",
    "        self.model = None\n",
    "        self._load_model()\n",
    "\n",
    "    def _load_model(self):\n",
    "        \"\"\"Load the SentenceTransformer model\"\"\"\n",
    "        try:\n",
    "            print(f\"Loading embedding model: {self.model_name}\")\n",
    "            self.model = SentenceTransformer(self.model_name)\n",
    "            print(f\"Model loaded successfully. Embedding dimension: {self.model.get_sentence_embedding_dimension()}\")\n",
    "        except Exception as e:\n",
    "            print(f\"Error loading model {self.model_name}: {e}\")\n",
    "            raise\n",
    "\n",
    "    def generate_embeddings(self, texts: List[str]) -> np.ndarray:\n",
    "        \"\"\"\n",
    "        Generate embeddings for a list of texts\n",
    "        \n",
    "        Args:\n",
    "            texts: List of text strings to embed\n",
    "            \n",
    "        Returns:\n",
    "            numpy array of embeddings with shape (len(texts), embedding_dim)\n",
    "        \"\"\"\n",
    "        if not self.model:\n",
    "            raise ValueError(\"Model not loaded\")\n",
    "        \n",
    "        print(f\"Generating embeddings for {len(texts)} texts...\")\n",
    "        embeddings = self.model.encode(texts, show_progress_bar=True)\n",
    "        print(f\"Generated embeddings with shape: {embeddings.shape}\")\n",
    "        return embeddings\n",
    "\n",
    "\n",
    "## initialize the embedding manager\n",
    "\n",
    "embedding_manager=EmbeddingManager()\n",
    "embedding_manager"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89990a07-7d94-4a62-8bd4-3fba5d4cab47",
   "metadata": {},
   "source": [
    "# Vectordb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "83f0c056-a067-4fda-8d3f-5fc2837a77ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "class VectorStore:\n",
    "    \"\"\"Manages document embeddings in a ChromaDB vector store\"\"\"\n",
    "    \n",
    "    def __init__(self, collection_name: str = \"medical_papers\", persist_directory: str = r'D:\\Users\\User\\VectorDBZaRagVNP\\data\\vector_store'):\n",
    "        \"\"\"\n",
    "        Initialize the vector store\n",
    "        \n",
    "        Args:\n",
    "            collection_name: Name of the ChromaDB collection\n",
    "            persist_directory: Directory to persist the vector store\n",
    "        \"\"\"\n",
    "        self.collection_name = collection_name\n",
    "        self.persist_directory = persist_directory\n",
    "        self.client = None\n",
    "        self.collection = None\n",
    "        self._initialize_store()\n",
    "\n",
    "    def _initialize_store(self):\n",
    "        \"\"\"Initialize ChromaDB client and collection\"\"\"\n",
    "        try:\n",
    "            # Create persistent ChromaDB client\n",
    "            os.makedirs(self.persist_directory, exist_ok=True)\n",
    "            self.client = chromadb.PersistentClient(path=self.persist_directory)\n",
    "            \n",
    "            # Get or create collection\n",
    "            self.collection = self.client.get_or_create_collection(\n",
    "                name=self.collection_name,\n",
    "                metadata={\"description\": \"Medical papers chunks embedding for RAG\"}\n",
    "            )\n",
    "            print(f\"Vector store initialized. Collection: {self.collection_name}\")\n",
    "            print(f\"Existing documents in collection: {self.collection.count()}\")\n",
    "            \n",
    "        except Exception as e:\n",
    "            print(f\"Error initializing vector store: {e}\")\n",
    "            raise\n",
    "    def empty_vector_store(self):\n",
    "        \n",
    "        self.client.delete_collection(self.collection_name)\n",
    "       \n",
    "        self.collection = self.client.get_or_create_collection(\n",
    "            name=self.collection_name,\n",
    "            metadata={\"description\": \"Medical papers chunks embedding for RAG\"}\n",
    "        )\n",
    "        print(\"Collection dropped and recreated. Count:\", self.collection.count())\n",
    "\n",
    "    def collection_count():\n",
    "         print(f\"Existing documents in collection: {self.collection.count()}\")\n",
    "        \n",
    "    def add_documents(self, chunks: List[Any], embeddings: np.ndarray):\n",
    "        \"\"\"\n",
    "        Add documents and their embeddings to the vector store\n",
    "        \n",
    "        Args:\n",
    "            chunks: chunks of a medical paper \n",
    "            embeddings: Corresponding embeddings for the documents\n",
    "        \"\"\"\n",
    "        if len(chunks) != len(embeddings):\n",
    "            raise ValueError(\"Number of documents must match number of embeddings\")\n",
    "        \n",
    "        print(f\"Adding {len(chunks)} documents to vector store...\")\n",
    "        \n",
    "        \n",
    "        ids = []\n",
    "        metadatas = []\n",
    "        documents_text = []\n",
    "        embeddings_list = []\n",
    "        \n",
    "        for i, (chunk, embedding) in enumerate(zip(chunks, embeddings)):\n",
    "           \n",
    "            chunk_id = f\"chunk_{uuid.uuid4().hex[:8]}_{i}\"\n",
    "            ids.append(chunk_id)\n",
    "            \n",
    "           \n",
    "            metadata = dict()\n",
    "           \n",
    "            metadata['content_length'] = len(chunk)\n",
    "            metadatas.append(metadata)\n",
    "            \n",
    "          \n",
    "            documents_text.append(chunk)\n",
    "            \n",
    "          \n",
    "            embeddings_list.append(embedding.tolist())\n",
    "        \n",
    "        \n",
    "        try:\n",
    "            self.collection.add(\n",
    "                ids=ids,\n",
    "                embeddings=embeddings_list,\n",
    "                metadatas=metadatas,\n",
    "                documents=documents_text\n",
    "            )\n",
    "            print(f\"Successfully added {len(chunks)} documents to vector store\")\n",
    "            print(f\"Total documents in collection: {self.collection.count()}\")\n",
    "            \n",
    "        except Exception as e:\n",
    "            print(f\"Error adding documents to vector store: {e}\")\n",
    "            raise\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "7a4dafb3-9256-4a5d-88d6-8b1200776582",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Vector store initialized. Collection: medical_papers\n",
      "Existing documents in collection: 170\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<__main__.VectorStore at 0x262c4809fd0>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorstore=VectorStore()\n",
    "vectorstore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "419f0f00-b9c1-4a83-8903-adc7e9b8e850",
   "metadata": {},
   "outputs": [],
   "source": [
    "# vectorstore.empty_vector_store()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "1c1768c4-ae24-49e2-a76f-338e1609092d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating embeddings for 53 texts...\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "02078ec2193c4d348092185039c0b246",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Batches:   0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generated embeddings with shape: (53, 768)\n",
      "Adding 53 documents to vector store...\n",
      "Successfully added 53 documents to vector store\n",
      "Total documents in collection: 170\n"
     ]
    }
   ],
   "source": [
    "texts=[n.page_content for n in normalized]\n",
    "embeddings=embedding_manager.generate_embeddings(texts)\n",
    "vectorstore.add_documents(texts,embeddings)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b68a0662-0c6d-4730-9e78-a0dc141ce575",
   "metadata": {},
   "source": [
    "# RAG Retriever"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "4391ccd7-28db-4286-b73f-ade2a3b03bc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "class RAGRetriever:\n",
    "    \"\"\"Handles query-based retrieval from the vector store\"\"\"\n",
    "    \n",
    "    def __init__(self, vector_store: VectorStore, embedding_manager: EmbeddingManager):\n",
    "        \"\"\"\n",
    "        Initialize the retriever\n",
    "        \n",
    "        Args:\n",
    "            vector_store: Vector store containing document embeddings\n",
    "            embedding_manager: Manager for generating query embeddings\n",
    "        \"\"\"\n",
    "        self.vector_store = vector_store\n",
    "        self.embedding_manager = embedding_manager\n",
    "\n",
    "    def retrieve(self, query: str, top_k: int = 5, score_threshold: float = 0.0) -> List[Dict[str, Any]]:\n",
    "        \"\"\"\n",
    "        Retrieve relevant documents for a query\n",
    "        \n",
    "        Args:\n",
    "            query: The search query\n",
    "            top_k: Number of top results to return\n",
    "            score_threshold: Minimum similarity score threshold\n",
    "            \n",
    "        Returns:\n",
    "            List of dictionaries containing retrieved documents and metadata\n",
    "        \"\"\"\n",
    "        print(f\"Retrieving paper chunks for query: '{query}'\")\n",
    "        print(f\"Top K: {top_k}, Score threshold: {score_threshold}\")\n",
    "        \n",
    "        \n",
    "        query_embedding = self.embedding_manager.generate_embeddings([query])[0]\n",
    "        \n",
    "       \n",
    "        try:\n",
    "            results = self.vector_store.collection.query(\n",
    "                query_embeddings=[query_embedding.tolist()],\n",
    "                n_results=top_k\n",
    "            )\n",
    "           \n",
    "            retrieved_docs = []\n",
    "            \n",
    "            if results['documents'] and results['documents'][0]:\n",
    "                papers = results['documents'][0]\n",
    "                metadatas = results['metadatas'][0]\n",
    "                distances = results['distances'][0]\n",
    "                ids = results['ids'][0]\n",
    "                \n",
    "                for i, (doc_id, paper, metadata, distance) in enumerate(zip(ids, papers, metadatas, distances)):\n",
    "                    retrieved_docs.append({\n",
    "                        'id': doc_id,\n",
    "                        'content': paper,\n",
    "                        'metadata': metadata,\n",
    "                        \n",
    "                        'distance': distance,\n",
    "                        'rank': i + 1\n",
    "                    })\n",
    "                \n",
    "                print(f\"Retrieved {len(retrieved_docs)} paper chunks (after filtering)\")\n",
    "            else:\n",
    "                print(\"No documents found\")\n",
    "            \n",
    "            return retrieved_docs\n",
    "            \n",
    "        except Exception as e:\n",
    "            print(f\"Error during retrieval: {e}\")\n",
    "            return []\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f8676765-94be-4093-a759-3341dc898675",
   "metadata": {},
   "outputs": [],
   "source": [
    "rag_retriever=RAGRetriever(vectorstore,embedding_manager)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "8e5d6534-5121-461f-b5d6-ecc5466c6409",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<__main__.RAGRetriever at 0x262f874bed0>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rag_retriever"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "379f4bd3-1024-41be-bf8c-6e518df18155",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving paper chunks for query: 'How does DEA affect the Keap1Nrf2 proteinprotein interaction (PPI) and GSK3 activity in ALD model cells?'\n",
      "Top K: 5, Score threshold: 0.0\n",
      "Generating embeddings for 1 texts...\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f2694e45f4824f7f9d6542d3831d3ae7",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Batches:   0%|          | 0/1 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generated embeddings with shape: (1, 768)\n",
      "Retrieved 5 paper chunks (after filtering)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[{'id': 'chunk_22b35c5c_25',\n",
       "  'content': 'Recent studies on ALD are scarce compared to research on other liver diseases, and there are still no highly satisfactory therapeutic options for ALD. To our knowledge, no dual inhibitor of the Keap1Nrf2 PPI and GSK3 to treat alcohol-induced liver injury has been reported. Compound DEA (1) potently inhibits Keap1-Nrf2 binding and GSK3 activity, enhancing ARE activity in vitro, and exhibits low cytotoxicity in various cell lines. Compound DEA (1) releases Nrf2 by binding with Keap1 and GSK3, enhancing its thermal stability and reducing protein degradation. Compound DEA (1) regulates antioxidant effects by targeting Keap1 and GSK3, as evidenced by decreased expression of antioxidant proteins in cells with knockdown of either Keap1 or GSK3.',\n",
       "  'metadata': {'content_length': 753},\n",
       "  'distance': 104.04415893554688,\n",
       "  'rank': 1},\n",
       " {'id': 'chunk_f92d20e1_8',\n",
       "  'content': 'The ability of DEA (1) to disrupt the Keap1Nrf2 binding in cellulo was monitored through a CoIP experiment using human normal liver cell line LO2 cells (A). After treatment with DEA (1) (10 M) or ML334 (10 M) for 8 h, there was a 45% reduction of Keap1 co-precipitated with Nrf2 for DEA (1) as compared to 12% inhibition for ML334, indicating that DEA (1) could more effectively inhibit the Keap1Nrf2 interaction compared to ML334 in living cells (B). We investigated the ability of DEA (1) to target Keap1, Nrf2, and GSK3 in the cellular environment using CETSA. LO2 cell lysates were incubated with 10 M of DEA (1) at room temperature for 30 min. Then, aliquots were heated individually at different temperatures and the protein in the soluble fraction was quantified by WB. Compound DEA (1) significantly stabilized Keap1 (Tm: ca. 4.5 C) and GSK3 (Tm: ca. 3.9 C), while having no observable effect on the thermal stability of Nrf2 and actin (C,D). This result suggests that DEA (1) can bind with Keap1 and GSK3, even within the complicated environment of cell lysates.',\n",
       "  'metadata': {'content_length': 1086},\n",
       "  'distance': 107.53489685058594,\n",
       "  'rank': 2},\n",
       " {'id': 'chunk_2c6dbda8_2',\n",
       "  'content': 'In addition, Nrf2 activation increases the expression levels of nuclear respiratory factor-1 (Nrf-1) and peroxisome proliferator-activated receptor  coactivator 1 (PGC1), which in turn modulates the expression of mitochondrial respiratory subunits and translational components [16]. The genes activated by the antioxidant responsive element (ARE) protect cells against any damage caused by reactive oxygen species (ROS) [17]. In addition to Keap1, glycogen synthase kinase 3 (GSK3) is another negative regulator of Nrf2 that acts to downregulate the antioxidant stress response [18,19]. GSK3 promotes the nuclear exclusion and degradation of Nrf2 in a Keap1-independent fashion in stressed or injured cells with active Nrf2, thereby reducing the degree of protection conferred by Nrf2 activity [20,21]. Therefore, increasing the Nrf2 antioxidant activity via targeting Keap1 or GSK3 could be a potential approach for treating ALD [22]. Most pharmacological approaches towards Nrf2 activation are focused on the inhibition of Keap1 [23,24]. Several Keap1 inhibitors have been reported in the last decade, and the vast majority of these covalently modify cysteine residues in Keap1 [25]. Covalently targeting cysteines in Keap1 may lack selectivity due to the presence of other reactive cysteine residues in the cell, resulting in adverse side effects [26,27]. However, because cysteine residues are abundant in cells, the safety and specificity of covalent drugs are a concern [28]. Recently, proteinprotein interaction (PPI) inhibitors of Keap1 and Nrf2, developed based on the X-ray structure of Keap1, have emerged as a new class of Nrf2 activators [29,30,31].',\n",
       "  'metadata': {'content_length': 1670},\n",
       "  'distance': 120.5280532836914,\n",
       "  'rank': 3},\n",
       " {'id': 'chunk_84a3b074_10',\n",
       "  'content': 'Therefore, DEA (1) exerts its antioxidant effects by targeting both Keap1 and GSK3. Inhibition of the Keap1Nrf2 interaction is known to increase Nrf2 accumulation in the nucleus [47]. Meanwhile, GSK3 is also a negative regulator of Nrf2 accumulation by acting independently of Keap1 [49]. Therefore, we evaluated the Nrf2 level in the cytoplasm and nucleus of LO2 cells via WB after the treatment of cells with DEA (1) (10 M) for 8 h. As shown in A, DEA (1) significantly increased Nrf2 translocation into the nucleus in a dose-dependent manner. GSK3 knockdown also phenocopied compound DEA (1) treatment at inducing Nrf2 nuclear accumulation without influencing Keap1 level (B). In summary, DEA (1) disrupts the Keap1Nrf2 PPI and inhibits GSK3 to increase the translocation of liberated Nrf2 into the nucleus, thereby presumably increasing ARE transcriptional activity.',\n",
       "  'metadata': {'content_length': 877},\n",
       "  'distance': 121.01292419433594,\n",
       "  'rank': 4},\n",
       " {'id': 'chunk_8a562a1a_9',\n",
       "  'content': 'To verify whether Keap1 or GSK3 are direct targets of DEA (1), knockdown assays were performed. NQO1 and HO1 are target antioxidant genes of Nrf2 that increase cytoprotection against oxidative stress [48] As expected, Keap1 knockdown in LO2 cells induced a noticeable rise of the expression levels of both Nrf2 and its gene targets, NQO1 and HO1 (A,B). However, treating Keap1 knockdown cells with DEA (1) led to no further rises in the NQO1 and HO1 level, in contrast to the control cells, where DEA (1) treatment produced noticeable increases of NQO1 and HO1. These results suggest that DEA (1) activates NQO1 and HO1 through targeting Keap1. Similarly, GSK3 knockdown induced Nrf2 and downstream antioxidant protein expression but without influencing Keap1 protein level (C,D). In addition, DEA (1) had a diminished effect on NQO1 and HO1 level in GSK3 knockdown cells as compared to the control cells. Finally, we knocked down GSK3 and Keap1 at the same time to verify the dual binding mechanism of DEA (1) on GSK3 and Keap1 (E,F). The results showed that treating the GSK3 and Keap1 knockdown cells with DEA (1) did not cause a further significant induction of NQO1 and HO1 expression. Evidence for the hypothesis that DEA (1) acts through targeting both Keap1 and GSK3 comes from considering the Nrf2 level. In either the Keap1 knockdown or GSK3 knockdown cells, DEA (1) treatment resulted in slight increases of Nrf2 expression relative to DMSO. However, in the double Keap1 and GSK3 knockdown cells, treatment with DEA (1) did not lead to increases of Nrf2 expression. These results suggest that DEA (1) treatment mimicked the effects of double knockdown in increasing Nrf2 expression.',\n",
       "  'metadata': {'content_length': 1709},\n",
       "  'distance': 127.15357971191406,\n",
       "  'rank': 5}]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rag_retriever.retrieve(\"How does DEA affect the Keap1Nrf2 proteinprotein interaction (PPI) and GSK3 activity in ALD model cells?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e99e4641-115e-4a78-9d47-359adbb1db9e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09ab6f58-c725-471d-a63e-1412275c95a5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "decaf393-4ceb-4530-aa42-182c4107b1d3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "d386a934-b8f7-49a0-95ea-57007da617fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "from typing import Any, List\n",
    "from langchain.prompts import PromptTemplate\n",
    "from langchain.schema import HumanMessage\n",
    "\n",
    "\n",
    "try:\n",
    "    from langchain_community.chat_models import ChatOllama\n",
    "except ImportError:\n",
    "    \n",
    "    from langchain_ollama import ChatOllama\n",
    "\n",
    "\n",
    "class LocalOllamaLLM:\n",
    "    def __init__(\n",
    "        self,\n",
    "        model_name: str = \"mistral:7b-instruct-v0.3-q4_0\", # 32,768 tokens.\n",
    "        base_url: str = \"http://localhost:11434\",\n",
    "        temperature: float = 0.1,\n",
    "        max_tokens: int = 1024,\n",
    "        num_ctx: int = 8192,\n",
    "        extra_model_kwargs: dict | None = None,\n",
    "    ):\n",
    "        \"\"\"\n",
    "        Initialize a local Ollama LLM.\n",
    "\n",
    "        Args:\n",
    "            model_name: Ollama model tag you've pulled locally.\n",
    "            base_url: Ollama server URL.\n",
    "            temperature: Sampling temperature.\n",
    "            max_tokens: Max tokens to generate (mapped to num_predict).\n",
    "            num_ctx: Context window to request from Ollama (if supported by model).\n",
    "            extra_model_kwargs: Extra model kwargs to pass through to Ollama.\n",
    "        \"\"\"\n",
    "        self.model_name = model_name\n",
    "        self.base_url = base_url\n",
    "        self.temperature = temperature\n",
    "        self.max_tokens = max_tokens\n",
    "\n",
    "        model_kwargs = {\"num_ctx\": num_ctx}\n",
    "        if extra_model_kwargs:\n",
    "            model_kwargs.update(extra_model_kwargs)\n",
    "\n",
    "        \n",
    "        self.llm = ChatOllama(\n",
    "            model=self.model_name,\n",
    "            temperature=self.temperature,\n",
    "            base_url=self.base_url,\n",
    "            \n",
    "            num_predict=self.max_tokens,\n",
    "            \n",
    "            model_kwargs=model_kwargs,\n",
    "        )\n",
    "\n",
    "        print(f\"Initialized local Ollama LLM with model: {self.model_name} at {self.base_url}\")\n",
    "\n",
    "    def generate_response(self, query: str, context: str, max_length: int = 500) -> str:\n",
    "        \"\"\"\n",
    "        Generate response using retrieved context (structured prompt).\n",
    "        \"\"\"\n",
    "        prompt_template = PromptTemplate(\n",
    "            input_variables=[\"context\", \"question\"],\n",
    "            template=(\n",
    "                \"You are a helpful AI assistant. Use the following context to answer the question accurately and concisely.\\n\\n\"\n",
    "                \"Context:\\n{context}\\n\\n\"\n",
    "                \"Question: {question}\\n\\n\"\n",
    "                \"Answer: Provide a clear and informative answer based on the context above. \"\n",
    "                \"If the context doesn't contain enough information to answer the question, say so.\"\n",
    "            ),\n",
    "        )\n",
    "        formatted_prompt = prompt_template.format(context=context, question=query)\n",
    "\n",
    "        try:\n",
    "            messages = [HumanMessage(content=formatted_prompt)]\n",
    "            \n",
    "            response = self.llm.invoke(messages)\n",
    "            \n",
    "            return response.content[:max_length]\n",
    "        except Exception as e:\n",
    "            return f\"Error generating response: {str(e)}\"\n",
    "\n",
    "    def generate_response_simple(self, query: str, context: str) -> str:\n",
    "        \"\"\"\n",
    "        Simple response generation without complex prompting.\n",
    "        \"\"\"\n",
    "        simple_prompt = f\"\"\"Based on this context: {context}\n",
    "\n",
    "        Question: {query}\n",
    "        \n",
    "        Answer:\"\"\"\n",
    "        try:\n",
    "            messages = [HumanMessage(content=simple_prompt)]\n",
    "            response = self.llm.invoke(messages)\n",
    "            return response.content\n",
    "        except Exception as e:\n",
    "            return f\"Error: {str(e)}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "4a748eaa-8d02-44cc-b84f-d3662ca1832b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initialized local Ollama LLM with model: mistral:7b-instruct-v0.3-q4_0 at http://localhost:11434\n",
      "Local Ollama LLM initialized successfully!\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\User\\AppData\\Local\\Temp\\ipykernel_19152\\3365943129.py:44: LangChainDeprecationWarning: The class `ChatOllama` was deprecated in LangChain 0.3.1 and will be removed in 1.0.0. An updated version of the class exists in the :class:`~langchain-ollama package and should be used instead. To use it run `pip install -U :class:`~langchain-ollama` and import as `from :class:`~langchain_ollama import ChatOllama``.\n",
      "  self.llm = ChatOllama(\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    ollama_llm = LocalOllamaLLM()\n",
    "    print(\"Local Ollama LLM initialized successfully!\")\n",
    "except Exception as e:\n",
    "    print(f\"Initialization error: {e}\")\n",
    "    ollama_llm = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "b1a01ae7-205e-462d-a376-9c559278c771",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain.schema import HumanMessage\n",
    "\n",
    "def rag_simple(\n",
    "    query: str,\n",
    "    retriever,\n",
    "    llm,\n",
    "    top_k: int = 3,\n",
    "    return_debug: bool = False,\n",
    ") -> str | Tuple[str, dict]:\n",
    "    \"\"\"\n",
    "    Retrieve top_k chunks, build a concise prompt, and get an answer from the LLM.\n",
    "\n",
    "    Works with:\n",
    "      - LocalOllamaLLM wrapper you defined (has .generate_response and .llm.invoke)\n",
    "      - Any LangChain chat model exposing .invoke([...])\n",
    "\n",
    "    If return_debug=True, returns (answer, debug_info) where debug_info includes the\n",
    "    used context and the raw retrieved docs.\n",
    "    \"\"\"\n",
    "   \n",
    "    results: List[Dict[str, Any]] = retriever.retrieve(query, top_k=top_k)\n",
    "    context = \"\\n\\n\".join([doc.get(\"content\", \"\") for doc in results if doc.get(\"content\")]) if results else \"\"\n",
    "    if not context.strip():\n",
    "        out = \"No relevant context found to answer the question.\"\n",
    "        return (out, {\"context\": \"\", \"results\": results}) if return_debug else out\n",
    "\n",
    "    \n",
    "    prompt = (\n",
    "        \"Use the following context to answer the question accurately and concisely.\\n\\n\"\n",
    "        f\"Context:\\n{context}\\n\\n\"\n",
    "        f\"Question: {query}\\n\\n\"\n",
    "        \"Answer:\"\n",
    "    )\n",
    "\n",
    "    \n",
    "    try:\n",
    "        if hasattr(llm, \"generate_response\"):\n",
    "            \n",
    "            answer = llm.generate_response(query=query, context=context)\n",
    "        elif hasattr(llm, \"invoke\"):\n",
    "            \n",
    "            answer = llm.invoke([HumanMessage(content=prompt)]).content\n",
    "        elif hasattr(llm, \"llm\") and hasattr(llm.llm, \"invoke\"):\n",
    "            \n",
    "            answer = llm.llm.invoke([HumanMessage(content=prompt)]).content\n",
    "        else:\n",
    "            raise TypeError(\"Unsupported llm object: expected a LangChain chat model or LocalOllamaLLM.\")\n",
    "    except Exception as e:\n",
    "        answer = f\"Error generating response: {e}\"\n",
    "\n",
    "    return (answer, {\"context\": context, \"results\": results}) if return_debug else answer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "f2e5edd2-23b9-41a0-ae39-7a633d661f78",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving paper chunks for query: 'What is basketball?'\n",
      "Top K: 5, Score threshold: 0.0\n",
      "Generating embeddings for 1 texts...\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2861e8ae7fdf42b6ba51811a5c35233d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Batches:   0%|          | 0/1 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generated embeddings with shape: (1, 768)\n",
      "Retrieved 5 paper chunks (after filtering)\n",
      " The provided context does not contain any information about basketball.\n"
     ]
    }
   ],
   "source": [
    "q = \"What is dehydroeburicoic acid and from which natural source is it isolated?\"\n",
    "# q = \"What is basketball?\"\n",
    "answer, dbg = rag_simple(q, rag_retriever, ollama_llm, top_k=5, return_debug=True)\n",
    "print(answer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52f05ebf-3de1-4b1d-ba8c-62d172fd4a62",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27903477-0e79-4c54-8f13-0ffdad12533c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2108ca3-58de-4fd8-a6e6-9204d5e1aa00",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (vnp_llm_env)",
   "language": "python",
   "name": "vnp_llm_env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
